---
title: "Stinson-STE-Microbes"
author: "Jessica Bullington"
date: "2024-03-23"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Part 0: Set up

## 1. Load libraries
```{r}
library(phyloseq)
library(ggplot2)
library(vegan)
library(ape)
library(dplyr)
library(viridis)
library(tidyr)
library(genefilter)
library(reshape2)
library(pheatmap)
library(RColorBrewer)
library(GGally)
library(Hmisc) 
library(DESeq2)


```

## 2. Load data
```{r}
ps = readRDS("new_ps_Stinson_2024.03.14.rds") # load phyloseq object
#ps_pw_sw = subset_samples(ps, Type %in% c("Porewater", "Seawater")) # subset to porewater and seawater samples
```

```{r add tide regime}

sample_data(ps)$Tide.Regime[sample_data(ps)$Date_Tide == "Feb-23 L"] <- "Neap"

sample_data(ps)$Tide.Regime = case_when(
  sample_data(ps)$Date_Tide %in% c("Feb-23 L", "Feb-24 H", "Feb-25 H", "Feb-25 L", "Oct-14 M", "Oct-17 M") ~ "Neap",
  sample_data(ps)$Date_Tide %in% c("Feb-27 H", "Feb-27 L", "Mar-01 H", "Mar-01 L", "Mar-03 H", "Mar-03 L", "Oct-11 M") ~ "Spring")

meta = data.frame(sample_data(ps))


chem$cluster = case_when( # cluster region

  chem$Well %in% c("Well 1", "Well 2", "Well 3") & chem$Elevation > -0.2 ~ "Shallow",
  chem$Well %in% c("Well 1", "Well 2", "Well 3") & chem$Elevation < -0.2 & chem$Elevation > -0.5 ~ "Intermediate",
  chem$Well %in% c("Well 1", "Well 2", "Well 3") & chem$Elevation < -0.5 ~ "Deep",
  
  chem$Well == "Well 4" ~ "Well 4",
  chem$Well == "Seawater"  ~ "Seawater"
  
)


```

## 3. Basic data assessment
```{r reads per sample}
# number of reads per sample
reads = data.frame(sample_sums(ps_pw_sw))

mdata = data.frame(sample_data(ps_pw_sw))

# ggplot histogram of reads
ggplot(mdata, aes(x = Reads)) +
  geom_histogram(binwidth = 5000) +
  theme_bw() +
  labs(x = "\nNumber of reads", y = "Number of samples\n") +
  theme(axis.text.x = element_text(hjust = 0.6)) # default is hjust = 0.5
  

# ggsave("Hist_reads_2024.03.22.png", plot = last_plot(),
#       units = "cm", width = 15, height = 11, dpi = 300)
```



# Part 1: Assess and trim environmental covariates

## 1. Extract sample data from ps object
```{r}
env = data.frame(sample_data(ps)) # extract sample metadata
env = subset(env, Type != "Sand") # remove sand samples
env$DO_uM = env$DO *  31.2512 # convert DO to ÂµM
env$Phosphate[env$Phosphate < 0] = 0 # assign negative values to zero
env2 = subset(env, Nitrite < 100) # remove outlier nitrite value
```

## 2. Trim sample data to only environmental data
```{r}
cols_keep = c("DO_uM", "Salinity", "Temperature", "pH", "DOC", "Nitrite", "Nitrate", "Ammonium", "Phosphate", "DOC", "TN", "CobleA",  "CobleB",  "CobleC", "CobleM", "CobleT", "Fpeak",  "Stedmon_D", "Optical_Brighteners", "Lignin",  "BIX",  "HIX",  "FI", "M_to_C" )

trim.env = env2[, (colnames(env2) %in% cols_keep)]
```

## 3. Assess pairwise correlations
```{r}
pl = ggpairs(trim.env) # ignore warnings about missing values

# ggsave("Chem_ggpairs_untrans_2024.03.22.png", plot = pl,
#        units = "cm", width = 40, height = 40, dpi = 300)

# assess bimodal distributions, caused by campaign, well, group?

combo.env = data.frame(Campaign = env2$Campaign, Well = env2$Well, Group = env2$Group, trim.env)

pl1 = ggpairs(combo.env, columns = 4:ncol(combo.env), ggplot2::aes(colour=Campaign))
# ggsave("Chem_ggpairs_untrans_Campaign_2024.03.22.png", plot = pl1,
#        units = "cm", width = 60, height = 40, dpi = 300)

pl2 = ggpairs(combo.env, columns = 4:ncol(combo.env), ggplot2::aes(colour=Well))
# ggsave("Chem_ggpairs_untrans_Well_2024.03.22.png", plot = pl2,
#        units = "cm", width = 60, height = 60, dpi = 300)

pl3 = ggpairs(combo.env, columns = 4:ncol(combo.env), ggplot2::aes(colour=Group))
# ggsave("Chem_ggpairs_untrans_Group_2024.03.22.png", plot = pl3,
#        units = "cm", width = 60, height = 60, dpi = 300)

# separate seasons
combo.env.feb = subset(combo.env, Campaign == "February 2022" & Well != "Seawater")
pl4 = ggpairs(combo.env.feb, columns = 4:ncol(combo.env.feb), ggplot2::aes(colour=Well))
# ggsave("Chem_ggpairs_untrans_Feb_Well_2024.03.22.png", plot = pl4,
#        units = "cm", width = 60, height = 40, dpi = 300)

combo.env.oct = subset(combo.env, Campaign == "October 2022")
pl5 = ggpairs(combo.env.oct, columns = 4:ncol(combo.env.oct))

```

## 4. Generate correlation matrix
```{r}
# method 1
res = cor(trim.env, method = "pearson", use = "complete.obs")

# method 2
res2 <- rcorr(as.matrix(trim.env))
```

## 5. Visualize correlation matrix
```{r full matrix}
# define color scale
col = rev(brewer.pal(10, "RdBu"))

# method 1 - simple heatmap
# png('Chemistry_Correlation_Matrix1_2024.03.22.png', units="in", width=10, height=10, res=300)
heatmap(x = res, col = rev(col), symm = TRUE)
# dev.off()

# method 2 - simple heatmap
# png('Chemistry_Correlation_Matrix2_2024.03.22.png', units="in", width=10, height=10, res=300)
heatmap(x = res2$r, col = col, symm = TRUE)
# dev.off()

# method 1 - pheatmap
xx <- pheatmap( 
  res,
  color = col,
  breaks = seq(-1, 1, by = 0.2),
  clustering_method = "complete",  # Clustering method
  display_numbers = T,  # Display numbers in cells
  number_color = "grey35",
  show_rownames = T,
  legend = TRUE,  # Display legend
  legend_breaks = seq(-1, 1, by = 0.2)
)

save_pheatmap_pdf <- function(x, filename, width=10, height=10) {
   stopifnot(!missing(x))
   stopifnot(!missing(filename))
   pdf(filename, width=width, height=height)
   grid::grid.newpage()
   grid::grid.draw(x$gtable)
   dev.off()
}

# save_pheatmap_pdf(xx, "Chem_corr_untrans_rowNames_2024.03.23.pdf")

xx <- pheatmap( 
  res,
  color = col,
  breaks = seq(-1, 1, by = 0.2),
  clustering_method = "complete",  # Clustering method
  display_numbers = T,  # Display numbers in cells
  number_color = "grey35",
  show_rownames = F,
  legend = TRUE,  # Display legend
  legend_breaks = seq(-1, 1, by = 0.2)
)

# save_pheatmap_pdf(xx, "Chem_corr_untrans_2024.03.23.pdf")


```

```{r reduced matrix}
# assess and remove covariates above 0.8 correlation coefficient # use white box in preview
cols_keep_red = c("DO_uM", "Salinity", "Temperature", "pH", "DOC", "Nitrite", "Nitrate", "Ammonium", "Phosphate", "HIX", "Fpeak" )
trim.env_red = env2[, (colnames(env2) %in% cols_keep_red)]

# reduced correlation matrix
res_red = cor(trim.env_red, method = "pearson", use = "complete.obs")

# reduced pheatmap
xx_red <- pheatmap( 
  res_red,
  color = col,
  breaks = seq(-1, 1, by = 0.2),
  clustering_method = "complete",  # Clustering method
  display_numbers = T,  # Display numbers in cells
  number_color = "grey35",
  show_rownames = T,
  legend = TRUE,  # Display legend
  legend_breaks = seq(-1, 1, by = 0.2)
)

# save_pheatmap_pdf(xx_red, "Chem_corr_Reduced_2024.03.23.pdf", width=6, height=5)
```


## 6. Repeat for each well individually
```{r well 1}
# Well 1
env_W1 = subset(env2, Well == "Well 1")
cols_keep_W1 = c("DO_uM", "Salinity", "Temperature", "pH", "DOC", "Nitrite", "Nitrate", "Ammonium", "Phosphate", "DOC", "TN", "CobleA",  "CobleB",  "CobleC", "CobleM", "CobleT", "Fpeak",  "Stedmon_D", "Optical_Brighteners", "Lignin",  "BIX",  "HIX",  "FI", "M_to_C" )
trim.env_W1 = env_W1[, (colnames(env_W1) %in% cols_keep_W1)]
res_W1 = cor(trim.env_W1, method = "pearson", use = "complete.obs")
xx_W1 <- pheatmap( 
  res_W1,
  color = col,
  breaks = seq(-1, 1, by = 0.2),
  clustering_method = "complete",  # Clustering method
  display_numbers = T,  # Display numbers in cells
  number_color = "grey35",
  show_rownames = T,
  legend = TRUE,  # Display legend
  legend_breaks = seq(-1, 1, by = 0.2)
)

# save_pheatmap_pdf(xx_W1, "Chem_corr_Well_1_2024.03.22.pdf", width=12, height=10)

```

```{r well 1 reduced}
# Well 1 - reduced
cols_keep_red_W1 = c("DO_uM", "Salinity", "Temperature", "pH", "DOC", "Nitrite", "Nitrate", "Ammonium", "Phosphate",  "HIX", "Fpeak")
trim.env_red_W1 = env_W1[, (colnames(env_W1) %in% cols_keep_red_W1)]
res_red_W1 = cor(trim.env_red_W1, method = "pearson", use = "complete.obs")
xx_red_W1 <- pheatmap( 
  res_red_W1,
  color = col,
  breaks = seq(-1, 1, by = 0.2),
  clustering_method = "complete",  # Clustering method
  display_numbers = T,  # Display numbers in cells
  number_color = "grey35",
  show_rownames = T,
  legend = TRUE,  # Display legend
  legend_breaks = seq(-1, 1, by = 0.2)
)

# save_pheatmap_pdf(xx_red_W1, "Chem_corr_W1_Reduced_2024.03.23.pdf", width=7, height=6)

```

```{r well 2}
# Well 2
env_W2 = subset(env2, Well == "Well 2")
cols_keep_W2 = c("DO_uM", "Salinity", "Temperature", "pH", "DOC", "Nitrite", "Nitrate", "Ammonium", "Phosphate", "DOC", "TN", "CobleA",  "CobleB",  "CobleC", "CobleM", "CobleT", "Fpeak",  "Stedmon_D", "Optical_Brighteners", "Lignin",  "BIX",  "HIX",  "FI", "M_to_C" )
trim.env_W2 = env_W2[, (colnames(env_W2) %in% cols_keep_W2)]
res_W2 = cor(trim.env_W2, method = "pearson", use = "complete.obs")
xx_W2 <- pheatmap( 
  res_W2,
  color = col,
  breaks = seq(-1, 1, by = 0.2),
  clustering_method = "complete",  # Clustering method
  display_numbers = T,  # Display numbers in cells
  number_color = "grey35",
  show_rownames = T,
  legend = TRUE,  # Display legend
  legend_breaks = seq(-1, 1, by = 0.2)
)

# save_pheatmap_pdf(xx_W2, "Chem_corr_W2_2024.03.22.pdf", width=12, height=10)

```


```{r well 2 reduced}
# Well 2 - reduced
cols_keep_red_W2 = c("DO_uM", "Salinity", "Temperature", "DOC", "Nitrite", "Nitrate", "Ammonium", "Phosphate",  "HIX", "Fpeak" )
trim.env_red_W2 = env_W2[, (colnames(env_W2) %in% cols_keep_red_W2)]
res_red_W2 = cor(trim.env_red_W2, method = "pearson", use = "complete.obs")
xx_red_W2 <- pheatmap( 
  res_red_W2,
  color = col,
  breaks = seq(-1, 1, by = 0.2),
  clustering_method = "complete",  # Clustering method
  display_numbers = T,  # Display numbers in cells
  number_color = "grey35",
  show_rownames = T,
  legend = TRUE,  # Display legend
  legend_breaks = seq(-1, 1, by = 0.2)
)

# save_pheatmap_pdf(xx_red_W2, "Chem_corr_W2_Reduced_2024.03.23.pdf", width=7, height=6)

```


```{r well 3}
# Well 3
env_W3 = subset(env2, Well == "Well 3")
cols_keep_W3 = c("DO_uM", "Salinity", "Temperature", "pH", "DOC", "Nitrite", "Nitrate", "Ammonium", "Phosphate", "DOC", "TN", "CobleA",  "CobleB",  "CobleC", "CobleM", "CobleT", "Fpeak",  "Stedmon_D", "Optical_Brighteners", "Lignin",  "BIX",  "HIX",  "FI", "M_to_C" )
trim.env_W3 = env_W3[, (colnames(env_W3) %in% cols_keep_W3)]
res_W3 = cor(trim.env_W3, method = "pearson", use = "complete.obs")
xx_W3 <- pheatmap( 
  res_W3,
  color = col,
  breaks = seq(-1, 1, by = 0.2),
  clustering_method = "complete",  # Clustering method
  display_numbers = T,  # Display numbers in cells
  number_color = "grey35",
  show_rownames = T,
  legend = TRUE,  # Display legend
  legend_breaks = seq(-1, 1, by = 0.2)
)

# save_pheatmap_pdf(xx_W3, "Chem_corr_W3_2024.03.22.pdf", width=12, height=10)

```

```{r well 3 reduced}
# Well 3 - reduced
cols_keep_red_W3 = c("DO_uM", "Salinity", "Temperature", "pH", "DOC", "Nitrite", "Nitrate", "Ammonium", "Phosphate",  "HIX", "Fpeak" )
trim.env_red_W3 = env_W3[, (colnames(env_W3) %in% cols_keep_red_W3)]
res_red_W3 = cor(trim.env_red_W3, method = "pearson", use = "complete.obs")
xx_red_W3 <- pheatmap( 
  res_red_W3,
  color = col,
  breaks = seq(-1, 1, by = 0.2),
  clustering_method = "complete",  # Clustering method
  display_numbers = T,  # Display numbers in cells
  number_color = "grey35",
  show_rownames = T,
  legend = TRUE,  # Display legend
  legend_breaks = seq(-1, 1, by = 0.2)
)

# save_pheatmap_pdf(xx_red_W3, "Chem_corr_W3_Reduced_2024.03.23.pdf", width=7, height=6)

```

```{r well 4}
# Well 4
env_W4 = subset(env2, Well == "Well 4")
cols_keep_W4 = c("DO_uM", "Salinity", "Temperature", "pH", "DOC", "Nitrite", "Nitrate", "Ammonium", "Phosphate", "DOC", "TN", "CobleA",  "CobleB",  "CobleC", "CobleM", "CobleT", "Fpeak",  "Stedmon_D", "Optical_Brighteners", "Lignin",  "BIX",  "HIX",  "FI", "M_to_C" )
trim.env_W4 = env_W4[, (colnames(env_W4) %in% cols_keep_W4)]
res_W4 = cor(trim.env_W4, method = "pearson", use = "complete.obs")
xx_W4 <- pheatmap( 
  res_W4,
  color = col,
  breaks = seq(-1, 1, by = 0.2),
  clustering_method = "complete",  # Clustering method
  display_numbers = T,  # Display numbers in cells
  number_color = "grey35",
  show_rownames = T,
  legend = TRUE,  # Display legend
  legend_breaks = seq(-1, 1, by = 0.2)
)

# save_pheatmap_pdf(xx_W4, "Chem_corr_W4_2024.03.22.pdf", width=12, height=10)

```

```{r well 4 reduced}
# Well 4 - reduced
cols_keep_red_W4 = c("DO_uM", "Salinity", "Temperature", "pH",  "Nitrite", "Nitrate", "Ammonium", "Phosphate", "DOC", "HIX", "Fpeak" )
trim.env_red_W4 = env_W4[, (colnames(env_W4) %in% cols_keep_red_W4)]
res_red_W4 = cor(trim.env_red_W4, method = "pearson", use = "complete.obs")
xx_red_W4 <- pheatmap( 
  res_red_W4,
  color = col,
  breaks = seq(-1, 1, by = 0.2),
  clustering_method = "complete",  # Clustering method
  display_numbers = T,  # Display numbers in cells
  number_color = "grey35",
  show_rownames = T,
  legend = TRUE,  # Display legend
  legend_breaks = seq(-1, 1, by = 0.2)
)

# save_pheatmap_pdf(xx_red_W4, "Chem_corr_W4_Reduced_2024.03.23.pdf", width=7, height=6)


```




## 7. Repeat for each cluster group

```{r shallow}
# Shallow Group 
env_S = subset(env2, Group == "Shallow")

cols_keep_S = c("DO_uM", "Salinity", "Temperature", "pH", "DOC", "Nitrite", "Nitrate", "Ammonium", "Phosphate", "DOC", "TN", "CobleA",  "CobleB",  "CobleC", "CobleM", "CobleT", "Fpeak",  "Stedmon_D", "Optical_Brighteners", "Lignin",  "BIX",  "HIX",  "FI", "M_to_C" )

trim.env_S = env_S[, (colnames(env_S) %in% cols_keep_S)]
res_S = cor(trim.env_S, method = "pearson", use = "complete.obs")
xx_S <- pheatmap( 
  res_S,
  color = col,
  breaks = seq(-1, 1, by = 0.2),
  clustering_method = "complete",  # Clustering method
  display_numbers = T,  # Display numbers in cells
  number_color = "grey35",
  show_rownames = T,
  legend = TRUE,  # Display legend
  legend_breaks = seq(-1, 1, by = 0.2)
)

save_pheatmap_pdf(xx_W1, "Chem_corr_Shallow_2024.05.22.pdf", width=12, height=10)
```

```{r shallow reduced}
# Shallow Group - reduced
cols_keep_red_S = c("DO_uM", "Salinity", "Temperature", "pH", "DOC", "Nitrite", "Nitrate", "Ammonium", "Phosphate", "HIX", "Fpeak" )
trim.env_red_S = env_S[, (colnames(env_S) %in% cols_keep_red_S)]

res_red_S = cor(trim.env_red_S, method = "pearson", use = "complete.obs")
xx_red_S <- pheatmap( 
  res_red_S,
  color = col,
  breaks = seq(-1, 1, by = 0.2),
  clustering_method = "complete",  # Clustering method
  display_numbers = T,  # Display numbers in cells
  number_color = "grey35",
  show_rownames = T,
  legend = TRUE,  # Display legend
  legend_breaks = seq(-1, 1, by = 0.2)
)

save_pheatmap_pdf(xx_W1, "Chem_corr_Shallow_red_2024.05.22.pdf", width=12, height=10)

```

```{r intermediate}
# Intermediate Group
env_I = subset(env2, Group == "Intermediate")

cols_keep_I = c("DO_uM", "Salinity", "Temperature", "pH", "DOC", "Nitrite", "Nitrate", "Ammonium", "Phosphate", "DOC", "TN", "CobleA",  "CobleB",  "CobleC", "CobleM", "CobleT", "Fpeak",  "Stedmon_D", "Optical_Brighteners", "Lignin",  "BIX",  "HIX",  "FI", "M_to_C" )

trim.env_I = env_I[, (colnames(env_I) %in% cols_keep_I)]
res_I = cor(trim.env_I, method = "pearson", use = "complete.obs")
xx_I <- pheatmap( 
  res_I,
  color = col,
  breaks = seq(-1, 1, by = 0.2),
  clustering_method = "complete",  # Clustering method
  display_numbers = T,  # Display numbers in cells
  number_color = "grey35",
  show_rownames = T,
  legend = TRUE,  # Display legend
  legend_breaks = seq(-1, 1, by = 0.2)
)


```

```{r intermediate reduced}
# Intermediate Group - reduced
cols_keep_red_I = c("DO_uM", "Salinity", "Temperature", "pH", "DOC", "Nitrite", "Nitrate", "Ammonium", "Phosphate", "HIX", "Fpeak" )

trim.env_red_I = env_I[, (colnames(env_I) %in% cols_keep_red_I)]
res_red_I = cor(trim.env_red_I, method = "pearson", use = "complete.obs")
xx_red_I <- pheatmap( 
  res_red_I,
  color = col,
  breaks = seq(-1, 1, by = 0.2),
  clustering_method = "complete",  # Clustering method
  display_numbers = T,  # Display numbers in cells
  number_color = "grey35",
  show_rownames = T,
  legend = TRUE,  # Display legend
  legend_breaks = seq(-1, 1, by = 0.2)
)

```

```{r deep}
# Deep Group
env_D = subset(env2, Group == "Deep")

cols_keep_D = c("DO_uM", "Salinity", "Temperature", "pH", "DOC", "Nitrite", "Nitrate", "Ammonium", "Phosphate", "DOC", "TN", "CobleA",  "CobleB",  "CobleC", "CobleM", "CobleT", "Fpeak",  "Stedmon_D", "Optical_Brighteners", "Lignin",  "BIX",  "HIX",  "FI", "M_to_C" )

trim.env_D = env_D[, (colnames(env_D) %in% cols_keep_D)]
res_D = cor(trim.env_D, method = "pearson", use = "complete.obs")
xx_D <- pheatmap( 
  res_D,
  color = col,
  breaks = seq(-1, 1, by = 0.2),
  clustering_method = "complete",  # Clustering method
  display_numbers = T,  # Display numbers in cells
  number_color = "grey35",
  show_rownames = T,
  legend = TRUE,  # Display legend
  legend_breaks = seq(-1, 1, by = 0.2)
)

```

```{r deep reduced}
# Deep Group - reduced
cols_keep_red_D = c("DO_uM", "Salinity", "Temperature", "pH", "DOC", "Nitrite", "Nitrate", "Ammonium", "Phosphate", "HIX", "Fpeak" )

trim.env_red_D = env_D[, (colnames(env_D) %in% cols_keep_red_D)]
res_red_D = cor(trim.env_red_D, method = "pearson", use = "complete.obs")
xx_red_D <- pheatmap( 
  res_red_D,
  color = col,
  breaks = seq(-1, 1, by = 0.2),
  clustering_method = "complete",  # Clustering method
  display_numbers = T,  # Display numbers in cells
  number_color = "grey35",
  show_rownames = T,
  legend = TRUE,  # Display legend
  legend_breaks = seq(-1, 1, by = 0.2)
)

```

```{r Well 4}
# Well 4
env_W4 = subset(env2, Group == "W4")

cols_keep_W4 = c("DO_uM", "Salinity", "Temperature", "pH", "DOC", "Nitrite", "Nitrate", "Ammonium", "Phosphate", "DOC", "TN", "CobleA",  "CobleB",  "CobleC", "CobleM", "CobleT", "Fpeak",  "Stedmon_D", "Optical_Brighteners", "Lignin",  "BIX",  "HIX",  "FI", "M_to_C" )

trim.env_W4 = env_W4[, (colnames(env_W4) %in% cols_keep_W4)]
res_W4 = cor(trim.env_W4, method = "pearson", use = "complete.obs")
xx_W4 <- pheatmap( 
  res_W4,
  color = col,
  breaks = seq(-1, 1, by = 0.2),
  clustering_method = "complete",  # Clustering method
  display_numbers = T,  # Display numbers in cells
  number_color = "grey35",
  show_rownames = T,
  legend = TRUE,  # Display legend
  legend_breaks = seq(-1, 1, by = 0.2)
)

```

```{r Well 4 reduced}
# Well 4 - reduced
cols_keep_red_W4 = c("DO_uM", "Salinity", "Temperature", "pH", "DOC", "Nitrite", "Nitrate", "Ammonium", "Phosphate", "HIX", "Fpeak" )

trim.env_red_W4 = env_W4[, (colnames(env_W4) %in% cols_keep_red_W4)]
res_red_W4 = cor(trim.env_red_W4, method = "pearson", use = "complete.obs")
xx_red_W4 <- pheatmap( 
  res_red_W4,
  color = col,
  breaks = seq(-1, 1, by = 0.2),
  clustering_method = "complete",  # Clustering method
  display_numbers = T,  # Display numbers in cells
  number_color = "grey35",
  show_rownames = T,
  legend = TRUE,  # Display legend
  legend_breaks = seq(-1, 1, by = 0.2)
)

```


```{r Seawater}
# Seawater Group
env_SW = subset(env2, Group == "SW")

cols_keep_SW = c("DO_uM", "Salinity", "Temperature", "pH", "DOC", "Nitrite", "Nitrate", "Ammonium", "Phosphate", "DOC", "TN", "CobleA",  "CobleB",  "CobleC", "CobleM", "CobleT", "Fpeak",  "Stedmon_D", "Optical_Brighteners", "Lignin",  "BIX",  "HIX",  "FI", "M_to_C" )

trim.env_SW = env_SW[, (colnames(env_SW) %in% cols_keep_SW)]
res_SW = cor(trim.env_SW, method = "pearson", use = "complete.obs")
xx_SW <- pheatmap( 
  res_SW,
  color = col,
  breaks = seq(-1, 1, by = 0.2),
  clustering_method = "complete",  # Clustering method
  display_numbers = T,  # Display numbers in cells
  number_color = "grey35",
  show_rownames = T,
  legend = TRUE,  # Display legend
  legend_breaks = seq(-1, 1, by = 0.2)
)

```

```{r Seawater reduced}
# Seawater Group - reduced
cols_keep_red_SW = c("DO_uM", "Salinity", "Temperature", "pH", "DOC", "Nitrite", "Nitrate", "Ammonium", "Phosphate", "HIX", "Fpeak" )

trim.env_red_SW = env_SW[, (colnames(env_SW) %in% cols_keep_red_SW)]
res_red_SW = cor(trim.env_red_SW, method = "pearson", use = "complete.obs")
xx_red_SW <- pheatmap( 
  res_red_SW,
  color = col,
  breaks = seq(-1, 1, by = 0.2),
  clustering_method = "complete",  # Clustering method
  display_numbers = T,  # Display numbers in cells
  number_color = "grey35",
  show_rownames = T,
  legend = TRUE,  # Display legend
  legend_breaks = seq(-1, 1, by = 0.2)
)

```


# Part 2: Constrained ordination of microbial data with environmental variables

## 1. Transform read counts
```{r}
pstrans = transform_sample_counts(ps, function(x) asinh(x)) 
```

## 2. Remove samples with incomplete environmental data
```{r}
pstrans_noNA <- pstrans %>%
  subset_samples(
    !is.na(Salinity) &
    !is.na(Temperature) &
    !is.na(DO) &
    !is.na(pH) &
    !is.na(Nitrate) &
    !is.na(Nitrite) &
    !is.na(Ammonium) &
    !is.na(Phosphate) &
    !is.na(DOC) &
    !is.na(HIX)
  )

n_distinct(sample_names(pstrans)) # 204
n_distinct(sample_names(pstrans_noNA)) #183

# NOTE: removed all sand samples and 4 porewater samples: 030322-L-W4D4 (no OC), 030322-L-W4D2 (no OC), 030322-L-W1D3 (no nutrients), 030322-L-W1D1 (no ammonia)

```

```{r remove seawater depth}
# remove seawater "depth" because it's arbitrary
sample_data(pstrans_noNA)$Elevation[sample_data(pstrans_noNA)$Type == "Seawater"] = NA 

```


## 3. Perform Constrained (canonical) Analysis of Principal Coordinates (CAP)
```{r}
# CAP ~ uses vegan's capscale function
# (partial) constrained analysis of principal coordinates or distance-based redundancy analysis
set.seed(123)
cap_litdir = ordinate(pstrans_noNA, formula= ~ Salinity + Temperature + DO + pH + Nitrate + Nitrite + Ammonium + Phosphate + DOC + HIX + Fpeak, "CAP") # default is distance = "bray
evals_bc = cap_litdir$CCA$eig

# scree plot
plot_ordination(pstrans_noNA, cap_litdir, type = "scree")  # three primary axes

# ellipse
#dataEllipse in car package

```


## 4. Visualize CAP results with environmental arrows
```{r}
# ordination with axes 1 and 2
p0 = plot_ordination(pstrans_noNA, cap_litdir, color = "Well", shape = "Campaign", axes = c(1,2)) +
  coord_fixed(sqrt(evals_bc[2] / evals_bc[1])) + # set aspect ratio to CAP eigenvalues
  theme_bw()+
  theme(legend.text = element_text(size = 11), axis.text = element_text(size = 11)) +
  theme(panel.border = element_rect(colour = "black", size = 1))+
  geom_point(alpha = 0.7, size = 3) +
  scale_color_manual(values = rev(viridis(5)))+
  labs(col = "", shape = "")
p0

ggsave(filename = "CAP_1_2_no_arrows_2024.06.19.png",
       plot = last_plot(),
       device = "png", units = "cm", width = 15, height = 10, dpi = 300) # adjust dimensions to axes %

# Now add the environmental variables as arrows
arrowmat = vegan::scores(cap_litdir, display = "bp") # Extract arrow coordinates
arrowdf <- data.frame(labels = rownames(arrowmat), arrowmat) 
arrow_map = aes(xend = CAP1*1.2, yend = CAP2*1.2, x = 0, y = 0, shape = NULL, color = NULL) 
label_map = aes(x = 1.2 * CAP1, y = 1.2 * CAP2, shape = NULL, color = NULL, label = labels) 
arrowhead = arrow(length = unit(0.02, "npc")) # length of arrow head
p1 = p0 + geom_segment(arrow_map, linewidth = 0.5, data = arrowdf, color = "black", 
    arrow = arrowhead) +
geom_text(label_map, size = 4, data = arrowdf) # comment out to remove labels
p1

ggsave(filename = "CAP_1_2_2024.03.12.png",
       plot = last_plot(),
       device = "png", units = "cm", width = 15, height = 10, dpi = 300) # adjust dimensions to axes %

p1 = p0 + geom_segment(arrow_map, linewidth = 0.7, data = arrowdf, color = "black", 
    arrow = arrowhead)
p1

ggsave(filename = "CAP_1_2_noLabels_2024.06.19.png",
       plot = last_plot(),
       device = "png", units = "cm", width = 15, height = 10, dpi = 300) # adjust dimensions to axes %

# ordination with axes 2 and 3
p0 = plot_ordination(pstrans_noNA, cap_litdir, color = "Elevation", shape = "Campaign", axes = c(3,2)) +
  coord_fixed(sqrt(evals_bc[2] / evals_bc[3])) + # set aspect ratio to CAP eigenvalues
  theme_bw() +
  theme(legend.text = element_text(size = 11), axis.text = element_text(size = 11)) +
  theme(panel.border = element_rect(colour = "black", size = 1))+
  geom_point(alpha = 0.7, size = 3) +
  scale_color_viridis()+
  labs(col = "Depth (m to MSL)", shape = "")
p0

ggsave(filename = "CAP_3_2_no_arrows_2024.06.19.png",
       plot = last_plot(),
       device = "png", units = "cm", width = 15, height = 10, dpi = 300) # adjust dimensions to axes %


# Now add the environmental variables as arrows
arrowmat = vegan::scores(cap_litdir, choices=c(2,3), display = "bp") # Extract arrow coordinates
arrowdf <- data.frame(labels = rownames(arrowmat), arrowmat) 
arrow_map = aes(xend = CAP3*1.5, yend = CAP2*1.5, x = 0, y = 0, shape = NULL, color = NULL) 
label_map = aes(x = 1.2 * CAP3, y = 1.2 * CAP2, shape = NULL, color = NULL, label = labels) 
arrowhead = arrow(length = unit(0.02, "npc")) # length of arrow head
p1 = p0 + geom_segment(arrow_map, linewidth = 0.5, data = arrowdf, color = "black", 
    arrow = arrowhead) +
geom_text(label_map, size = 4, data = arrowdf) # comment out to remove labels
p1

ggsave(filename = "CAP_3_2_2024.03.23.png",
       plot = last_plot(),
       device = "png", units = "cm", width = 15, height = 10, dpi = 300)

p1 = p0 + geom_segment(arrow_map, linewidth = 0.7, data = arrowdf, color = "black", 
    arrow = arrowhead)
p1

ggsave(filename = "CAP_3_2_noLabels_2024.06.19.png",
       plot = last_plot(),
       device = "png", units = "cm", width = 15, height = 10, dpi = 300)

```

## 5. Repeat for each well individually
```{r well subsets}
# subset data by well
ps_W1 = subset_samples(pstrans_noNA, Well == "Well 1")
ps_W2 = subset_samples(pstrans_noNA, Well == "Well 2")
ps_W3 = subset_samples(pstrans_noNA, Well == "Well 3")
ps_W4 = subset_samples(pstrans_noNA, Well == "Well 4")

# CAP
cap_W1 = ordinate(ps_W1, formula= ~ Salinity + Temperature + DO + pH + Nitrate + Nitrite + Ammonium + Phosphate + DOC + HIX + Fpeak, "CAP")
cap_W2 = ordinate(ps_W2, formula= ~ Salinity + Temperature + DO + pH + Nitrate + Nitrite + Ammonium + Phosphate + DOC + HIX + Fpeak, "CAP")
cap_W3 = ordinate(ps_W3, formula= ~ Salinity + Temperature + DO + pH + Nitrate + Nitrite + Ammonium + Phosphate + DOC + HIX + Fpeak, "CAP")
cap_W4 = ordinate(ps_W4, formula= ~ Salinity + Temperature + DO + pH + Nitrate + Nitrite + Ammonium + Phosphate + DOC + HIX + Fpeak, "CAP")

# eigenvalues
evals_bc_W1 = cap_W1$CCA$eig
evals_bc_W2 = cap_W2$CCA$eig
evals_bc_W3 = cap_W3$CCA$eig
evals_bc_W4 = cap_W4$CCA$eig

# scree plots
plot_ordination(ps_W1, cap_W1, type = "scree")  # two primary axes
plot_ordination(ps_W2, cap_W2, type = "scree")  # two primary axes
plot_ordination(ps_W3, cap_W3, type = "scree")  # two primary axes
plot_ordination(ps_W4, cap_W4, type = "scree")  # two primary axes

```

```{r well 1}
# Well 1
pW1 = plot_ordination(ps_W1, cap_W1, color = "Elevation", shape = "Campaign", axes = c(1,2)) +
  coord_fixed(sqrt(evals_bc_W1[2] / evals_bc_W1[1])) + # set aspect ratio to CAP eigenvalues
  theme_bw() +
  geom_point(alpha = 0.7, size = 3) +
  scale_color_viridis()+
  labs(col = "Depth (m to MSL)", shape = "")
pW1

arrowmat = vegan::scores(cap_W1, display = "bp") # Extract arrow coordinates
arrowdf <- data.frame(labels = rownames(arrowmat), arrowmat) 
arrow_map = aes(xend = CAP1*1, yend = CAP2*1, x = 0, y = 0, shape = NULL, color = NULL) 
label_map = aes(x = 1.2 * CAP1, y = 1.2 * CAP2, shape = NULL, color = NULL, label = labels) 
arrowhead = arrow(length = unit(0.02, "npc")) # length of arrow head
pW1_env = pW1 + geom_segment(arrow_map, linewidth = 0.5, data = arrowdf, color = "black", 
    arrow = arrowhead) +
geom_text(label_map, size = 4, data = arrowdf) # comment out to remove labels
pW1_env

ggsave(filename = "CAP1_2_Well1_2024.03.23.png",
       plot = last_plot(),
       device = "png", units = "cm", width = 15, height = 10, dpi = 300) 

pW1_env = pW1 + geom_segment(arrow_map, linewidth = 0.5, data = arrowdf, color = "black", 
    arrow = arrowhead) 
pW1_env

ggsave(filename = "CAP1_2_W1_noLabels_2024.03.23.png",
       plot = last_plot(),
       device = "png", units = "cm", width = 15, height = 10, dpi = 300) 

```

```{r well 2}
# Well 2
pW2 = plot_ordination(ps_W2, cap_W2, color = "Elevation", shape = "Campaign", axes = c(1,2)) +
  coord_fixed(sqrt(evals_bc_W2[2] / evals_bc_W2[1])) + # set aspect ratio to CAP eigenvalues
  theme_bw() +
  geom_point(alpha = 0.7, size = 3) +
  scale_color_viridis()+
  labs(col = "Depth (m to MSL)", shape = "") 
pW2

arrowmat = vegan::scores(cap_W2, display = "bp") # Extract arrow coordinates
arrowdf <- data.frame(labels = rownames(arrowmat), arrowmat)
arrow_map = aes(xend = CAP1*1, yend = CAP2*1, x = 0, y = 0, shape = NULL, color = NULL) 
label_map = aes(x = 1.2 * CAP1, y = 1.2 * CAP2, shape = NULL, color = NULL, label = labels) 
arrowhead = arrow(length = unit(0.02, "npc")) # length of arrow head

pW2_env = pW2 + geom_segment(arrow_map, linewidth = 0.5, data = arrowdf, color = "black", 
    arrow = arrowhead) +
geom_text(label_map, size = 4, data = arrowdf) # comment out to remove labels
pW2_env

ggsave(filename = "CAP1_2_Well2_2024.03.23.png",
       plot = last_plot(),
       device = "png", units = "cm", width = 15, height = 10, dpi = 300) 

pW2_env = pW2 + geom_segment(arrow_map, linewidth = 0.5, data = arrowdf, color = "black", 
    arrow = arrowhead) 
pW2_env

ggsave(filename = "CAP1_2_Well2_noLabels_2024.03.23.png",
       plot = last_plot(),
       device = "png", units = "cm", width = 15, height = 10, dpi = 300) 


```


```{r well 3}
# Well 3
pW3 = plot_ordination(ps_W3, cap_W3, color = "Elevation", shape = "Campaign", axes = c(1,2)) +
  coord_fixed(sqrt(evals_bc_W3[2] / evals_bc_W3[1])) + # set aspect ratio to CAP eigenvalues
  theme_bw() +
  geom_point(alpha = 0.7, size = 3) +
  scale_color_viridis()+
  labs(col = "Depth (m to MSL)", shape = "")
pW3

arrowmat = vegan::scores(cap_W3, display = "bp") # Extract arrow coordinates
arrowdf <- data.frame(labels = rownames(arrowmat), arrowmat)
arrow_map = aes(xend = CAP1*1, yend = CAP2*1, x = 0, y = 0, shape = NULL, color = NULL)
label_map = aes(x = 1.2 * CAP1, y = 1.2 * CAP2, shape = NULL, color = NULL, label = labels)
arrowhead = arrow(length = unit(0.02, "npc")) # length of arrow head

pW3_env = pW3 + geom_segment(arrow_map, linewidth = 0.5, data = arrowdf, color = "black", 
    arrow = arrowhead) +
geom_text(label_map, size = 4, data = arrowdf) # comment out to remove labels
pW3_env

ggsave(filename = "CAP1_2_Well3_2024.03.23.png",
       plot = last_plot(),
       device = "png", units = "cm", width = 15, height = 10, dpi = 300)

pW3_env = pW3 + geom_segment(arrow_map, linewidth = 0.5, data = arrowdf, color = "black", 
    arrow = arrowhead) 
pW3_env

ggsave(filename = "CAP1_2_Well3_noLabels_2024.03.23.png",
       plot = last_plot(),
       device = "png", units = "cm", width = 15, height = 10, dpi = 300)


```


```{r well 4}
# Well 4
pW4 = plot_ordination(ps_W4, cap_W4, color = "Elevation", shape = "Campaign", axes = c(1,2)) +
  coord_fixed(sqrt(evals_bc_W4[2] / evals_bc_W4[1])) + # set aspect ratio to CAP eigenvalues
  theme_bw() +
  geom_point(alpha = 0.7, size = 3) +
  scale_color_viridis()+
  labs(col = "Depth (m to MSL)", shape = "")
pW4

pW4_date = plot_ordination(ps_W4, cap_W4, color = "Date_Tide", shape = "Campaign", axes = c(1,2)) +
  coord_fixed(sqrt(evals_bc_W4[2] / evals_bc_W4[1])) + # set aspect ratio to CAP eigenvalues
  theme_bw() +
  geom_point(alpha = 0.7, size = 3) +
  scale_color_manual(values = viridis(13))+
  labs(col = "Date", shape = "")
pW4_date
  
arrowmat = vegan::scores(cap_W4, display = "bp") # Extract arrow coordinates
arrowdf <- data.frame(labels = rownames(arrowmat), arrowmat)
arrow_map = aes(xend = CAP1*1, yend = CAP2*1, x = 0, y = 0, shape = NULL, color = NULL)
label_map = aes(x = 1.2 * CAP1, y = 1.2 * CAP2, shape = NULL, color = NULL, label = labels)
arrowhead = arrow(length = unit(0.02, "npc")) # length of arrow head

pW4_env = pW4 + geom_segment(arrow_map, linewidth = 0.5, data = arrowdf, color = "black", 
    arrow = arrowhead) +
geom_text(label_map, size = 4, data = arrowdf) # comment out to remove labels
pW4_env

ggsave(filename = "CAP1_2_Well4_2024.03.23.png",
       plot = last_plot(),
       device = "png", units = "cm", width = 15, height = 10, dpi = 300)

pW4_env = pW4 + geom_segment(arrow_map, linewidth = 0.5, data = arrowdf, color = "black", 
    arrow = arrowhead) 
pW4_env

ggsave(filename = "CAP1_2_Well4_noLabels_2024.03.23.png",
       plot = last_plot(),
       device = "png", units = "cm", width = 15, height = 10, dpi = 300)

```



## 6. Repeat for each cluster group individually
```{r group subsets}
# subset data by group
ps_shallow = subset_samples(pstrans_noNA, Group == "Shallow")
ps_intermediate = subset_samples(pstrans_noNA, Group == "Intermediate")
ps_deep = subset_samples(pstrans_noNA, Group == "Deep")
ps_W4 = subset_samples(pstrans_noNA, Group == "W4")
ps_SW = subset_samples(pstrans_noNA, Group == "SW")

# check data
meta_shallow = sample_data(ps_shallow)
meta_intermediate = sample_data(ps_intermediate)
meta_deep = sample_data(ps_deep)
meta_W4 = sample_data(ps_W4)
meta_SW = sample_data(ps_SW)

# CAP
set.seed(123)
cap_shallow = ordinate(ps_shallow, formula= ~ Salinity + Temperature + DO + pH + Nitrate + Nitrite + Ammonium + Phosphate + DOC + HIX + Fpeak, "CAP")
cap_intermediate = ordinate(ps_intermediate, formula= ~ Salinity + Temperature + DO + pH + Nitrate + Nitrite + Ammonium + Phosphate + DOC + HIX + Fpeak, "CAP")
cap_deep = ordinate(ps_deep, formula= ~ Salinity + Temperature + DO + pH + Nitrate + Nitrite + Ammonium + Phosphate + DOC + HIX + Fpeak, "CAP")
cap_W4 = ordinate(ps_W4, formula= ~ Salinity + Temperature + DO + pH + Nitrate + Nitrite + Ammonium + Phosphate + DOC + HIX + Fpeak, "CAP")
cap_SW = ordinate(ps_SW, formula= ~ Salinity + Temperature + DO + pH + Nitrate + Nitrite + Ammonium + Phosphate + DOC + HIX + Fpeak, "CAP")

# eigenvalues
evals_bc_shallow = cap_shallow$CCA$eig
evals_bc_intermediate = cap_intermediate$CCA$eig
evals_bc_deep = cap_deep$CCA$eig
evals_bc_W4 = cap_W4$CCA$eig
evals_bc_SW = cap_SW$CCA$eig

# scree plots
plot_ordination(ps_shallow, cap_shallow, type = "scree")  # two primary axes
plot_ordination(ps_intermediate, cap_intermediate, type = "scree")  # three primary axes
plot_ordination(ps_deep, cap_deep, type = "scree")  # two primary axes
plot_ordination(ps_W4, cap_W4, type = "scree")  # two primary axes
plot_ordination(ps_SW, cap_SW, type = "scree")  # one primary axis

```

```{r shallow}
# Shallow
pshallow = plot_ordination(ps_shallow, cap_shallow, color = "Well", shape = "Campaign", axes = c(1,2)) +
  coord_fixed(sqrt(evals_bc_shallow[2] / evals_bc_shallow[1])) + # set aspect ratio to CAP eigenvalues
  theme_bw() +
  geom_point(alpha = 0.7, size = 3) +
  labs(col = "", shape = "")
pshallow

arrowmat = vegan::scores(cap_shallow, display = "bp") # Extract arrow coordinates
arrowdf <- data.frame(labels = rownames(arrowmat), arrowmat)
arrow_map = aes(xend = CAP1*1, yend = CAP2*1, x = 0, y = 0, shape = NULL, color = NULL)
label_map = aes(x = 1.2 * CAP1, y = 1.2 * CAP2, shape = NULL, color = NULL, label = labels)
arrowhead = arrow(length = unit(0.02, "npc")) # length of arrow head

pshallow_env = pshallow + geom_segment(arrow_map, linewidth = 0.5, data = arrowdf, color = "black", 
    arrow = arrowhead) +
geom_text(label_map, size = 4, data = arrowdf) # comment out to remove labels
pshallow_env

ggsave(filename = "CAP1_2_Shallow_2024.05.22.png",
       plot = last_plot(),
       device = "png", units = "cm", width = 15, height = 10, dpi = 300)

pshallow_env = pshallow + geom_segment(arrow_map, linewidth = 0.5, data = arrowdf, color = "black", 
    arrow = arrowhead) 
pshallow_env

ggsave(filename = "CAP1_2_Shallow_noLabels_2024.05.22.png",
       plot = last_plot(),
       device = "png", units = "cm", width = 15, height = 10, dpi = 300)


```

```{r intermediate}
# Intermediate
pintermediate = plot_ordination(ps_intermediate, cap_intermediate, color = "Well", shape = "Campaign", axes = c(1,2)) +
  coord_fixed(sqrt(evals_bc_intermediate[2] / evals_bc_intermediate[1])) + # set aspect ratio to CAP eigenvalues
  theme_bw() +
  geom_point(alpha = 0.7, size = 3) +
  labs(col = "", shape = "")
pintermediate

arrowmat = vegan::scores(cap_intermediate, display = "bp") # Extract arrow coordinates
arrowdf <- data.frame(labels = rownames(arrowmat), arrowmat)
arrow_map = aes(xend = CAP1*1, yend = CAP2*1, x = 0, y = 0, shape = NULL, color = NULL)
label_map = aes(x = 1.2 * CAP1, y = 1.2 * CAP2, shape = NULL, color = NULL, label = labels)
arrowhead = arrow(length = unit(0.02, "npc")) # length of arrow head

pintermediate_env = pintermediate + geom_segment(arrow_map, linewidth = 0.5, data = arrowdf, color = "black", 
    arrow = arrowhead) +
geom_text(label_map, size = 4, data = arrowdf) # comment out to remove labels
pintermediate_env

ggsave(filename = "CAP1_2_Intermediate_2024.05.22.png",
       plot = last_plot(),
       device = "png", units = "cm", width = 15, height = 10, dpi = 300)

pintermediate_env = pintermediate + geom_segment(arrow_map, linewidth = 0.5, data = arrowdf, color = "black",
    arrow = arrowhead)
pintermediate_env

ggsave(filename = "CAP1_2_Intermediate_noLabels_2024.05.22.png",
       plot = last_plot(),
       device = "png", units = "cm", width = 15, height = 10, dpi = 300)


```

```{r deep}
# Deep
pdeep = plot_ordination(ps_deep, cap_deep, color = "Well", shape = "Campaign", axes = c(1,2)) +
  coord_fixed(sqrt(evals_bc_deep[2] / evals_bc_deep[1])) + # set aspect ratio to CAP eigenvalues
  theme_bw() +
  geom_point(alpha = 0.7, size = 3) +
  labs(col = "", shape = "")
pdeep

arrowmat = vegan::scores(cap_deep, display = "bp") # Extract arrow coordinates
arrowdf <- data.frame(labels = rownames(arrowmat), arrowmat)
arrow_map = aes(xend = CAP1*1, yend = CAP2*1, x = 0, y = 0, shape = NULL, color = NULL)
label_map = aes(x = 1.2 * CAP1, y = 1.2 * CAP2, shape = NULL, color = NULL, label = labels)
arrowhead = arrow(length = unit(0.02, "npc")) # length of arrow head

pdeep_env = pdeep + geom_segment(arrow_map, linewidth = 0.5, data = arrowdf, color = "black",
    arrow = arrowhead) +
geom_text(label_map, size = 4, data = arrowdf) # comment out to remove labels
pdeep_env

ggsave(filename = "CAP1_2_Deep_2024.05.22.png",
       plot = last_plot(),
       device = "png", units = "cm", width = 15, height = 10, dpi = 300)

pdeep_env = pdeep + geom_segment(arrow_map, linewidth = 0.5, data = arrowdf, color = "black",
    arrow = arrowhead)
pdeep_env

ggsave(filename = "CAP1_2_Deep_noLabels_2024.05.22.png",
       plot = last_plot(),
       device = "png", units = "cm", width = 15, height = 10, dpi = 300)

```

```{r Well 4}
# Well 4
pW4 = plot_ordination(ps_W4, cap_W4, color = "Elevation", shape = "Campaign", axes = c(1,2)) +
  coord_fixed(sqrt(evals_bc_W4[2] / evals_bc_W4[1])) + # set aspect ratio to CAP eigenvalues
  theme_bw() +
  geom_point(alpha = 0.7, size = 3) +
  scale_color_viridis()+
  labs(col = "Depth (m to MSL)", shape = "")
pW4

arrowmat = vegan::scores(cap_W4, display = "bp") # Extract arrow coordinates
arrowdf <- data.frame(labels = rownames(arrowmat), arrowmat)
arrow_map = aes(xend = CAP1*1, yend = CAP2*1, x = 0, y = 0, shape = NULL, color = NULL)
label_map = aes(x = 1.2 * CAP1, y = 1.2 * CAP2, shape = NULL, color = NULL, label = labels)
arrowhead = arrow(length = unit(0.02, "npc")) # length of arrow head

pW4_env = pW4 + geom_segment(arrow_map, linewidth = 0.5, data = arrowdf, color = "black",
    arrow = arrowhead) +
    
geom_text(label_map, size = 4, data = arrowdf) # comment out to remove labels
pW4_env

ggsave(filename = "CAP1_2_Well4_2024.05.22.png",
       plot = last_plot(),
       device = "png", units = "cm", width = 15, height = 10, dpi = 300)

pW4_env = pW4 + geom_segment(arrow_map, linewidth = 0.5, data = arrowdf, color = "black",
    arrow = arrowhead)
pW4_env

ggsave(filename = "CAP1_2_Well4_noLabels_2024.05.22.png",
       plot = last_plot(),
       device = "png", units = "cm", width = 15, height = 10, dpi = 300)

```

```{r SW (remove)}
# Seawater # not enough samples
pSW = plot_ordination(ps_SW, cap_SW, color = "Tide", shape = "Campaign", axes = c(1,2)) +
  coord_fixed(sqrt(evals_bc_SW[2] / evals_bc_SW[1])) + # set aspect ratio to CAP eigenvalues
  theme_bw() +
  geom_point(alpha = 0.7, size = 3) +
  scale_color_manual(values = viridis(3))+
  labs(col = "Tide", shape = "")
pSW

arrowmat = vegan::scores(cap_SW, display = "bp") # Extract arrow coordinates
arrowdf <- data.frame(labels = rownames(arrowmat), arrowmat)
arrow_map = aes(xend = CAP1*1, yend = CAP2*1, x = 0, y = 0, shape = NULL, color = NULL)
label_map = aes(x = 1.2 * CAP1, y = 1.2 * CAP2, shape = NULL, color = NULL, label = labels)
arrowhead = arrow(length = unit(0.02, "npc")) # length of arrow head

pSW_env = pSW + geom_segment(arrow_map, linewidth = 0.5, data = arrowdf, color = "black",
    arrow = arrowhead) +
geom_text(label_map, size = 4, data = arrowdf) # comment out to remove labels
pSW_env

ggsave(filename = "CAP1_2_SW_2024.05.22.png",
       plot = last_plot(),
       device = "png", units = "cm", width = 15, height = 10, dpi = 300)

pSW_env = pSW + geom_segment(arrow_map, linewidth = 0.5, data = arrowdf, color = "black",
    arrow = arrowhead)
pSW_env

ggsave(filename = "CAP1_2_SW_noLabels_2024.05.22.png",
       plot = last_plot(),
       device = "png", units = "cm", width = 15, height = 10, dpi = 300)

```


# Part 3: Community-level statistical analyses

## 1. Compute Bray-Curtis distance matrix
```{r distance matrix}
# porewater only
ps_noSW = subset_samples(pstrans_noNA, Well != "Seawater")
meta_ps_noSW = sample_data(ps_noSW)

# february
ps_noSW_Feb = subset_samples(ps_noSW, Campaign == "February 2022")
meta_ps_noSW_Feb = sample_data(ps_noSW_Feb)

# october
ps_noSW_Oct = subset_samples(ps_noSW, Campaign == "October 2022")
meta_ps_noSW_Oct = sample_data(ps_noSW_Oct)


dist_all = phyloseq::distance(pstrans_noNA, method = "bray") # default type = "samples"
dist_noSW = phyloseq::distance(ps_noSW, method = "bray")
dist_feb = phyloseq::distance(ps_noSW_Feb, method = "bray")
dist_oct = phyloseq::distance(ps_noSW_Oct, method = "bray")


# wells
dist_W1 = phyloseq::distance(ps_W1, method = "bray") 
dist_W2 = phyloseq::distance(ps_W2, method = "bray") 
dist_W3 = phyloseq::distance(ps_W3, method = "bray")
dist_W4 = phyloseq::distance(ps_W4, method = "bray")

# region
dist_shallow = phyloseq::distance(ps_shallow, method = "bray")
dist_intermediate = phyloseq::distance(ps_intermediate, method = "bray")
dist_deep = phyloseq::distance(ps_deep, method = "bray")
dist_W4 = phyloseq::distance(ps_W4, method = "bray")
dist_SW = phyloseq::distance(ps_SW, method = "bray")

# region and season
ps_shallow_feb = subset_samples(ps_shallow, Campaign == "February 2022")
ps_intermediate_feb = subset_samples(ps_intermediate, Campaign == "February 2022")
ps_intermediate_oct = subset_samples(ps_intermediate, Campaign == "October 2022")
ps_deep_feb = subset_samples(ps_deep, Campaign == "February 2022")
ps_deep_oct = subset_samples(ps_deep, Campaign == "October 2022")
ps_W4_feb = subset_samples(ps_W4, Campaign == "February 2022")
ps_W4_oct = subset_samples(ps_W4, Campaign == "October 2022")

meta_ps_shallow_feb = sample_data(ps_shallow_feb)
meta_ps_intermediate_feb = sample_data(ps_intermediate_feb)
meta_ps_intermediate_oct = sample_data(ps_intermediate_oct)
meta_ps_deep_feb = sample_data(ps_deep_feb)
meta_ps_deep_oct = sample_data(ps_deep_oct)
meta_ps_W4_feb = sample_data(ps_W4_feb)
meta_ps_W4_oct = sample_data(ps_W4_oct)

dist_shallow_feb = phyloseq::distance(ps_shallow_feb, method = "bray")
dist_intermediate_feb = phyloseq::distance(ps_intermediate_feb, method = "bray")
dist_intermediate_oct = phyloseq::distance(ps_intermediate_oct, method = "bray")
dist_deep_feb = phyloseq::distance(ps_deep_feb, method = "bray")
dist_deep_oct = phyloseq::distance(ps_deep_oct, method = "bray")
dist_W4_feb = phyloseq::distance(ps_W4_feb, method = "bray")
dist_W4_oct = phyloseq::distance(ps_W4_oct, method = "bray")

# region, season, tide regime
ps_shallow_feb_neap = subset_samples(ps_shallow_feb, Tide.Regime == "Neap")
ps_shallow_feb_spring = subset_samples(ps_shallow_feb, Tide.Regime == "Spring")
ps_intermediate_feb_neap = subset_samples(ps_intermediate_feb, Tide.Regime == "Neap")
ps_intermediate_feb_spring = subset_samples(ps_intermediate_feb, Tide.Regime == "Spring")
ps_deep_feb_neap = subset_samples(ps_deep_feb, Tide.Regime == "Neap")
ps_deep_feb_spring = subset_samples(ps_deep_feb, Tide.Regime == "Spring")
ps_W4_feb_neap = subset_samples(ps_W4_feb, Tide.Regime == "Neap")
ps_W4_feb_spring = subset_samples(ps_W4_feb, Tide.Regime == "Spring")

meta_ps_shallow_feb_neap = sample_data(ps_shallow_feb_neap)
meta_ps_shallow_feb_spring = sample_data(ps_shallow_feb_spring)
meta_ps_intermediate_feb_neap = sample_data(ps_intermediate_feb_neap)
meta_ps_intermediate_feb_spring = sample_data(ps_intermediate_feb_spring)
meta_ps_deep_feb_neap = sample_data(ps_deep_feb_neap)
meta_ps_deep_feb_spring = sample_data(ps_deep_feb_spring)
meta_ps_W4_feb_neap = sample_data(ps_W4_feb_neap)
meta_ps_W4_feb_spring = sample_data(ps_W4_feb_spring)


dist_shallow_feb_neap = phyloseq::distance(ps_shallow_feb_neap, method = "bray")
dist_shallow_feb_spring = phyloseq::distance(ps_shallow_feb_spring, method = "bray")
dist_intermediate_feb_neap = phyloseq::distance(ps_intermediate_feb_neap, method = "bray")
dist_intermediate_feb_spring = phyloseq::distance(ps_intermediate_feb_spring, method = "bray")
dist_deep_feb_neap = phyloseq::distance(ps_deep_feb_neap, method = "bray")
dist_deep_feb_spring = phyloseq::distance(ps_deep_feb_spring, method = "bray")
dist_W4_feb_neap = phyloseq::distance(ps_W4_feb_neap, method = "bray")
dist_W4_feb_spring = phyloseq::distance(ps_W4_feb_spring, method = "bray")

```

## 2. Homogeneity of dispersion
```{r test season/campaign}
# test dispersion using vegan betadisper
disp_all = betadisper(dist_all, sample_data(pstrans_noNA)$Campaign) # campaign is season
disp_noSW = betadisper(dist_noSW, sample_data(ps_noSW)$Campaign)
disp_W1 = betadisper(dist_W1, sample_data(ps_W1)$Campaign) # can be a factor with one level
disp_W2 = betadisper(dist_W2, sample_data(ps_W2)$Campaign)
disp_W3 = betadisper(dist_W3, sample_data(ps_W3)$Campaign)
disp_W4 = betadisper(dist_W4, sample_data(ps_W4)$Campaign)

# region by season
#disp_shallow_season = betadisper(dist_shallow, sample_data(ps_shallow)$Campaign)
disp_intermediate_season = betadisper(dist_intermediate, sample_data(ps_intermediate)$Campaign)
disp_deep_season = betadisper(dist_deep, sample_data(ps_deep)$Campaign)
disp_W4_season = betadisper(dist_W4, sample_data(ps_W4)$Campaign)
disp_SW_season = betadisper(dist_SW, sample_data(ps_SW)$Campaign)


# hull ordiplot
plot(disp_all)
plot(disp_noSW)
plot(disp_W1,main = "Well 1")
plot(disp_W2,main = "Well 2")
plot(disp_W3,main = "Well 3")
plot(disp_W4,main = "Well 4")

#plot(disp_shallow_season, main = "Shallow")
plot(disp_intermediate_season, main = "Intermediate")
plot(disp_deep_season, main = "Deep")
plot(disp_W4_season, main = "Well 4")
plot(disp_SW_season, main = "Seawater")

# boxplot
boxplot(disp_all, ylab = "Distance to centroid", xlab = "", main = "All samples")
boxplot(disp_noSW, ylab = "Distance to centroid", xlab = "", main = "All samples (no SW)")
boxplot(disp_W1, ylab = "Distance to centroid", xlab = "", main = "Well 1")
boxplot(disp_W2, ylab = "Distance to centroid", xlab = "", main = "Well 2")
boxplot(disp_W3, ylab = "Distance to centroid", xlab = "", main = "Well 3")
boxplot(disp_W4, ylab = "Distance to centroid", xlab = "", main = "Well 4")

#boxplot(disp_shallow_season, ylab = "Distance to centroid", xlab = "", main = "Shallow")
boxplot(disp_intermediate_season, ylab = "Distance to centroid", xlab = "", main = "Intermediate")
boxplot(disp_deep_season, ylab = "Distance to centroid", xlab = "", main = "Deep")
boxplot(disp_W4_season, ylab = "Distance to centroid", xlab = "", main = "Well 4")
boxplot(disp_SW_season, ylab = "Distance to centroid", xlab = "", main = "Seawater")

# summary
print(disp_all, digits = max(3, getOption("digits") - 3), neigen = 8)
print(disp_noSW, digits = max(3, getOption("digits") - 3), neigen = 8)
print(disp_W1, digits = max(3, getOption("digits") - 3), neigen = 8)
print(disp_W2, digits = max(3, getOption("digits") - 3), neigen = 8)
print(disp_W3, digits = max(3, getOption("digits") - 3), neigen = 8)
print(disp_W4, digits = max(3, getOption("digits") - 3), neigen = 8)

#print(disp_shallow_season, digits = max(3, getOption("digits") - 3), neigen = 8)
print(disp_intermediate_season, digits = max(3, getOption("digits") - 3), neigen = 8)
print(disp_deep_season, digits = max(3, getOption("digits") - 3), neigen = 8)
print(disp_W4_season, digits = max(3, getOption("digits") - 3), neigen = 8)
print(disp_SW_season, digits = max(3, getOption("digits") - 3), neigen = 8)


# Tukey's HSD
TukeyHSD(disp_all, which = "group", ordered = FALSE, conf.level = 0.95)
TukeyHSD(disp_noSW, which = "group", ordered = FALSE, conf.level = 0.95)
TukeyHSD(disp_W1, which = "group", ordered = FALSE, conf.level = 0.95)
TukeyHSD(disp_W2, which = "group", ordered = FALSE, conf.level = 0.95)
TukeyHSD(disp_W3, which = "group", ordered = FALSE, conf.level = 0.95)
TukeyHSD(disp_W4, which = "group", ordered = FALSE, conf.level = 0.95)

#TukeyHSD(disp_shallow_season, which = "group", ordered = FALSE, conf.level = 0.95)
TukeyHSD(disp_intermediate_season, which = "group", ordered = FALSE, conf.level = 0.95)
TukeyHSD(disp_deep_season, which = "group", ordered = FALSE, conf.level = 0.95)
TukeyHSD(disp_W4_season, which = "group", ordered = FALSE, conf.level = 0.95)
TukeyHSD(disp_SW_season, which = "group", ordered = FALSE, conf.level = 0.95)

# permutation-based test for homogeneity of dispersion 
permutest(disp_all)
permutest(disp_noSW)
permutest(disp_W1)
permutest(disp_W2)
permutest(disp_W3)
permutest(disp_W4)

#permutest(disp_shallow_season)
permutest(disp_intermediate_season) # df is num samples minus num groups in test
permutest(disp_deep_season)
permutest(disp_W4_season)
permutest(disp_SW_season)

# ANOVA
anova(disp_all)
anova(disp_noSW)
anova(disp_W1)
anova(disp_W2)
anova(disp_W3)
anova(disp_W4)

#anova(disp_shallow_season)
anova(disp_intermediate_season)
anova(disp_deep_season)
anova(disp_W4_season)
anova(disp_SW_season)


```

```{r test tide regime}
# region and season by tide regime
disp_noSW_tide_regime = betadisper(dist_noSW, sample_data(ps_noSW)$Tide.Regime)
disp_noSW_Feb_tide_regime = betadisper(dist_feb, sample_data(ps_noSW_Feb)$Tide.Regime) # really a test of overtopping effect
disp_noSW_Oct_tide_regime = betadisper(dist_oct, sample_data(ps_noSW_Oct)$Tide.Regime)
disp_shallow_feb_tide_regime = betadisper(dist_shallow_feb, sample_data(ps_shallow_feb)$Tide.Regime)
disp_intermediate_feb_tide_regime = betadisper(dist_intermediate_feb, sample_data(ps_intermediate_feb)$Tide.Regime)
disp_intermediate_oct_tide_regime = betadisper(dist_intermediate_oct, sample_data(ps_intermediate_oct)$Tide.Regime)
disp_deep_feb_tide_regime = betadisper(dist_deep_feb, sample_data(ps_deep_feb)$Tide.Regime)
disp_deep_oct_tide_regime = betadisper(dist_deep_oct, sample_data(ps_deep_oct)$Tide.Regime)
disp_W4_feb_tide_regime = betadisper(dist_W4_feb, sample_data(ps_W4_feb)$Tide.Regime)
disp_W4_oct_tide_regime = betadisper(dist_W4_oct, sample_data(ps_W4_oct)$Tide.Regime)

# hull ordiplot
plot(disp_noSW_tide_regime)
plot(disp_noSW_Feb_tide_regime)
plot(disp_noSW_Oct_tide_regime)
plot(disp_shallow_feb_tide_regime)
plot(disp_intermediate_feb_tide_regime)
plot(disp_intermediate_oct_tide_regime)
plot(disp_deep_feb_tide_regime)
plot(disp_deep_oct_tide_regime)
plot(disp_W4_feb_tide_regime)
plot(disp_W4_oct_tide_regime)

# boxplot
boxplot(disp_noSW_tide_regime, ylab = "Distance to centroid", xlab = "", main = "All samples (no SW)")
boxplot(disp_noSW_Feb_tide_regime, ylab = "Distance to centroid", xlab = "", main = "All samples (Feb)")
boxplot(disp_noSW_Oct_tide_regime, ylab = "Distance to centroid", xlab = "", main = "All samples (Oct)")
boxplot(disp_shallow_feb_tide_regime, ylab = "Distance to centroid", xlab = "", main = "Shallow (Feb)")
boxplot(disp_intermediate_feb_tide_regime, ylab = "Distance to centroid", xlab = "", main = "Intermediate (Feb)")
boxplot(disp_intermediate_oct_tide_regime, ylab = "Distance to centroid", xlab = "", main = "Intermediate (Oct)")
boxplot(disp_deep_feb_tide_regime, ylab = "Distance to centroid", xlab = "", main = "Deep (Feb)")
boxplot(disp_deep_oct_tide_regime, ylab = "Distance to centroid", xlab = "", main = "Deep (Oct)")
boxplot(disp_W4_feb_tide_regime, ylab = "Distance to centroid", xlab = "", main = "Well 4 (Feb)")
boxplot(disp_W4_oct_tide_regime, ylab = "Distance to centroid", xlab = "", main = "Well 4 (Oct)")

# permutation-based test for homogeneity of dispersion 
permutest(disp_noSW_tide_regime)
permutest(disp_noSW_Feb_tide_regime)
permutest(disp_noSW_Oct_tide_regime)
permutest(disp_shallow_feb_tide_regime)
permutest(disp_intermediate_feb_tide_regime)
permutest(disp_intermediate_oct_tide_regime)
permutest(disp_deep_feb_tide_regime)
permutest(disp_deep_oct_tide_regime)
permutest(disp_W4_feb_tide_regime)
permutest(disp_W4_oct_tide_regime)



```


```{r test tide}
# region and Feb by tide 
disp_noSW_feb_tide = betadisper(dist_feb, sample_data(ps_noSW_Feb)$Tide)
disp_shallow_feb_tide = betadisper(dist_shallow_feb, sample_data(ps_shallow_feb)$Tide)
disp_intermediate_feb_tide = betadisper(dist_intermediate_feb, sample_data(ps_intermediate_feb)$Tide)
disp_deep_feb_tide = betadisper(dist_deep_feb, sample_data(ps_deep_feb)$Tide)
disp_W4_feb_tide = betadisper(dist_W4_feb, sample_data(ps_W4_feb)$Tide)

# region and Feb and tide regime by tide
disp_shallow_feb_neap_tide = betadisper(dist_shallow_feb_neap, sample_data(ps_shallow_feb_neap)$Tide)
disp_shallow_feb_spring_tide = betadisper(dist_shallow_feb_spring, sample_data(ps_shallow_feb_spring)$Tide)
disp_intermediate_feb_neap_tide = betadisper(dist_intermediate_feb_neap, sample_data(ps_intermediate_feb_neap)$Tide)
disp_intermediate_feb_spring_tide = betadisper(dist_intermediate_feb_spring, sample_data(ps_intermediate_feb_spring)$Tide)
disp_deep_feb_neap_tide = betadisper(dist_deep_feb_neap, sample_data(ps_deep_feb_neap)$Tide)
disp_deep_feb_spring_tide = betadisper(dist_deep_feb_spring, sample_data(ps_deep_feb_spring)$Tide)
disp_W4_feb_neap_tide = betadisper(dist_W4_feb_neap, sample_data(ps_W4_feb_neap)$Tide)
disp_W4_feb_spring_tide = betadisper(dist_W4_feb_spring, sample_data(ps_W4_feb_spring)$Tide)

# hull ordiplot
plot(disp_noSW_feb_tide)
plot(disp_shallow_feb_tide)
plot(disp_intermediate_feb_tide)
plot(disp_deep_feb_tide)
plot(disp_W4_feb_tide)

plot(disp_shallow_feb_neap_tide)
plot(disp_shallow_feb_spring_tide)
plot(disp_intermediate_feb_neap_tide)
plot(disp_intermediate_feb_spring_tide)
plot(disp_deep_feb_neap_tide)
plot(disp_deep_feb_spring_tide)
plot(disp_W4_feb_neap_tide)
plot(disp_W4_feb_spring_tide)

# permutation-based test for homogeneity of dispersion 
permutest(disp_noSW_feb_tide)
permutest(disp_shallow_feb_tide)
permutest(disp_intermediate_feb_tide)
permutest(disp_deep_feb_tide)
permutest(disp_W4_feb_tide)

permutest(disp_shallow_feb_neap_tide)
permutest(disp_shallow_feb_spring_tide)
permutest(disp_intermediate_feb_neap_tide)
permutest(disp_intermediate_feb_spring_tide)
permutest(disp_deep_feb_neap_tide)
permutest(disp_deep_feb_spring_tide)
permutest(disp_W4_feb_neap_tide)
permutest(disp_W4_feb_spring_tide)




```


```{r test depth groups}
# check definition of depth groups
range(sample_data(pstrans_noNA)$Elevation[sample_data(pstrans_noNA)$Group == "Shallow"]) # 0.16 0.84
range(sample_data(pstrans_noNA)$Elevation[sample_data(pstrans_noNA)$Group == "Intermediate"]) # -0.71  0.05 (W2D2 Oct)
range(sample_data(pstrans_noNA)$Elevation[sample_data(pstrans_noNA)$Group == "Deep"]) # -1.17 -0.58



# test dispersion using vegan betadisper
disp_all = betadisper(dist_all, sample_data(pstrans_noNA)$Group) # not all depths... remove SW for test
disp_all_cluster = betadisper(dist_all, sample_data(pstrans_noNA)$Cluster) # not all depths... remove SW for test

disp_noSW = betadisper(dist_noSW, sample_data(ps_noSW)$Group)
disp_W1 = betadisper(dist_W1, sample_data(ps_W1)$Group) # can be one group factor
disp_W2 = betadisper(dist_W2, sample_data(ps_W2)$Group)
disp_W3 = betadisper(dist_W3, sample_data(ps_W3)$Group)
#disp_W4 = betadisper(dist_W4, sample_data(ps_W4)$depth_bin) # not depth...

# hull ordiplot
#png('Ordihull_all_Group_2024.03.25.png', units="in", width=5, height=5, res=300)
plot(disp_all, main = "all group")
plot(disp_all_cluster, main = "")

#dev.off()

plot(disp_noSW)
plot(disp_W1,main = "Well 1")
plot(disp_W2,main = "Well 2")
plot(disp_W3,main = "Well 3")
#plot(disp_W4,main = "Well 4")

# boxplot
boxplot(disp_all, ylab = "Distance to centroid", xlab = "", main = "All samples")
boxplot(disp_noSW, ylab = "Distance to centroid", xlab = "", main = "All samples (no SW)")
boxplot(disp_W1, ylab = "Distance to centroid", xlab = "", main = "Well 1")
boxplot(disp_W2, ylab = "Distance to centroid", xlab = "", main = "Well 2")
boxplot(disp_W3, ylab = "Distance to centroid", xlab = "", main = "Well 3")
#boxplot(disp_W4, ylab = "Distance to centroid", xlab = "", main = "Well 4")

# summary
print(disp_all, digits = max(3, getOption("digits") - 3), neigen = 8)
print(disp_noSW, digits = max(3, getOption("digits") - 3), neigen = 8)
print(disp_W1, digits = max(3, getOption("digits") - 3), neigen = 8)
print(disp_W2, digits = max(3, getOption("digits") - 3), neigen = 8)
print(disp_W3, digits = max(3, getOption("digits") - 3), neigen = 8)
#print(disp_W4, digits = max(3, getOption("digits") - 3), neigen = 8)

# Tukey's HSD
TukeyHSD(disp_all, which = "group", ordered = FALSE, conf.level = 0.95)
TukeyHSD(disp_noSW, which = "group", ordered = FALSE, conf.level = 0.95)
TukeyHSD(disp_W1, which = "group", ordered = FALSE, conf.level = 0.95)
TukeyHSD(disp_W2, which = "group", ordered = FALSE, conf.level = 0.95)
TukeyHSD(disp_W3, which = "group", ordered = FALSE, conf.level = 0.95)
#TukeyHSD(disp_W4, which = "group", ordered = FALSE, conf.level = 0.95)

# permutation-based test for homogeneity of dispersion 
permutest(disp_all)
permutest(disp_noSW)
permutest(disp_W1)
permutest(disp_W2)
permutest(disp_W3)
#permutest(disp_W4)

# ANOVA
anova(disp_all)
anova(disp_noSW)
anova(disp_W1)
anova(disp_W2)
anova(disp_W3)
#anova(disp_W4)

```

## 2. Analysis of Similarities - Anosim
```{r test of season/campaign}
# ANOSIM rank-order dissimilarities similar to NMDS
# NOTE: anosim can confound group differences with dispersion differences (want section before to be non-significant)
set.seed(123)
anosim_all = anosim(dist_all, sample_data(pstrans_noNA)$Campaign, permutations = 999)
anosim_noSW = anosim(dist_noSW, sample_data(ps_noSW)$Campaign, permutations = 999)
anosim_W1 = anosim(dist_W1, sample_data(ps_W1)$Campaign, permutations = 999)
anosim_W2 = anosim(dist_W2, sample_data(ps_W2)$Campaign, permutations = 999)
anosim_W3 = anosim(dist_W3, sample_data(ps_W3)$Campaign, permutations = 999)
anosim_W4 = anosim(dist_W4, sample_data(ps_W4)$Campaign, permutations = 999)

anosim_intermediate = anosim(dist_intermediate, sample_data(ps_intermediate)$Campaign, permutations = 999)
anosim_deep = anosim(dist_deep, sample_data(ps_deep)$Campaign, permutations = 999)
anosim_W4 = anosim(dist_W4, sample_data(ps_W4)$Campaign, permutations = 999)

# summary
print(anosim_all) 
print(anosim_noSW)
print(anosim_W1)
print(anosim_W2)
print(anosim_W3)
print(anosim_W4)

print(anosim_intermediate)
print(anosim_deep)
print(anosim_W4)

```

```{r test of tide regime}
# test
anosim_noSW_tide_regime = anosim(dist_noSW, sample_data(ps_noSW)$Tide.Regime, permutations = 999)
anosim_feb_tide_regime = anosim(dist_feb, sample_data(ps_noSW_Feb)$Tide.Regime, permutations = 999)
anosim_oct_tide_regime = anosim(dist_oct, sample_data(ps_noSW_Oct)$Tide.Regime, permutations = 999)
anosim_shallow_feb_tide_regime = anosim(dist_shallow_feb, sample_data(ps_shallow_feb)$Tide.Regime, permutations = 999)
anosim_intermediate_feb_tide_regime = anosim(dist_intermediate_feb, sample_data(ps_intermediate_feb)$Tide.Regime, permutations = 999)
anosim_intermediate_oct_tide_regime = anosim(dist_intermediate_oct, sample_data(ps_intermediate_oct)$Tide.Regime, permutations = 999)
anosim_deep_feb_tide_regime = anosim(dist_deep_feb, sample_data(ps_deep_feb)$Tide.Regime, permutations = 999)
anosim_deep_oct_tide_regime = anosim(dist_deep_oct, sample_data(ps_deep_oct)$Tide.Regime, permutations = 999)
anosim_W4_feb_tide_regime = anosim(dist_W4_feb, sample_data(ps_W4_feb)$Tide.Regime, permutations = 999)
anosim_W4_oct_tide_regime = anosim(dist_W4_oct, sample_data(ps_W4_oct)$Tide.Regime, permutations = 999)

# summary
print(anosim_noSW_tide_regime)
print(anosim_feb_tide_regime)
print(anosim_oct_tide_regime)
print(anosim_shallow_feb_tide_regime)
print(anosim_intermediate_feb_tide_regime)
print(anosim_intermediate_oct_tide_regime)
print(anosim_deep_feb_tide_regime)
print(anosim_deep_oct_tide_regime)
print(anosim_W4_feb_tide_regime)
print(anosim_W4_oct_tide_regime)

```

```{r test of tide}
#test
anosim_noSW_feb_tide = anosim(dist_feb, sample_data(ps_noSW_Feb)$Tide, permutations = 999)
anosim_shallow_feb_tide = anosim(dist_shallow_feb, sample_data(ps_shallow_feb)$Tide, permutations = 999)
anosim_intermediate_feb_tide = anosim(dist_intermediate_feb, sample_data(ps_intermediate_feb)$Tide, permutations = 999)
anosim_deep_feb_tide = anosim(dist_deep_feb, sample_data(ps_deep_feb)$Tide, permutations = 999)
anosim_W4_feb_tide = anosim(dist_W4_feb, sample_data(ps_W4_feb)$Tide, permutations = 999)

anosim_shallow_feb_neap_tide = anosim(dist_shallow_feb_neap, sample_data(ps_shallow_feb_neap)$Tide, permutations = 999)
anosim_shallow_feb_spring_tide = anosim(dist_shallow_feb_spring, sample_data(ps_shallow_feb_spring)$Tide, permutations = 999)
anosim_intermediate_feb_neap_tide = anosim(dist_intermediate_feb_neap, sample_data(ps_intermediate_feb_neap)$Tide, permutations = 999)
anosim_intermediate_feb_spring_tide = anosim(dist_intermediate_feb_spring, sample_data(ps_intermediate_feb_spring)$Tide, permutations = 999)
anosim_deep_feb_neap_tide = anosim(dist_deep_feb_neap, sample_data(ps_deep_feb_neap)$Tide, permutations = 999)
anosim_deep_feb_spring_tide = anosim(dist_deep_feb_spring, sample_data(ps_deep_feb_spring)$Tide, permutations = 999)
anosim_W4_feb_neap_tide = anosim(dist_W4_feb_neap, sample_data(ps_W4_feb_neap)$Tide, permutations = 999)
anosim_W4_feb_spring_tide = anosim(dist_W4_feb_spring, sample_data(ps_W4_feb_spring)$Tide, permutations = 999)

# summary
print(anosim_noSW_feb_tide)
print(anosim_shallow_feb_tide)
print(anosim_intermediate_feb_tide)
print(anosim_deep_feb_tide)
print(anosim_W4_feb_tide)

print(anosim_shallow_feb_neap_tide)
print(anosim_shallow_feb_spring_tide)
print(anosim_intermediate_feb_neap_tide)
print(anosim_intermediate_feb_spring_tide)
print(anosim_deep_feb_neap_tide)
print(anosim_deep_feb_spring_tide)
print(anosim_W4_feb_neap_tide)
print(anosim_W4_feb_spring_tide)

```


```{r test of depth groups}
# ANOSIM rank-order dissimilarities similar to NMDS
# NOTE: anosim can confound group differences with dispersion differences (want section before to be non-significant)
anosim_all = anosim(dist_all, sample_data(pstrans_noNA)$Group, permutations = 999) # CHANGE
anosim_noSW = anosim(dist_noSW, sample_data(ps_noSW)$Group, permutations = 999)
anosim_W1 = anosim(dist_W1, sample_data(ps_W1)$Group, permutations = 999)
anosim_W2 = anosim(dist_W2, sample_data(ps_W2)$Group, permutations = 999)
anosim_W3 = anosim(dist_W3, sample_data(ps_W3)$Group, permutations = 999)
#anosim_W4 = anosim(dist_W4, sample_data(ps_W4)$Group, permutations = 999) # CHANGE

# summary
print(anosim_all) 
print(anosim_noSW)
print(anosim_W1)
print(anosim_W2)
print(anosim_W3)
#print(anosim_W4)

```


## 3. Analysis of Variance - PERMANOVA
```{r test of season/campaign}
# adonis2
adonis_all = adonis2(dist_all ~ Campaign, data = data.frame(sample_data(pstrans_noNA)), permutations = 999)
adonis_noSW = adonis2(dist_noSW ~ Campaign, data = data.frame(sample_data(ps_noSW)), permutations = 999)
adonis_W1 = adonis2(dist_W1 ~ Campaign, data = data.frame(sample_data(ps_W1)), permutations = 999)
adonis_W2 = adonis2(dist_W2 ~ Campaign, data = data.frame(sample_data(ps_W2)), permutations = 999)
adonis_W3 = adonis2(dist_W3 ~ Campaign, data = data.frame(sample_data(ps_W3)), permutations = 999)
adonis_W4 = adonis2(dist_W4 ~ Campaign, data = data.frame(sample_data(ps_W4)), permutations = 999)

adonis_intermediate = adonis2(dist_intermediate ~ Campaign, data = data.frame(sample_data(ps_intermediate)), permutations = 999)
adonis_deep = adonis2(dist_deep ~ Campaign, data = data.frame(sample_data(ps_deep)), permutations = 999)
adonis_W4 = adonis2(dist_W4 ~ Campaign, data = data.frame(sample_data(ps_W4)), permutations = 999)

# summary
print(adonis_all)
print(adonis_noSW)
print(adonis_W1)
print(adonis_W2)
print(adonis_W3)
print(adonis_W4)

print(adonis_intermediate)
print(adonis_deep)
print(adonis_W4)

```

```{r test of tide regime}
adonis_noSW_tide_regime = adonis2(dist_noSW ~ Tide.Regime, data = data.frame(sample_data(ps_noSW)), permutations = 999)
adonis_feb_tide_regime = adonis2(dist_feb ~ Tide.Regime, data = data.frame(sample_data(ps_noSW_Feb)), permutations = 999)
adonis_oct_tide_regime = adonis2(dist_oct ~ Tide.Regime, data = data.frame(sample_data(ps_noSW_Oct)), permutations = 999)
adonis_shallow_feb_tide_regime = adonis2(dist_shallow_feb ~ Tide.Regime, data = data.frame(sample_data(ps_shallow_feb)), permutations = 999)
adonis_intermediate_feb_tide_regime = adonis2(dist_intermediate_feb ~ Tide.Regime, data = data.frame(sample_data(ps_intermediate_feb)), permutations = 999)
adonis_intermediate_oct_tide_regime = adonis2(dist_intermediate_oct ~ Tide.Regime, data = data.frame(sample_data(ps_intermediate_oct)), permutations = 999)
adonis_deep_feb_tide_regime = adonis2(dist_deep_feb ~ Tide.Regime, data = data.frame(sample_data(ps_deep_feb)), permutations = 999)
adonis_deep_oct_tide_regime = adonis2(dist_deep_oct ~ Tide.Regime, data = data.frame(sample_data(ps_deep_oct)), permutations = 999)
adonis_W4_feb_tide_regime = adonis2(dist_W4_feb ~ Tide.Regime, data = data.frame(sample_data(ps_W4_feb)), permutations = 999)
adonis_W4_oct_tide_regime = adonis2(dist_W4_oct ~ Tide.Regime, data = data.frame(sample_data(ps_W4_oct)), permutations = 999)

# summary
print(adonis_noSW_tide_regime)
print(adonis_feb_tide_regime)
print(adonis_oct_tide_regime)
print(adonis_shallow_feb_tide_regime)
print(adonis_intermediate_feb_tide_regime)
print(adonis_intermediate_oct_tide_regime)
print(adonis_deep_feb_tide_regime)
print(adonis_deep_oct_tide_regime)
print(adonis_W4_feb_tide_regime)
print(adonis_W4_oct_tide_regime)

```

```{r test of tide}
adonis_noSW_feb_tide = adonis2(dist_feb ~ Tide, data = data.frame(sample_data(ps_noSW_Feb)), permutations = 999)
adonis_shallow_feb_tide = adonis2(dist_shallow_feb ~ Tide, data = data.frame(sample_data(ps_shallow_feb)), permutations = 999)
adonis_intermediate_feb_tide = adonis2(dist_intermediate_feb ~ Tide, data = data.frame(sample_data(ps_intermediate_feb)), permutations = 999)
adonis_deep_feb_tide = adonis2(dist_deep_feb ~ Tide, data = data.frame(sample_data(ps_deep_feb)), permutations = 999)
adonis_W4_feb_tide = adonis2(dist_W4_feb ~ Tide, data = data.frame(sample_data(ps_W4_feb)), permutations = 999)

adonis_shallow_feb_neap_tide = adonis2(dist_shallow_feb_neap ~ Tide, data = data.frame(sample_data(ps_shallow_feb_neap)), permutations = 999)
adonis_shallow_feb_spring_tide = adonis2(dist_shallow_feb_spring ~ Tide, data = data.frame(sample_data(ps_shallow_feb_spring)), permutations = 999)
adonis_intermediate_feb_neap_tide = adonis2(dist_intermediate_feb_neap ~ Tide, data = data.frame(sample_data(ps_intermediate_feb_neap)), permutations = 999)
adonis_intermediate_feb_spring_tide = adonis2(dist_intermediate_feb_spring ~ Tide, data = data.frame(sample_data(ps_intermediate_feb_spring)), permutations = 999)
adonis_deep_feb_neap_tide = adonis2(dist_deep_feb_neap ~ Tide, data = data.frame(sample_data(ps_deep_feb_neap)), permutations = 999)
adonis_deep_feb_spring_tide = adonis2(dist_deep_feb_spring ~ Tide, data = data.frame(sample_data(ps_deep_feb_spring)), permutations = 999)
adonis_W4_feb_neap_tide = adonis2(dist_W4_feb_neap ~ Tide, data = data.frame(sample_data(ps_W4_feb_neap)), permutations = 999)
adonis_W4_feb_spring_tide = adonis2(dist_W4_feb_spring ~ Tide, data = data.frame(sample_data(ps_W4_feb_spring)), permutations = 999)

# summary
print(adonis_noSW_feb_tide)
print(adonis_shallow_feb_tide)
print(adonis_intermediate_feb_tide)
print(adonis_deep_feb_tide)
print(adonis_W4_feb_tide)
  
print(adonis_shallow_feb_neap_tide)
print(adonis_shallow_feb_spring_tide)
print(adonis_intermediate_feb_neap_tide)
print(adonis_intermediate_feb_spring_tide)
print(adonis_deep_feb_neap_tide)
print(adonis_deep_feb_spring_tide)
print(adonis_W4_feb_neap_tide)
print(adonis_W4_feb_spring_tide)

```

```{r test of depth groups}
adonis_all = adonis2(dist_all ~ Group, data = data.frame(sample_data(pstrans_noNA)), permutations = 999)
adonis_noSW = adonis2(dist_noSW ~ Group, data = data.frame(sample_data(ps_noSW)), permutations = 999)
adonis_W1 = adonis2(dist_W1 ~ Group, data = data.frame(sample_data(ps_W1)), permutations = 999)
adonis_W2 = adonis2(dist_W2 ~ Group, data = data.frame(sample_data(ps_W2)), permutations = 999)
adonis_W3 = adonis2(dist_W3 ~ Group, data = data.frame(sample_data(ps_W3)), permutations = 999)
#adonis_W4 = adonis2(dist_W4 ~ Group, data = data.frame(sample_data(ps_W4)), permutations = 999)

# summary
print(adonis_all)
print(adonis_noSW)
print(adonis_W1)
print(adonis_W2)
print(adonis_W3)
#print(adonis_W4)

```


```{r test of depth bin}
adonis_all = adonis2(dist_all ~ depth_bin, data = data.frame(sample_data(pstrans_noNA)), permutations = 999)
adonis_W1 = adonis2(dist_W1 ~ depth_bin, data = data.frame(sample_data(ps_W1)), permutations = 999)
adonis_W2 = adonis2(dist_W2 ~ depth_bin, data = data.frame(sample_data(ps_W2)), permutations = 999)
adonis_W3 = adonis2(dist_W3 ~ depth_bin, data = data.frame(sample_data(ps_W3)), permutations = 999)
adonis_W4 = adonis2(dist_W4 ~ depth_bin, data = data.frame(sample_data(ps_W4)), permutations = 999)

# summary
print(adonis_all)
print(adonis_W1)
print(adonis_W2)
print(adonis_W3)
print(adonis_W4)

```

```{r test of continuous depth}
sample_data(pstrans_noNA)$Elevation[sample_data(pstrans_noNA)$Type == "Seawater"] = 1 

adonis_all = adonis2(dist_all ~ Elevation, data = data.frame(sample_data(pstrans_noNA)), permutations = 999)
adonis_W1 = adonis2(dist_W1 ~ depth_bin, data = data.frame(sample_data(ps_W1)), permutations = 999)
adonis_W2 = adonis2(dist_W2 ~ depth_bin, data = data.frame(sample_data(ps_W2)), permutations = 999)
adonis_W3 = adonis2(dist_W3 ~ depth_bin, data = data.frame(sample_data(ps_W3)), permutations = 999)
adonis_W4 = adonis2(dist_W4 ~ depth_bin, data = data.frame(sample_data(ps_W4)), permutations = 999)

# summary
print(adonis_all)
print(adonis_W1)
print(adonis_W2)
print(adonis_W3)
print(adonis_W4)

```



## 4. Mantel environmental correlation - Bioenv 
```{r test of best env correlates}
# default spearman is non-parametric
# bioenv selects variables to maximize the Mantel correlation
bioenv_all = bioenv(dist_all, sample_data(pstrans_noNA)[,c("Salinity", "Temperature", "DO", "pH", "Nitrate", "Nitrite", "Ammonium", "Phosphate", "DOC", "HIX", "Fpeak")]) 
bioenv_noSW = bioenv(dist_noSW, sample_data(ps_noSW)[,c("Salinity", "Temperature", "DO", "pH", "Nitrate", "Nitrite", "Ammonium", "Phosphate", "DOC", "HIX", "Fpeak")])

bioenv_feb = bioenv(dist_feb, sample_data(ps_noSW_Feb)[,c("Salinity", "Temperature", "DO", "pH", "Nitrate", "Nitrite", "Ammonium", "Phosphate", "DOC", "HIX", "Fpeak")])
bioenv_oct = bioenv(dist_oct, sample_data(ps_noSW_Oct)[,c("Salinity", "Temperature", "DO", "pH", "Nitrate", "Nitrite", "Ammonium", "Phosphate", "DOC", "HIX", "Fpeak")])

bioenv_W1 = bioenv(dist_W1, sample_data(ps_W1)[,c("Salinity", "Temperature", "DO", "pH", "Nitrate", "Nitrite", "Ammonium", "Phosphate", "DOC", "HIX", "Fpeak")])
bioenv_W2 = bioenv(dist_W2, sample_data(ps_W2)[,c("Salinity", "Temperature", "DO", "pH", "Nitrate", "Nitrite", "Ammonium", "Phosphate", "DOC", "HIX", "Fpeak")],)
bioenv_W3 = bioenv(dist_W3, sample_data(ps_W3)[,c("Salinity", "Temperature", "DO", "pH", "Nitrate", "Nitrite", "Ammonium", "Phosphate", "DOC", "HIX", "Fpeak")])
bioenv_W4 = bioenv(dist_W4, sample_data(ps_W4)[,c("Salinity", "Temperature", "DO", "pH", "Nitrate", "Nitrite", "Ammonium", "Phosphate", "DOC", "HIX", "Fpeak")])

bioenv_shallow = bioenv(dist_shallow, sample_data(ps_shallow)[,c("Salinity", "Temperature", "DO", "pH", "Nitrate", "Nitrite", "Ammonium", "Phosphate", "DOC", "HIX", "Fpeak")])
bioenv_intermediate = bioenv(dist_intermediate, sample_data(ps_intermediate)[,c("Salinity", "Temperature", "DO", "pH", "Nitrate", "Nitrite", "Ammonium", "Phosphate", "DOC", "HIX", "Fpeak")])
bioenv_deep = bioenv(dist_deep, sample_data(ps_deep)[,c("Salinity", "Temperature", "DO", "pH", "Nitrate", "Nitrite", "Ammonium", "Phosphate", "DOC", "HIX", "Fpeak")])

bioenv_shallow_feb = bioenv(dist_shallow_feb, sample_data(ps_shallow_feb)[,c("Salinity", "Temperature", "DO", "pH", "Nitrate", "Nitrite", "Ammonium", "Phosphate", "DOC", "HIX", "Fpeak")])
bioenv_intermediate_feb = bioenv(dist_intermediate_feb, sample_data(ps_intermediate_feb)[,c("Salinity", "Temperature", "DO", "pH", "Nitrate", "Nitrite", "Ammonium", "Phosphate", "DOC", "HIX", "Fpeak")])
bioenv_deep_feb = bioenv(dist_deep_feb, sample_data(ps_deep_feb)[,c("Salinity", "Temperature", "DO", "pH", "Nitrate", "Nitrite", "Ammonium", "Phosphate", "DOC", "HIX", "Fpeak")])
bioenv_W4_feb = bioenv(dist_W4_feb, sample_data(ps_W4_feb)[,c("Salinity", "Temperature", "DO", "pH", "Nitrate", "Nitrite", "Ammonium", "Phosphate", "DOC", "HIX", "Fpeak")])


bioenv_intermediate_oct = bioenv(dist_intermediate_oct, sample_data(ps_intermediate_oct)[,c("Salinity", "Temperature", "DO", "pH", "Nitrate", "Nitrite", "Ammonium", "Phosphate", "DOC", "HIX", "Fpeak")])
bioenv_deep_oct = bioenv(dist_deep_oct, sample_data(ps_deep_oct)[,c("Salinity", "Temperature", "DO", "pH", "Nitrate", "Nitrite", "Ammonium", "Phosphate", "DOC", "HIX", "Fpeak")])
bioenv_W4_oct = bioenv(dist_W4_oct, sample_data(ps_W4_oct)[,c("Salinity", "Temperature", "DO", "pH", "Nitrate", "Nitrite", "Ammonium", "Phosphate", "DOC", "HIX", "Fpeak")])


ps_W4Feb = subset_samples(pstrans_noNA, Well == "Well 4" & Campaign == "February 2022")

dist_W4Feb = phyloseq::distance(ps_W4Feb, method = "bray")

bioenv_W4Feb = bioenv(dist_W4Feb, sample_data(ps_W4Feb)[,c("Salinity", "Temperature", "DO", "pH", "Nitrate", "Nitrite", "Ammonium", "Phosphate", "DOC", "HIX", "Fpeak")])


# summary
print(bioenv_all)
print(bioenv_noSW)

print(bioenv_feb)
print(bioenv_oct)

print(bioenv_W1)
print(bioenv_W2)
print(bioenv_W3)
print(bioenv_W4)
print(bioenv_W4Feb)

print(bioenv_shallow)
print(bioenv_intermediate)
print(bioenv_deep)

print(bioenv_shallow_feb)
print(bioenv_intermediate_feb)
print(bioenv_deep_feb)
print(bioenv_W4_feb)

print(bioenv_intermediate_oct)
print(bioenv_deep_oct)
print(bioenv_W4_oct)


summary(bioenv_all)
summary(bioenv_noSW)

summary(bioenv_feb)
summary(bioenv_oct)

summary(bioenv_W1)
summary(bioenv_W2)
summary(bioenv_W3)
summary(bioenv_W4)
summary(bioenv_W4Feb)

summary(bioenv_shallow)
summary(bioenv_intermediate)
summary(bioenv_deep)

summary(bioenv_shallow_feb)
summary(bioenv_intermediate_feb)
summary(bioenv_deep_feb)
summary(bioenv_W4_feb)

summary(bioenv_intermediate_oct)
summary(bioenv_deep_oct)
summary(bioenv_W4_oct)

```


```{r}
# check number of samples in each dataset
nrow(sample_data(pstrans_noNA))
nrow(sample_data(ps_noSW))
nrow(sample_data(ps_W1))
nrow(sample_data(ps_W2))
nrow(sample_data(ps_W3))
nrow(sample_data(ps_W4))
nrow(sample_data(ps_W4Feb))

nrow(sample_data(ps_shallow))
nrow(sample_data(ps_intermediate))
nrow(sample_data(ps_deep))

nrow(sample_data(ps_shallow_feb))
nrow(sample_data(ps_intermediate_feb))
nrow(sample_data(ps_deep_feb))
nrow(sample_data(ps_W4_feb))

nrow(sample_data(ps_intermediate_oct))
nrow(sample_data(ps_deep_oct))
nrow(sample_data(ps_W4_oct))



```



## 7. Ordination of environmental data
```{r}
# extract standardized distance matrix from bioenv
env_dist_all = data.frame(as.matrix(bioenvdist(bioenv_all)))
rownames(env_dist_all)
colnames(env_dist_all) <- rownames(env_dist_all)

pcoa_env = capscale(env_dist_all ~ 1)
scores(pcoa_env)

# CAP
set.seed(123)
pcoa_comm = ordinate(pstrans_noNA, formula= ~ 1, "CAP", distance = "bray") # same as MDS
scores(pcoa_comm)

# NOTE: pcoa() function is from ape package and not compatible with vegan's scores() function
```



## 8. Procrustes to compare community and environmental ordinations
```{r}
# procrustes compares two ordinations
proc_all = procrustes(pcoa_env, pcoa_comm)
plot(proc_all) 


```



## 9. Hierarchical Clustering
```{r}
set.seed(123)

# bray dissimilarity matrix
#pstrans = transform_sample_counts(ps, function(x) asinh(x)) 
#mat = phyloseq::distance(physeq = pstrans, method = "bray", type = "samples")
#mat2 = (as.matrix(mat))

# bray dissimilarity matrix ~ untransformed
mat = phyloseq::distance(physeq = ps, method = "bray", type = "samples")
mat2 = (as.matrix(mat))

# bray dissimilarity matrix ~ proportional
#ps_prop = transform_sample_counts(ps, function(x) 100*x/sum(x))


# heatmap of ordination with dendrogram
png('Bray_Matrix_Untransformed_All_2024.02.26.png', units="in", width=25, height=25, res=300)
heatmap(x = mat2, col = magma(20), symm = T) 
dev.off()

h1 = heatmap(x = mat2, col = magma(20), symm = T) 
r1list = h1$rowInd

r1list_sample = as.data.frame(chem$SampleID[r1list[1:187]])
colnames(r1list_sample) = c("ClusterSampleID")

# cutoffs: 43, 56, 95, 138, 187
r1list_sample$Cluster = rep(c("cluster1","cluster2", "cluster3", "cluster4", "cluster5"), c(43, 13, 39, 43, 49)) # by hand, search
r1list_sample$Cluster1 = rep(c("yes","no"), c(43, 144))
r1list_sample$Cluster2 = rep(c("no", "yes","no"), c(43, 13, 131))
r1list_sample$Cluster3 = rep(c("no", "yes","no"), c(56, 39, 92))
r1list_sample$Cluster4 = rep(c("no", "yes","no"), c(95, 43, 49))
r1list_sample$Cluster5 = rep(c("no", "yes"), c(138, 49))
r1list_sample$ClusterOrder = 1:nrow(r1list_sample)

# cutoffs: 40, 50, 110, 138, 187 # updated 2024.02.22 with asinh transform ~ dont use!
#r1list_sample$Cluster = rep(c("cluster1","cluster2", "cluster3", "cluster4", "cluster5"), c(40, 10, 60, 28, 49)) # by hand, search
# r1list_sample$Cluster1 = rep(c("yes","no"), c(43, 144)) 
# r1list_sample$Cluster2 = rep(c("no", "yes","no"), c(43, 13, 131)) 
# r1list_sample$Cluster3 = rep(c("no", "yes","no"), c(56, 39, 92)) 
# r1list_sample$Cluster4 = rep(c("no", "yes","no"), c(95, 43, 49)) 
# r1list_sample$Cluster5 = rep(c("no", "yes"), c(138, 49)) 

# add to phyloseq object
row.names(r1list_sample) = r1list_sample$ClusterSampleID
sam.new <- sample_data(r1list_sample)
psclust <- merge_phyloseq(ps, sam.new)
saveRDS(psclust, "ps_Clusters_Stinson_2024.02.26.rds")
check = sample_data(psclust)
check_data = data.frame(sample_data(psclust))

# png('Bray_Matrix_Untransformed_All_Complex_2024.02.15.png', units="in", width=25, height=25, res=300)
# Heatmap(mat2)
# dev.off()

h2 = Heatmap(mat2)
r2list = row_order(h2)

reduced = data.frame(check_data$ClusterSampleID, check_data$Cluster, check_data$ClusterOrder)
colnames(reduced) = c("SampleID", "Cluster", "ClusterOrder")
write.csv(reduced, "Bray_Clusters_SampleID.csv", row.names = F)

```


# Part 4. Phylum-level analyses

## 1. Stacked Bar Phyla
```{r define colors}
colors_matt_shade =c( "#242526", "#a7b8c2", "#f0ece2", "#dfd3c3", "#c7b198" ,"#9c4f41", "#bf2651", "#4d6a94", "#b3785d", "#e6cc8a", "#997d76", "#ffd3a3" ,"#52216e", "#ffac7f" ,"#ff8766" ,"#e6a3a3", "#d6a57a", "#4b3d53" ,"#911d55", "#fafac3", "#99c2db", "#995c95", "#93c2a7", "#66333d", "#524a63", "#4c8f82" ,"#b2c5c7" ,"#357985" ,"#728794",  "#dfeded","#242b4a")

```


```{r extract abundant phyla}

#psclust_long = psmelt(psclust) # takes a while
#write.csv(psclust_long, "ps_MELTED_Clusters_Stinson_2024.02.26.csv", row.names = F)
psclust_long = read.csv("ps_MELTED_Clusters_Stinson_2024.02.26.csv")

# summarize by sample ID
psclust_sum_phyla = psclust_long %>% group_by(SampleID, Phylum) %>% summarise(All = sum(Abundance)) # not proportional!

# total ASVs with abundance > 0
psclust_long_asvs = psclust_long %>% group_by(OTU) %>% summarise(All = sum(Abundance))
psclust_long_asvs = subset(psclust_long_asvs, All > 0)

# total reads
sum(psclust_long$Abundance) # 7613584

# phyla comprising 0.1% of all reads (Ruiz Gonzales 2022)
all_phyla_reads = psclust_long %>% group_by(Phylum) %>% summarise(Total = sum(Abundance)) 
all_phyla_reads$Proportion = (all_phyla_reads$Total)/(sum(psclust_long$Abundance))*100
all_phyla_reads = subset(all_phyla_reads, Total > 0)

ggplot(all_phyla_reads, aes(x = reorder(Phylum, Proportion), y = Proportion)) +
  geom_bar(stat = "identity") +
  coord_flip() +
  theme_bw() +
  labs(x = "", y = "Percent of Total Reads (%)")

ggsave("phyla_rank_percent_2024.03.26.png", plot = last_plot(), 
       units = "cm", width = 25, height = 25, dpi = 300)

ggplot(all_phyla_reads, aes(x = reorder(Phylum, Total), y = Total)) +
  geom_bar(stat = "identity") +
  coord_flip() +
  theme_bw() +
  labs(x = "", y = "\nTotal Read Count")

ggsave("phyla_rank_reads_2024.03.26.png", plot = last_plot(), 
       units = "cm", width = 25, height = 25, dpi = 300)


sub_phyla_reads = subset(all_phyla_reads, Proportion > 0.1 & Phylum != "unclassified Bacteria")
sub_phyla_reads = sub_phyla_reads[order(-sub_phyla_reads$Proportion),]
sub_phyla_reads$Phylum

ggplot(sub_phyla_reads, aes(x = reorder(Phylum, Proportion), y = Proportion)) +
  geom_bar(stat = "identity") +
  coord_flip() +
  theme_bw() +
  labs(x = "", y = "\nPercent of Total Reads (%)")

ggsave("reduced_phyla_rank_percent_2024.03.27.png", plot = last_plot(), 
       units = "cm", width = 15, height = 15, dpi = 300)

# how many of the reads are represented by the phyla > 0.1% ?
sum(sub_phyla_reads$Total)/sum(psclust_long$Abundance) # 0.954094

phyla_list_0.1 = sub_phyla_reads$Phylum # keeps the order for plotting!

# total reads per sample
#psclust_sum_sample = psclust_long %>% group_by(SampleID) %>% summarise(Reads = sum(Abundance)) 

# check_data = subset(data.frame(sample_data(ps)), Type != "Sand")
# 
# check_data2 = merge(psclust_sum_sample, check_data, by = "SampleID")
# 
# check_data2$SampleID = as.factor(check_data2$SampleID)
# check_data2$SampleID = factor(check_data2$SampleID, levels = c(rev(r1list_sample$ClusterSampleID)))
# 
# ggplot(check_data2, aes(x = SampleID, y = Reads)) +
#   geom_bar(stat = "identity") +
#   coord_flip() +
#   theme_bw() +
#   labs(x = "", y = "\nTotal Read Count")
# 
# ggsave("sample_cluster_order_reads_2024.02.26.png", plot = last_plot(), 
#        units = "cm", width = 25, height = 50, dpi = 300)
# 
# ggplot(check_data2, aes(x = reorder(SampleID, Reads), y = Reads)) +
#   geom_bar(stat = "identity") +
#   coord_flip() +
#   theme_bw() +
#   labs(x = "", y = "\nTotal Read Count")
# 
# ggsave("sample_abun_order_reads_2024.02.26.png", plot = last_plot(), 
#        units = "cm", width = 25, height = 50, dpi = 300)

# merge sample separated with total per sample
psclust_sum_phyla2 = left_join(psclust_sum_phyla, psclust_sum_sample, by = "SampleID")

# take proportion of sample-phylum per sample total
psclust_sum_phyla2$Proportion = (psclust_sum_phyla2$All)/(psclust_sum_phyla2$Reads)
psclust_sum_phyla2$Percent = (psclust_sum_phyla2$All)/(psclust_sum_phyla2$Reads)*100

# subset to phyla > 0.1% of total reads
psclust_sum_phyla2_sub = subset(psclust_sum_phyla2, Phylum %in% c(phyla_list_0.1))
unique(psclust_sum_phyla2_sub$Phylum) # check

# create an "other" phyla group
psclust_sum_phyla2_other = psclust_sum_phyla2_sub %>% group_by(SampleID) %>% summarise(Percent = 100 - sum(Percent))
psclust_sum_phyla2_other$Phylum = "Other"

# Note: another method starting with ps object
# Rename all other phyla not in this list to "Other"
#phyla_keep = phyla_list_0.1 # how to do this trim in phyloseq?
#'%!in%' <- function(x,y)!('%in%'(x,y))
#tax_table(ps)[tax_table(ps)[,2] %!in% phyla_keep,2] <- "Other"

# trim table of phyla over 0.1%
psclust_sum_phyla_0.1 = data.frame(psclust_sum_phyla2_sub$SampleID, psclust_sum_phyla2_sub$Percent, psclust_sum_phyla2_sub$Phylum)
colnames(psclust_sum_phyla_0.1) = c("SampleID", "Percent", "Phylum")

# bind dataframes
psclust_phyla = rbind(psclust_sum_phyla_0.1, psclust_sum_phyla2_other)

# merge with metadata
phyla_table = merge(check_data2, psclust_phyla, by = "SampleID")

```


```{r plot ordered by cluster}
# plot in order of dendrogram
class(phyla_table$SampleID)
levels(phyla_table$SampleID)

class(phyla_table$Phylum)
phyla_table$Phylum = as.factor(phyla_table$Phylum)
levels(phyla_table$Phylum)
phyla_order = c("Proteobacteria", "Nanoarchaeota",  "Verrucomicrobiota", "Patescibacteria", "Bacteroidota", "Bdellovibrionota", "Chloroflexi", "Planctomycetota", "Crenarchaeota", "Firmicutes", "Acidobacteriota", "Campylobacterota",  "Myxococcota", "Desulfobacterota", "Actinobacteriota", "Dependentiae", "Elusimicrobiota",   "Thermoplasmatota", "Nitrospirota", "Methylomirabilota", "Aenigmarchaeota", "Gemmatimonadota", "SAR324 clade(Marine group B)", "Cyanobacteria", "Nitrospinota", "Marinimicrobia (SAR406 clade)", "DTB120", "Latescibacterota", "Micrarchaeota", "WOR-1", "Other")
phyla_table$Phylum = factor(phyla_table$Phylum, levels = c(rev(phyla_order)))

ggplot(phyla_table, aes(x = Percent, y = reorder(SampleID, ClusterOrder))) + # rev()
  #facet_wrap(vars(Cluster), nrow = 3, ncol = 2, scales = "free") +
  geom_bar(stat = "identity", aes(fill = Phylum)) +
  theme_bw() +
  theme(strip.text = element_text(size = 11)) +
  #theme(legend.position = "none") +
  #scale_y_continuous(limits = c(0, 50)) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) + 
  scale_fill_manual(values = colors_matt_shade, name = "Phyla") +
  labs(x = "Relative Abundance (%)\n", y = "")

ggsave("Phyla_Matt_Ordered_Dendrogram_Stacked_2024.02.16.png", plot = last_plot(),
       units = "cm", width = 40, height = 50, dpi = 300)


# reverse for legend
phyla_table$Phylum = factor(phyla_table$Phylum, levels = c((phyla_order)))

ggplot(phyla_table, aes(x = Percent, y = SampleID)) +
  #facet_wrap(vars(Cluster), nrow = 3, ncol = 2, scales = "free") +
  geom_bar(stat = "identity", aes(fill = Phylum)) +
  theme_bw() +
  theme(strip.text = element_text(size = 11)) +
  #theme(legend.position = "none") +
  #scale_y_continuous(limits = c(0, 50)) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) + 
  scale_fill_manual(values = c(rev(colors_matt)), name = "Phyla") +
  labs(x = "Relative Abundance (%)\n", y = "")

ggsave("Phyla_Reverse_Ordered_Dendrogram_Stacked_2024.02.16.png", plot = last_plot(),
       units = "cm", width = 40, height = 50, dpi = 300)


phyla_table$Phylum = factor(phyla_table$Phylum, levels = c(rev(phyla_order)))

# separate by cluster ~ doesn't keep consistent width of bars
ggplot(phyla_table, aes(x = Percent, y = SampleID)) +
  facet_wrap(vars(Cluster), nrow = 5, ncol = 1, scales = "free") +
  geom_bar(stat = "identity", aes(fill = Phylum)) +
  theme_bw() +
  theme(strip.text = element_text(size = 11)) +
  #theme(legend.position = "none") +
  #scale_y_continuous(limits = c(0, 50)) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) + 
  scale_fill_manual(values = colors_matt, name = "Phyla") +
  labs(x = "Relative Abundance (%)\n", y = "")
```


```{r plot ordered by well, depth, date, tide}
# sort in excel by Well, Depth, Date_Tide
#write.csv(phyla_table, "Phylum_Other_Sample_2024.02.26.csv", row.names = F)

#phyla_table_WD = read.csv("Phylum_Other_Sample_NEW_ORDER_2024.02.26.csv")

#write.csv(check_data2, "Metadata_clusters_2024.02.72.csv", row.names = F)

check_data2_sort = read.csv("Metadata_clusters_sorted_2024.02.72.csv")
check_data2_sort = read.csv("Metadata_clusters_ORDER_depthbin_2024.02.27.csv")

phyla_table_2 = phyla_table

phyla_table_2$SampleID = factor(phyla_table_2$SampleID, levels = c(rev(check_data2_sort$SampleID)))
levels(phyla_table_2$SampleID)

# ordered by well depth
ggplot(phyla_table_2, aes(x = Percent, y = SampleID)) +
  geom_bar(stat = "identity", aes(fill = Phylum)) +
  theme_bw() +
  theme(strip.text = element_text(size = 11)) +
  #theme(legend.position = "none") +
  #scale_y_continuous(limits = c(0, 50)) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) + 
  scale_fill_manual(values = colors_matt_shade, name = "Phyla") +
  labs(x = "Relative Abundance (%)\n", y = "")

ggsave("Phyla_WellDepthbin_Ordered_2024.03.27.png", plot = last_plot(),
       units = "cm", width = 40, height = 50, dpi = 300)


# including date tide
#check_data2_sort$depth_bin = as.character(check_data2_sort$depth_bin)
check_data2_sort$SampleWDBDT = paste(check_data2_sort$W, " ", check_data2_sort$depth_bin, " ", check_data2_sort$Date_Tide)

#phyla_table_2$depth_bin = as.character(phyla_table_2$depth_bin)
phyla_table_2$SampleWDBDT = paste(phyla_table_2$W, " ", phyla_table_2$depth_bin, " ", phyla_table_2$Date_Tide)

phyla_table_2$SampleWDBDT = factor(phyla_table_2$SampleWDBDT, levels = c(rev(check_data2_sort$SampleWDBDT)))
levels(phyla_table_2$SampleWDBDT)

ggplot(phyla_table_2, aes(x = Percent, y = SampleWDBDT)) +
  geom_bar(stat = "identity", aes(fill = Phylum)) +
  theme_bw() +
  theme(strip.text = element_text(size = 11)) +
  #theme(legend.position = "none") +
  #scale_y_continuous(limits = c(0, 50)) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) + 
  scale_fill_manual(values = colors_matt_shade, name = "Phyla") +
  labs(x = "Relative Abundance (%)\n", y = "")

ggsave("Phyla_WellDepthbin_DateTide_Ordered_2024.03.27.png", plot = last_plot(),
       units = "cm", width = 40, height = 50, dpi = 300)

write.csv(phyla_table_2, "Phylum_Table_2024.03.27.csv", row.names = F)

```


```{r bars of other variables to compare}
# plot other things with same sample order
class(check_data2_sort$SampleWDBDT)
check_data2_sort$SampleWDBDT = as.factor(check_data2_sort$SampleWDBDT)
levels(check_data2_sort$SampleWDBDT)
check_data2_sort$SampleWDBDT = factor(check_data2_sort$SampleWDBDT, levels = c(rev(check_data2_sort$SampleWDBDT)))
levels(check_data2_sort$SampleWDBDT)

ggplot(check_data2_sort, aes(x = log10(Cells_mL), y = SampleWDBDT)) +
  geom_bar(stat = "identity", aes(fill = "identity")) +
  scale_fill_manual(values = "grey") +
  #theme_bw() +
  theme(strip.text = element_text(size = 11)) +
  theme(legend.position = "none") +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  theme(panel.background = element_blank()) +
  labs(x = "Cell Density (ml-1)\n", y = "")


ggsave("Log_Cell_Density_DateTide_Ordered_2024.02.28.png", plot = last_plot(),
       units = "cm", width = 10, height = 50, dpi = 300)

ggplot(check_data2_sort, aes(x = (Cells_mL), y = SampleWDBDT)) +
  geom_bar(stat = "identity", aes(fill = "identity")) +
  scale_fill_manual(values = "grey") +
  #theme_bw() +
  theme(strip.text = element_text(size = 11)) +
  theme(legend.position = "none") +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  theme(panel.background = element_blank()) +
  labs(x = "Cell Density (ml-1)\n", y = "")


ggsave("Cell_Density_DateTide_Ordered_2024.02.28.png", plot = last_plot(),
       units = "cm", width = 10, height = 50, dpi = 300)
```


## 2. Environmental correlates with abundant phyla
```{r proportional reads and spearman}

cols_keep_red = c("SampleID", "DO_uM", "Salinity", "Temperature", "pH", "DOC", "Nitrite", "Nitrate", "Ammonium", "Phosphate", "HIX", "Fpeak" ) 
trim.env2 = env[, (colnames(env) %in% cols_keep_red)]
env_noNA = na.omit(trim.env2)

# spread data set so taxa are columns
phylum_spread = psclust_sum_phyla_0.1 %>%
  pivot_wider(names_from = Phylum, values_from = Percent) # note proportional not raw reads

# remove samples with NA in env data
remove = c("030322-L-W1D1", "030322-L-W1D3", "030322-L-W4D2", "030322-L-W4D4") 
phylum_spread_noNA = subset(phylum_spread, !(SampleID %in% remove))

# merge phlya with env data
phyl_env = merge(phylum_spread, trim.env2, by = "SampleID")
phyl_env_noNA = na.omit(phyl_env)

phyl_env_corr = cor(phyl_env_noNA[2:31], phyl_env_noNA[32:ncol(phyl_env_noNA)], method = "spearman")

library(psych)
phyl_env_corr_test = corr.test(phyl_env_noNA[2:31], phyl_env_noNA[32:ncol(phyl_env_noNA)], method = "spearman")
phyl_env_corr_padj = phyl_env_corr_test$p.adj
write.csv(phyl_env_corr_padj, "Phylum_Env_Spearman_pvalues_holm_2024.06.20.csv")

phyl_env_corr_test = corr.test(phyl_env_noNA[2:31], phyl_env_noNA[32:ncol(phyl_env_noNA)], method = "spearman", adjust = "bonferroni")
phyl_env_corr_padj = phyl_env_corr_test$p.adj
write.csv(phyl_env_corr_padj, "Phylum_Env_Spearman_pvalues_bonferroni_2024.06.20.csv")


phy <- pheatmap( 
  phyl_env_corr,
  color = col,
  breaks = seq(-1, 1, by = 0.2),
  clustering_method = "complete",  # Clustering method
  display_numbers = T,  # Display numbers in cells
  number_color = "grey35",
  show_rownames = T,
  legend = TRUE,  # Display legend
  legend_breaks = seq(-1, 1, by = 0.2)
)


save_pheatmap_pdf(phy, "Phyla_env_prop_2024.03.23.pdf", width = 8, height = 9)

phy <- pheatmap( 
  phyl_env_corr,
  color = col,
  breaks = seq(-1, 1, by = 0.2),
  clustering_method = "complete",  # Clustering method
  display_numbers = F,  # Display numbers in cells
  number_color = "grey35",
  show_rownames = T,
  legend = TRUE,  # Display legend
  legend_breaks = seq(-1, 1, by = 0.2)
)


save_pheatmap_pdf(phy, "Phyla_env_noNum_2024.03.23.pdf", width = 6, height = 6)


# using transformed raw reads
# psclust_reads_phyla_0.1 = data.frame(psclust_sum_phyla2_sub$SampleID, psclust_sum_phyla2_sub$All, psclust_sum_phyla2_sub$Phylum)
# colnames(psclust_reads_phyla_0.1) = c("SampleID", "Reads", "Phylum")
# psclust_reads_phyla_0.1$trans_reads = asinh(psclust_reads_phyla_0.1$Reads)
# 
# phylum_spread_trans = psclust_reads_phyla_0.1 %>%
#   pivot_wider(names_from = Phylum, values_from = trans_reads)
# 
# phylum_spread_trans_noNA = subset(phylum_spread_trans, !(SampleID %in% remove))


```


```{r rank and spearman (non-parametric)}

## This is actually the same just computed manually!!

phyla_rank = data.frame(lapply(phylum_spread[,-1], rank, ties.method = "max"))
phyla_rank$SampleID = phylum_spread$SampleID

# merge phlya with env data
phyl_env = merge(phyla_rank, trim.env2, by = "SampleID")

# compute correlation matrix
phyl_env_corr = cor(phyl_env[2:31], phyl_env[32:ncol(phyl_env)], method = "spearman", use = "complete.obs")

# remove NAs manually and compute matrix (same result, just to check)
phyl_env_noNA = na.omit(phyl_env)
phyl_env_corr = cor(phyl_env_noNA[2:31], phyl_env_noNA[32:ncol(phyl_env_noNA)], method = "spearman")

symnum(phyl_env_corr)

phyl_env_corr_test = cor.test(phyl_env_noNA$Nitrospirota, phyl_env_noNA$HIX, method = "spearman")


# visualize correlation matrix
pheatmap( 
  phyl_env_corr,
  color = col,
  breaks = seq(-1, 1, by = 0.2),
  clustering_method = "complete",  # Clustering method
  display_numbers = T,  # Display numbers in cells
  number_color = "grey35",
  show_rownames = T,
  legend = TRUE,  # Display legend
  legend_breaks = seq(-1, 1, by = 0.2)
)


#save_pheatmap_pdf(xx, "Chem_corr_untrans_rowNames_2024.03.23.pdf")




```



## 3. DESeq2 to identify phyla differences between groups
```{r new ps phyla}
# aggregate ps by phylum
ps_pw_sw = subset_samples(ps, Type != "Sand")

#agg_phylum = tax_glom(ps_pw_sw, taxrank = "Phylum", NArm = F) ######### takes a while



# recreate phyloseq object with phlya reads
psclust_sum_phyla_piv = psclust_sum_phyla %>%
  pivot_wider(names_from = Phylum, values_from = All) 

psclust_sum_phyla_piv2 = psclust_sum_phyla_piv[,-1]

row.names(psclust_sum_phyla_piv2) <- psclust_sum_phyla_piv$SampleID

colnames(psclust_sum_phyla_piv2) <- 1:ncol(psclust_sum_phyla_piv2)

phyla = as.data.frame(unique(psclust_sum_phyla$Phylum))
colnames(phyla) = "Phylum"

#row.names(phyla) = phyla$Phylum
phyla$OTU = 1:nrow(phyla)


meta = sample_data(ps_pw_sw)

otu = as.data.frame(otu_table(ps_pw_sw))

tax = data.frame(tax_table(ps_pw_sw))


# extract information from phyloseq object
otu = otu_table(as.matrix(psclust_sum_phyla_piv2), taxa_are_rows = F)
tax = tax_table(as.matrix(phyla))
#seq = refseq(ps_origin)

taxa_names(ps_pw_sw)

# construct new phyloseq object with new sample data
ps_new = phyloseq(otu, tax, meta)

# check new sample data was integrated
sam = sample_data(ps_new)

# save new phyloseq object
saveRDS(ps_new, "ps_Stinson_2024.01.16.rds")


#ps_pw_sw = subset_samples(Type != "Sand")




```



```{r proportion ~ season/campaign}
ps_phyla_prop <- readRDS("ps_agg_phylum_2024.01.16.rds")
ps_pw_sw = subset_samples(ps, Type != "Sand")
sample_data(ps_phyla_prop) = sample_data(ps_pw_sw) # replace old sample data


diagdds <- phyloseq_to_deseq2(ps_phyla_prop, ~ Campaign)
diagdds <- DESeq(diagdds, test = "Wald", fitType = "parametric", sfType = "poscounts")

incubation = "Campaign"
treat = "February 2022"
control = "October 2022"

# Inspect results with significance threshold
res <- results(diagdds, cooksCutoff = FALSE, contrast = c(incubation, treat, control))
alpha <- 0.05
sigtab <- res[which(res$padj < alpha), ]
#sigtab <- sigtab[which(res$log2FoldChange > 0), ] # only want enriched
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(ps_phyla_prop)[rownames(sigtab), ], "matrix"))
sigtab <- data.frame(baseMean = sigtab$baseMean, log2FoldChange = sigtab$log2FoldChange, lfcSE = sigtab$lfcSE, stat = sigtab$stat, pvalue = sigtab$pvalue, padj = sigtab$padj, Phylum = sigtab$Phylum)
write.csv(sigtab, "DESeq_Campaign_Feb_Phyla_2024.03.28.csv", row.names = F)

# October 2022

control = "February 2022"
treat = "October 2022"

# Inspect results with significance threshold
res <- results(diagdds, cooksCutoff = FALSE, contrast = c(incubation, treat, control))
alpha <- 0.05
sigtab <- res[which(res$padj < alpha), ]
#sigtab <- sigtab[which(res$log2FoldChange > 0), ] # only want enriched
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(ps_phyla_prop)[rownames(sigtab), ], "matrix"))
sigtab <- data.frame(baseMean = sigtab$baseMean, log2FoldChange = sigtab$log2FoldChange, lfcSE = sigtab$lfcSE, stat = sigtab$stat, pvalue = sigtab$pvalue, padj = sigtab$padj, Phylum = sigtab$Phylum)
write.csv(sigtab, "DESeq_Campaign_Oct_Phyla_2024.03.28.csv", row.names = F)
```


```{r proportion ~ location groups}
# Cluster 1
diagdds <- phyloseq_to_deseq2(ps_phyla_prop, ~ Cluster1)
diagdds <- DESeq(diagdds, test = "Wald", fitType = "parametric", sfType = "poscounts")

incubation = "Cluster1" # deep
treat = "yes"
control = "no"

# Inspect results with significance threshold
res <- results(diagdds, cooksCutoff = FALSE, contrast = c(incubation, treat, control))
alpha <- 0.05
sigtab <- res[which(res$padj < alpha), ]
#sigtab <- sigtab[which(res$log2FoldChange > 0), ] # only want enriched
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(ps_pw)[rownames(sigtab), ], "matrix"))
sigtab <- data.frame(baseMean = sigtab$baseMean, log2FoldChange = sigtab$log2FoldChange, lfcSE = sigtab$lfcSE, stat = sigtab$stat, pvalue = sigtab$pvalue, padj = sigtab$padj, Phylum = sigtab$Phylum)
write.csv(sigtab, "DESeq_Cluster1_phyla_deep_2024.03.31.csv", row.names = F)

# Cluster 2
diagdds <- phyloseq_to_deseq2(ps_phyla_prop, ~ Cluster2)
diagdds <- DESeq(diagdds, test = "Wald", fitType = "parametric", sfType = "poscounts")

incubation = "Cluster2" # seawater
treat = "yes"
control = "no"

# Inspect results with significance threshold
res <- results(diagdds, cooksCutoff = FALSE, contrast = c(incubation, treat, control))
alpha <- 0.05
sigtab <- res[which(res$padj < alpha), ]
#sigtab <- sigtab[which(res$log2FoldChange > 0), ] # only want enriched
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(ps_pw)[rownames(sigtab), ], "matrix"))
sigtab <- data.frame(baseMean = sigtab$baseMean, log2FoldChange = sigtab$log2FoldChange, lfcSE = sigtab$lfcSE, stat = sigtab$stat, pvalue = sigtab$pvalue, padj = sigtab$padj, Phylum = sigtab$Phylum)
write.csv(sigtab, "DESeq_Cluster2_phyla_sw_2024.03.31.csv", row.names = F)

# Seawater only
diagdds <- phyloseq_to_deseq2(ps_phyla_prop, ~ Type)
diagdds <- DESeq(diagdds, test = "Wald", fitType = "parametric", sfType = "poscounts")

incubation = "Type" 
treat = "Seawater"
control = "Porewater"

# Inspect results with significance threshold
res <- results(diagdds, cooksCutoff = FALSE, contrast = c(incubation, treat, control))
alpha <- 0.05
sigtab <- res[which(res$padj < alpha), ]
#sigtab <- sigtab[which(res$log2FoldChange > 0), ] # only want enriched
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(ps_pw_sw)[rownames(sigtab), ], "matrix"))
sigtab <- data.frame(baseMean = sigtab$baseMean, log2FoldChange = sigtab$log2FoldChange, lfcSE = sigtab$lfcSE, stat = sigtab$stat, pvalue = sigtab$pvalue, padj = sigtab$padj, Phylum = sigtab$Phylum)
write.csv(sigtab, "DESeq_SW_only_phyla_2024.03.31.csv", row.names = F)

# Cluster 3
diagdds <- phyloseq_to_deseq2(ps_phyla_prop, ~ Cluster3)
diagdds <- DESeq(diagdds, test = "Wald", fitType = "parametric", sfType = "poscounts")

incubation = "Cluster3" # well 4
treat = "yes"
control = "no"

# Inspect results with significance threshold
res <- results(diagdds, cooksCutoff = FALSE, contrast = c(incubation, treat, control))
alpha <- 0.05
sigtab <- res[which(res$padj < alpha), ]
#sigtab <- sigtab[which(res$log2FoldChange > 0), ] # only want enriched
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(ps_pw)[rownames(sigtab), ], "matrix"))
sigtab <- data.frame(baseMean = sigtab$baseMean, log2FoldChange = sigtab$log2FoldChange, lfcSE = sigtab$lfcSE, stat = sigtab$stat, pvalue = sigtab$pvalue, padj = sigtab$padj, Phylum = sigtab$Phylum)
write.csv(sigtab, "DESeq_Cluster3_phyla_W4_2024.03.31.csv", row.names = F)

# Cluster 4
diagdds <- phyloseq_to_deseq2(ps_phyla_prop, ~ Cluster4)
diagdds <- DESeq(diagdds, test = "Wald", fitType = "parametric", sfType = "poscounts")

incubation = "Cluster4" # intermediate
treat = "yes"
control = "no"

# Inspect results with significance threshold
res <- results(diagdds, cooksCutoff = FALSE, contrast = c(incubation, treat, control))
alpha <- 0.05
sigtab <- res[which(res$padj < alpha), ]
#sigtab <- sigtab[which(res$log2FoldChange > 0), ] # only want enriched
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(ps_pw)[rownames(sigtab), ], "matrix"))
sigtab <- data.frame(baseMean = sigtab$baseMean, log2FoldChange = sigtab$log2FoldChange, lfcSE = sigtab$lfcSE, stat = sigtab$stat, pvalue = sigtab$pvalue, padj = sigtab$padj, Phylum = sigtab$Phylum)
write.csv(sigtab, "DESeq_Cluster4_iphyla_ntermediate_2024.03.31.csv", row.names = F)

# Cluster 5
diagdds <- phyloseq_to_deseq2(ps_phyla_prop, ~ Cluster5)
diagdds <- DESeq(diagdds, test = "Wald", fitType = "parametric", sfType = "poscounts")

incubation = "Cluster5" # shallow
treat = "yes"
control = "no"

# Inspect results with significance threshold
res <- results(diagdds, cooksCutoff = FALSE, contrast = c(incubation, treat, control))
alpha <- 0.05
sigtab <- res[which(res$padj < alpha), ]
#sigtab <- sigtab[which(res$log2FoldChange > 0), ] # only want enriched
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(ps_pw)[rownames(sigtab), ], "matrix"))
sigtab <- data.frame(baseMean = sigtab$baseMean, log2FoldChange = sigtab$log2FoldChange, lfcSE = sigtab$lfcSE, stat = sigtab$stat, pvalue = sigtab$pvalue, padj = sigtab$padj, Phylum = sigtab$Phylum)
reorder(sigtab, sigtab$log2FoldChange)
write.csv(sigtab, "DESeq_Cluster5_phyla_shallow_2024.03.31.csv", row.names = F)
```


# Part 5. Genus-level analyses

## 1. Stacked Bar Genera
```{r}
# define Phylum_Genus
psclust_long$Phylum_Genus = paste(psclust_long$Phylum, psclust_long$Genus, sep = "; ")

# sum read counts by sample ID and phylum
psclust_sum_genera = psclust_long %>% group_by(SampleID, Phylum_Genus) %>% summarise(All = sum(Abundance)) # not proportional!

# total reads
sum(psclust_long$Abundance) # 7613584

# phyla comprising 0.1% of all reads (Ruiz Gonzales 2022)
all_genera_reads = psclust_long %>% group_by(Phylum_Genus) %>% summarise(Total = sum(Abundance)) 
all_genera_reads$Proportion = (all_genera_reads$Total)/(sum(psclust_long$Abundance))*100
all_genera_reads = subset(all_genera_reads, Total > 0)
n_distinct(all_genera_reads$Phylum_Genus) #1749 (includes unclassifieds)

# ggplot(all_genera_reads, aes(x = reorder(Genus, Proportion), y = Proportion)) +
#   geom_bar(stat = "identity") +
#   coord_flip() +
#   theme_bw() +
#   labs(x = "", y = "Percent of Total Reads (%)")
# 
# ggsave("phyla_rank_percent_2024.03.14.png", plot = last_plot(), 
#        units = "cm", width = 28, height = 28, dpi = 300)
# 
# ggplot(all_phyla_reads, aes(x = reorder(Phylum, Total), y = Total)) +
#   geom_bar(stat = "identity") +
#   coord_flip() +
#   theme_bw() +
#   labs(x = "", y = "\nTotal Read Count")
# 
# ggsave("phyla_rank_reads_2024.03.14.png", plot = last_plot(), 
#        units = "cm", width = 28, height = 28, dpi = 300)


sub_genera_reads = subset(all_genera_reads, Proportion >= 0.3 )  #0.5%

sub_genera_reads = sub_genera_reads[order(-sub_genera_reads$Proportion),]
sub_genera_reads$Phylum_Genus # 49/1749

# how much of the reads are they ?
sum(sub_genera_reads$Total)/sum(psclust_long$Abundance) # 0.633369

# store the order for plotting
genera_list_0.3 = sub_genera_reads$Phylum_Genus

# total reads per sample
psclust_sum_sample = psclust_long %>% group_by(SampleID) %>% summarise(Reads = sum(Abundance)) 
check_data = data.frame(sample_data(ps_pw_sw))
check_data2 = merge(psclust_sum_sample, check_data, by = "SampleID")

# merge sample separated with total per sample
psclust_sum_genera2 = left_join(psclust_sum_genera, psclust_sum_sample, by = "SampleID")

# take proportion of sample-phylum per sample total
psclust_sum_genera2$Proportion = (psclust_sum_genera2$All)/(psclust_sum_genera2$Reads)
psclust_sum_genera2$Percent = (psclust_sum_genera2$All)/(psclust_sum_genera2$Reads)*100

# subset to genera > 0.3% of total reads
psclust_sum_genera2_sub = subset(psclust_sum_genera2, Phylum_Genus %in% c(genera_list_0.3))

unique(psclust_sum_genera2_sub$Phylum_Genus) # check

# create an "other"  group
sub_genera_total = psclust_sum_genera2_sub %>% group_by(SampleID) %>% summarise(Percent = sum(Percent)) # highest is 83
psclust_sum_genera2_other = psclust_sum_genera2_sub %>% group_by(SampleID) %>% summarise(Percent = 83 - sum(Percent))
psclust_sum_genera2_other$Phylum_Genus = "Other"

# trim table 
psclust_sum_genera_0.3 = data.frame(psclust_sum_genera2_sub$SampleID, psclust_sum_genera2_sub$Percent, psclust_sum_genera2_sub$Phylum_Genus)
colnames(psclust_sum_genera_0.3) = c("SampleID", "Percent", "Phylum_Genus")

# bind dataframes
psclust_genera = rbind(psclust_sum_genera_0.3, psclust_sum_genera2_other)

# merge with metadata
genera_table = merge(check_data2, psclust_genera, by = "SampleID")

# specify order of samples by metadata file # well, depth order
# genera_table$SampleID = as.factor(genera_table$SampleID)
# meta = read.csv("Stinson_16S_metadata_2024.03.14.csv") #204 with sand
# meta = subset(meta, Type != "Sand")
# genera_table$SampleID = factor(genera_table$SampleID, levels = c(rev(meta$SampleID)))
# levels(genera_table$SampleID) # reverse order for plot

# specify order 
genera_list_0.3_other = c(genera_list_0.3,  "Other")
genera_table$Phylum_Genus = as.factor(genera_table$Phylum_Genus)
genera_table$Phylum_Genus = factor(genera_table$Phylum_Genus, levels = c((genera_list_0.3_other)))
levels(genera_table$Phylum_Genus)

# define colors
colors_matt_30 =c( "white", "#a7b8c2", "#f0ece2", "#dfd3c3", "#c7b198" ,"#9c4f41", "#bf2651", "#4d6a94", "#b3785d", "#e6cc8a", "#997d76", "#ffd3a3" ,"#52216e", "#ffac7f" ,"#ff8766" ,"#e6a3a3", "#d6a57a", "#4b3d53" ,"#911d55", "#fafac3", "#99c2db", "#995c95", "#93c2a7", "#66333d", "#524a63", "#4c8f82" ,"#b2c5c7" ,"#357985" ,"#728794",  "#dfeded","#242b4a")

colors_matt_60 =c( "#a7b8c2", "#f0ece2", "#dfd3c3", "#c7b198" ,"#9c4f41", "#bf2651","#f6f694", "#4d6a94", "#b3785d", "#e6cc8a", "#997d76", "#ffd3a3" ,"#52216e", "#ffac7f" ,"#ff8766" ,"#e6a3a3", "#d6a57a", "#4b3d53" ,"#911d55", "#fafac3", "#99c2db", "#995c95", "#93c2a7", "#66333d", "#524a63", "#4c8f82" ,"#b2c5c7" ,"#357985" ,"#728794",  "#dfeded","#242b4a",
                  "#5161a6", "#744d8b",  "#9779a8", "#b9a6c5", "#aab7be", "#d4dbde", "#e5795b","#ff9375","#ff9f84","#ffab93","#ffb7a3","#ffc3b2","#ffcfc1","#ffdbd1","#ffe7e0",  "#6fa59b", "#93bbb4", "#666666", "#7f7f7f", "#999999", "#b2b2b2", "#d4dbde", "#66333d", "#7a9baf", "#89aec5", "#99c2db", "#43364a", "#5d5064", "#6e6375")

colors_matt_50  = sample(colors_matt_60, 49)
colors_matt_50 = c(colors_matt_50, "white")
 

# ordered by well depth
ggplot(genera_table, aes(x = Percent, y = reorder(SampleID, ClusterOrder, decreasing = T))) +
  geom_bar(stat = "identity", aes(fill = Phylum_Genus))+
  theme_bw() +
  theme(strip.text = element_text(size = 11)) +
  #theme(legend.position = "none") +
  #scale_y_continuous(limits = c(0, 50)) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) + 
  scale_fill_manual(values = colors_matt_50, name = "Taxa") +
  labs(x = "Relative Abundance (%)\n", y = "")

ggsave("Genera_Stacked_0.3_2024.04.02.png", plot = last_plot(),
       units = "cm", width = 40, height = 55, dpi = 300)

# # flip order for legend
# phyla_table$Phylum = factor(phyla_table$Phylum, levels = c((phyla_list_0.1_other)))
# levels(phyla_table$Phylum)
# ggplot(phyla_table, aes(x = Percent, y = SampleID)) +
#   geom_bar(stat = "identity", aes(fill = Phylum)) +
#   theme_bw() +
#   theme(strip.text = element_text(size = 11)) +
#   #theme(legend.position = "none") +
#   #scale_y_continuous(limits = c(0, 50)) +
#   theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
#   #theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) + 
#   scale_fill_manual(values = rev(colors_match), name = "Phyla") +
#   labs(x = "Relative Abundance (%)\n", y = "")
# 
# ggsave("Phyla_Legend_0.15_2024.03.14.png", plot = last_plot(),
#        units = "cm", width = 40, height = 55, dpi = 300)
# 
# 
# # including date tide
# #check_data2_sort$depth_bin = as.character(check_data2_sort$depth_bin)
# check_data3_sort = meta
# check_data3_sort$SampleWDBDT = paste(check_data3_sort$W, " ", check_data3_sort$depth_bin, " ", check_data3_sort$Date_Tide)
# 
# #phyla_table_2$depth_bin = as.character(phyla_table_2$depth_bin)
# phyla_table_3 = phyla_table
# phyla_table_3$SampleWDBDT = paste(phyla_table_3$W, " ", phyla_table_3$depth_bin, " ", phyla_table_3$Date_Tide)
# 
# phyla_table_3$SampleWDBDT = factor(phyla_table_3$SampleWDBDT, levels = c(rev(check_data3_sort$SampleWDBDT)))
# levels(phyla_table_3$SampleWDBDT)
# 
# ggplot(phyla_table_3, aes(x = Percent, y = SampleWDBDT)) +
#   geom_bar(stat = "identity", aes(fill = Phylum)) +
#   theme_bw() +
#   theme(strip.text = element_text(size = 11)) +
#   #theme(legend.position = "none") +
#   #scale_y_continuous(limits = c(0, 50)) +
#   theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
#   #theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) + 
#   scale_fill_manual(values = colors_match, name = "Phyla") +
#   labs(x = "Relative Abundance (%)\n", y = "") +
#   theme(panel.border = element_blank())
# 
# ggsave("Phyla_Stacked_DateTide_2024.03.14.png", plot = last_plot(),
#        units = "cm", width = 40, height = 55, dpi = 300)


# only samples already sent for sequencing
mgs = c("240222-H-W1D2", "240222-H-W1D3", "240222-H-W2D2", "240222-H-W2D3", "240222-H-W3D2", "240222-H-W3D3", "240222-H-W3D4", "240222-H-W4D1", "240222-H-W4D3", "250222-L-SW")

ggplot(subset(genera_table, SampleID %in% mgs), aes(x = Percent, y = SampleID)) +
  geom_bar(stat = "identity", aes(fill = Phylum_Genus)) +
  theme_bw() +
  theme(strip.text = element_text(size = 11)) +
  theme(legend.position = "none") +
  #scale_y_continuous(limits = c(0, 50)) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) + 
  scale_fill_manual(values = colors_match, name = "Taxa") +
  labs(x = "Relative Abundance (%)\n", y = "") +
  theme(panel.border = element_blank())

ggsave("MGs_Genera_2024.03.15.png", plot = last_plot(),
       units = "cm", width = 20, height = 5, dpi = 300)



```

```{r unclassifieds}

psclust_long_unk_bacteria = subset(psclust_long, Phylum == "unclassified Bacteria")

# total ASVs?
psclust_asvs_bacteria = psclust_long_unk_bacteria %>% group_by(OTU) %>% summarise(All = sum(Abundance)) 
psclust_asvs_bacteria = subset(psclust_asvs_bacteria, All > 0) # 3570 total ASVs

# total reads
sum(psclust_long_unk_bacteria$Abundance) # 225149/7613584 = 0.02957201 = 2.96 %

psclust_long_unk_arch = subset(psclust_long, Phylum == "unclassified Archaea")

# total ASVs?
psclust_asvs_arch = psclust_long_unk_arch %>% group_by(OTU) %>% summarise(All = sum(Abundance)) 
psclust_asvs_arch = subset(psclust_asvs_arch, All > 0) # 119 total ASVs

# total reads
sum(psclust_long_unk_arch$Abundance) # 6884/7613584 = 0.0009041734 = 0.09 %

# group by cluster
psclust_cluster_all = psclust_long %>% group_by(Cluster) %>% summarise(Total = sum(Abundance)) 
psclust_cluster_unk_bacteria = psclust_long_unk_bacteria %>% group_by(Cluster) %>% summarise(Unk_Bact = sum(Abundance)) 
psclust_cluster_unk_arch = psclust_long_unk_arch %>% group_by(Cluster) %>% summarise(Unk_Arch = sum(Abundance)) 

unknown_cluster = data.frame(Cluster = psclust_cluster_all$Cluster, Total = psclust_cluster_all$Total, Unk_Bact = psclust_cluster_unk_bacteria$Unk_Bact, Unk_Arch = psclust_cluster_unk_arch$Unk_Arch, Unk_Bact_Percent = (psclust_cluster_unk_bacteria$Unk_Bact)/(psclust_cluster_all$Total)*100, Unk_Arch_Percent = (psclust_cluster_unk_arch$Unk_Arch)/(psclust_cluster_all$Total)*100)
```



```{r Proteobacteria}
# define Phylum_Genus
psclust_long_proteo = subset(psclust_long, Phylum == "Proteobacteria")
#psclust_long$Phylum_Genus = paste(psclust_long$Phylum, psclust_long$Genus, sep = "; ")

# sum read counts by sample ID and phylum
psclust_sum_genera_proteo = psclust_long_proteo %>% group_by(SampleID, Genus) %>% summarise(All = sum(Abundance)) # not proportional!

sample_all = psclust_long %>% group_by(SampleID) %>% summarise(All = sum(Abundance))
sample_proteo = psclust_long_proteo %>% group_by(SampleID) %>% summarise(All = sum(Abundance))
sample_proteo$Proportion = (sample_proteo$All)/(sample_all$All)*100

# total ASVs?
psclust_asvs_proteo = psclust_long_proteo %>% group_by(OTU) %>% summarise(All = sum(Abundance)) 
psclust_asvs_proteo = subset(psclust_asvs_proteo, All > 0)


# total reads
sum(psclust_long$Abundance) # 7613584

# phyla comprising 0.1% of all reads (Ruiz Gonzales 2022)
all_genera_reads_proteo = psclust_long_proteo %>% group_by(Genus) %>% summarise(Total = sum(Abundance)) 
all_genera_reads_proteo$Proportion_Total = (all_genera_reads_proteo$Total)/(sum(psclust_long$Abundance))*100
all_genera_reads_proteo$Proportion_Prot = (all_genera_reads_proteo$Total)/(sum(psclust_long_proteo$Abundance))*100

all_genera_reads_proteo = subset(all_genera_reads_proteo, Total > 0)
n_distinct(all_genera_reads_proteo$Genus) #609 (includes unclassifieds)


# for each cluster
# Cluster 5 shallow
psclust_long_proteo_shallow = subset(psclust_long, Phylum == "Proteobacteria" & Cluster == "cluster5")

# sum read counts by sample ID and phylum
#psclust_sum_genera_proteo_shallow = psclust_long_proteo_shallow %>% group_by(SampleID, Genus) %>% summarise(All = sum(Abundance)) # not proportional!

all_genera_reads_proteo_shallow = psclust_long_proteo_shallow %>% group_by(Genus) %>% summarise(Total = sum(Abundance)) 
all_genera_reads_proteo_shallow$Proportion_Total = (all_genera_reads_proteo_shallow$Total)/(sum(psclust_long$Abundance))*100
all_genera_reads_proteo_shallow$Proportion_Prot = (all_genera_reads_proteo_shallow$Total)/(sum(psclust_long_proteo_shallow$Abundance))*100

all_genera_reads_proteo_shallow = subset(all_genera_reads_proteo_shallow, Total > 0)
n_distinct(all_genera_reads_proteo_shallow$Genus) #571 (includes unclassifieds)

# Cluster 4 intermediate
psclust_long_proteo_inter = subset(psclust_long, Phylum == "Proteobacteria" & Cluster == "cluster4")

# sum read counts by sample ID and phylum
#psclust_sum_genera_proteo_inter = psclust_long_proteo_inter %>% group_by(SampleID, Genus) %>% summarise(All = sum(Abundance)) # not proportional!

all_genera_reads_proteo_inter = psclust_long_proteo_inter %>% group_by(Genus) %>% summarise(Total = sum(Abundance)) 
all_genera_reads_proteo_inter$Proportion_Total = (all_genera_reads_proteo_inter$Total)/(sum(psclust_long$Abundance))*100
all_genera_reads_proteo_inter$Proportion_Prot = (all_genera_reads_proteo_inter$Total)/(sum(psclust_long_proteo_inter$Abundance))*100

all_genera_reads_proteo_inter = subset(all_genera_reads_proteo_inter, Total > 0)
n_distinct(all_genera_reads_proteo_inter$Genus) #503 (includes unclassifieds)

# Cluster 1 deep
psclust_long_proteo_deep = subset(psclust_long, Phylum == "Proteobacteria" & Cluster == "cluster1")

# sum read counts by sample ID and phylum
#psclust_sum_genera_proteo_inter = psclust_long_proteo_inter %>% group_by(SampleID, Genus) %>% summarise(All = sum(Abundance)) # not proportional!

all_genera_reads_proteo_deep = psclust_long_proteo_deep %>% group_by(Genus) %>% summarise(Total = sum(Abundance)) 
all_genera_reads_proteo_deep$Proportion_Total = (all_genera_reads_proteo_deep$Total)/(sum(psclust_long$Abundance))*100
all_genera_reads_proteo_deep$Proportion_Prot = (all_genera_reads_proteo_deep$Total)/(sum(psclust_long_proteo_deep$Abundance))*100

all_genera_reads_proteo_deep = subset(all_genera_reads_proteo_deep, Total > 0)
n_distinct(all_genera_reads_proteo_deep$Genus) #401 (includes unclassifieds)

# Cluster 3 W4 without overtopping
psclust_long_proteo_c3 = subset(psclust_long, Phylum == "Proteobacteria" & Cluster == "cluster3")

# sum read counts by sample ID and phylum
#psclust_sum_genera_proteo_inter = psclust_long_proteo_inter %>% group_by(SampleID, Genus) %>% summarise(All = sum(Abundance)) # not proportional!

all_genera_reads_proteo_c3 = psclust_long_proteo_c3 %>% group_by(Genus) %>% summarise(Total = sum(Abundance)) 
all_genera_reads_proteo_c3$Proportion_Total = (all_genera_reads_proteo_c3$Total)/(sum(psclust_long$Abundance))*100
all_genera_reads_proteo_c3$Proportion_Prot = (all_genera_reads_proteo_c3$Total)/(sum(psclust_long_proteo_c3$Abundance))*100

all_genera_reads_proteo_c3 = subset(all_genera_reads_proteo_c3, Total > 0)
n_distinct(all_genera_reads_proteo_c3$Genus) #480 (includes unclassifieds)

# Well 4 all
psclust_long_proteo_W4 = subset(psclust_long, Phylum == "Proteobacteria" & Well == "Well 4")

# sum read counts by sample ID and phylum
#psclust_sum_genera_proteo_inter = psclust_long_proteo_inter %>% group_by(SampleID, Genus) %>% summarise(All = sum(Abundance)) # not proportional!

all_genera_reads_proteo_W4 = psclust_long_proteo_W4 %>% group_by(Genus) %>% summarise(Total = sum(Abundance)) 
all_genera_reads_proteo_W4$Proportion_Total = (all_genera_reads_proteo_W4$Total)/(sum(psclust_long$Abundance))*100
all_genera_reads_proteo_W4$Proportion_Prot = (all_genera_reads_proteo_W4$Total)/(sum(psclust_long_proteo_W4$Abundance))*100

all_genera_reads_proteo_W4 = subset(all_genera_reads_proteo_W4, Total > 0)
n_distinct(all_genera_reads_proteo_W4$Genus) #492 (includes unclassifieds)

# Cluster2 SW and overtopping
psclust_long_proteo_c2 = subset(psclust_long, Phylum == "Proteobacteria" & Cluster == "cluster2")

# sum read counts by sample ID and phylum
#psclust_sum_genera_proteo_inter = psclust_long_proteo_inter %>% group_by(SampleID, Genus) %>% summarise(All = sum(Abundance)) # not proportional!

all_genera_reads_proteo_c2 = psclust_long_proteo_c2 %>% group_by(Genus) %>% summarise(Total = sum(Abundance)) 
all_genera_reads_proteo_c2$Proportion_Total = (all_genera_reads_proteo_c2$Total)/(sum(psclust_long$Abundance))*100
all_genera_reads_proteo_c2$Proportion_Prot = (all_genera_reads_proteo_c2$Total)/(sum(psclust_long_proteo_c2$Abundance))*100

all_genera_reads_proteo_c2 = subset(all_genera_reads_proteo_c2, Total > 0)
n_distinct(all_genera_reads_proteo_c2$Genus) #433 (includes unclassifieds)

# Seawater only
psclust_long_proteo_SW = subset(psclust_long, Phylum == "Proteobacteria" & Well == "Seawater")

# sum read counts by sample ID and phylum
#psclust_sum_genera_proteo_inter = psclust_long_proteo_inter %>% group_by(SampleID, Genus) %>% summarise(All = sum(Abundance)) # not proportional!

all_genera_reads_proteo_SW = psclust_long_proteo_SW %>% group_by(Genus) %>% summarise(Total = sum(Abundance)) 
all_genera_reads_proteo_SW$Proportion_Total = (all_genera_reads_proteo_SW$Total)/(sum(psclust_long$Abundance))*100
all_genera_reads_proteo_SW$Proportion_Prot = (all_genera_reads_proteo_SW$Total)/(sum(psclust_long_proteo_SW$Abundance))*100

all_genera_reads_proteo_SW = subset(all_genera_reads_proteo_SW, Total > 0)
n_distinct(all_genera_reads_proteo_SW$Genus) #295 (includes unclassifieds)



# ggplot(all_genera_reads, aes(x = reorder(Genus, Proportion), y = Proportion)) +
#   geom_bar(stat = "identity") +
#   coord_flip() +
#   theme_bw() +
#   labs(x = "", y = "Percent of Total Reads (%)")
# 
# ggsave("phyla_rank_percent_2024.03.14.png", plot = last_plot(), 
#        units = "cm", width = 28, height = 28, dpi = 300)
# 
# ggplot(all_phyla_reads, aes(x = reorder(Phylum, Total), y = Total)) +
#   geom_bar(stat = "identity") +
#   coord_flip() +
#   theme_bw() +
#   labs(x = "", y = "\nTotal Read Count")
# 
# ggsave("phyla_rank_reads_2024.03.14.png", plot = last_plot(), 
#        units = "cm", width = 28, height = 28, dpi = 300)


sub_genera_reads = subset(all_genera_reads_proteo, Proportion_Prot >= 0.5 )  #0.5%

sub_genera_reads = sub_genera_reads[order(-sub_genera_reads$Proportion_Prot),]
sub_genera_reads$Genus # 49/1749

# how much of the reads are they ?
sum(sub_genera_reads$Total)/sum(psclust_long$Abundance) # 0.1979181 # 19.8% of total reads

# store the order for plotting
genera_list_0.3 = sub_genera_reads$Genus

# total reads per sample
psclust_sum_sample = psclust_long %>% group_by(SampleID) %>% summarise(Reads = sum(Abundance)) # reads out of proteo
check_data2 = data.frame(sample_data(ps_pw_sw))
#check_data2 = merge(psclust_sum_sample, check_data, by = "SampleID")

# merge sample separated with total per sample
psclust_sum_genera2 = left_join(psclust_sum_genera_proteo, psclust_sum_sample, by = "SampleID")

# take proportion  per sample total # proteo!!
psclust_sum_genera2$Proportion = (psclust_sum_genera2$All)/(psclust_sum_genera2$Reads)
psclust_sum_genera2$Percent = (psclust_sum_genera2$All)/(psclust_sum_genera2$Reads)*100

# subset to genera > 0.3% of total reads
psclust_sum_genera2_sub = subset(psclust_sum_genera2, Genus %in% c(genera_list_0.3))

unique(psclust_sum_genera2_sub$Genus) # check

# create an "other"  group
sub_genera_total = psclust_sum_genera2_sub %>% group_by(SampleID) %>% summarise(Percent = sum(Percent)) 
psclust_sum_genera2_other = psclust_sum_genera2_sub %>% group_by(SampleID) %>% summarise(Percent = 55.5 - sum(Percent)) # chagne to 100 for proteo
psclust_sum_genera2_other$Genus = "Other"

# trim table 
psclust_sum_genera_0.3 = data.frame(psclust_sum_genera2_sub$SampleID, psclust_sum_genera2_sub$Percent, psclust_sum_genera2_sub$Genus)
colnames(psclust_sum_genera_0.3) = c("SampleID", "Percent", "Genus")



# bind dataframes
psclust_genera = rbind(psclust_sum_genera_0.3, psclust_sum_genera2_other)

# merge with metadata
genera_table = merge(check_data2, psclust_genera, by = "SampleID")


# specify order of samples by metadata file # well, depth order
genera_table$SampleID = as.factor(genera_table$SampleID)
meta = read.csv("Stinson_16S_metadata_2024.03.14.csv") #204 with sand
meta = subset(meta, Type != "Sand")
genera_table$SampleID = factor(genera_table$SampleID, levels = c(rev(meta$SampleID)))
levels(genera_table$SampleID) # reverse order for plot

# specify order 
genera_list_0.3_other = c(genera_list_0.3,  "Other")
genera_table$Genus = as.factor(genera_table$Genus)
genera_table$Genus = factor(genera_table$Genus, levels = c(rev(genera_list_0.3_other))) # reverse
levels(genera_table$Genus)

# define colors
colors_matt_30 =c( "white", "#a7b8c2", "#f0ece2", "#dfd3c3", "#c7b198" ,"#9c4f41", "#bf2651", "#4d6a94", "#b3785d", "#e6cc8a", "#997d76", "#ffd3a3" ,"#52216e", "#ffac7f" ,"#ff8766" ,"#e6a3a3", "#d6a57a", "#4b3d53" ,"#911d55", "#fafac3", "#99c2db", "#995c95", "#93c2a7", "#66333d", "#524a63", "#4c8f82" ,"#b2c5c7" ,"#357985" ,"#728794",  "#dfeded","#242b4a")

colors_matt_60 =c( "#a7b8c2", "#f0ece2", "#dfd3c3", "#c7b198" ,"#9c4f41", "#bf2651","#f6f694", "#4d6a94", "#b3785d", "#e6cc8a", "#997d76", "#ffd3a3" ,"#52216e", "#ffac7f" ,"#ff8766" ,"#e6a3a3", "#d6a57a", "#4b3d53" ,"#911d55", "#fafac3", "#99c2db", "#995c95", "#93c2a7", "#66333d", "#524a63", "#4c8f82" ,"#b2c5c7" ,"#357985" ,"#728794",  "#dfeded","#242b4a",
                  "#5161a6", "#744d8b",  "#9779a8", "#b9a6c5", "#aab7be", "#d4dbde", "#e5795b","#ff9375","#ff9f84","#ffab93","#ffb7a3","#ffc3b2","#ffcfc1","#ffdbd1","#ffe7e0",  "#6fa59b", "#93bbb4", "#666666", "#7f7f7f", "#999999", "#b2b2b2", "#d4dbde", "#66333d", "#7a9baf", "#89aec5", "#99c2db", "#43364a", "#5d5064", "#6e6375")

colors_matt_60 =c( "#a7b8c2", "#f0ece2", "#dfd3c3", "#c7b198" , "#bf2651","#f6f694", "#4d6a94",  "#e6cc8a", "#997d76", "#ffd3a3" ,"#52216e", "#ffac7f" ,"#ff8766" ,"#e6a3a3", "#d6a57a", "#4b3d53" ,"#911d55", "#fafac3", "#99c2db", "#995c95", "#93c2a7", "#66333d", "#524a63", "#4c8f82" ,"#b2c5c7" ,"#357985" ,"#728794",
                   
                   
                  "#5161a6", "#744d8b",  "#9779a8", "#b9a6c5", "#aab7be", "#d4dbde", "#6fa59b", "#93bbb4", "#666666", "#7f7f7f","#66333d", "#ffcfc1","#ffdbd1","#ffe7e0")
                  
                  
                  "#e5795b","#ff9375","#ff9f84","#ffab93","#ffb7a3","#ffc3b2",  "#999999", "#b2b2b2", "#d4dbde", "#7a9baf", "#89aec5", "#6e6375")

colors_matt_50  = sample(colors_matt_60, 49)
colors_matt_50 = c(colors_matt_50, "white")


# 43 colors

 
colors_matt_41  = sample(colors_matt_60, 41)
colors_matt_43 = c("#242b4a", colors_matt_41, "white")

version1_colors = "#242b4a" "#ffcfc1" "#744d8b" "#ffac7f" "#fafac3" "#b2c5c7" "#ffd3a3" "#911d55" "#aab7be" "#357985" "#e6cc8a" "#66333d"
[13] "#ffe7e0" "#c7b198" "#d6a57a" "#4d6a94" "#a7b8c2" "#995c95" "#dfd3c3" "#5161a6" "#ff8766" "#6fa59b" "#66333d" "#bf2651"
[25] "#d4dbde" "#93c2a7" "#997d76" "#7f7f7f" "#e6a3a3" "#728794" "#ffdbd1" "#9779a8" "#52216e" "#f6f694" "#666666" "#99c2db"
[37] "#b9a6c5" "#93bbb4" "#f0ece2" "#4b3d53" "#4c8f82" "#524a63" "white"  


# cluster order
ggplot(genera_table, aes(x = Percent, y = reorder(SampleID, ClusterOrder, decreasing = T))) +
  geom_bar(stat = "identity", aes(fill = Genus))+
  theme_bw() +
  theme(strip.text = element_text(size = 11)) +
  #theme(legend.position = "none") +
  #scale_y_continuous(limits = c(0, 50)) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) + 
  scale_fill_manual(values = c(rev(colors_matt_43)), name = "Taxa") +
  labs(x = "Relative Abundance (%)\n", y = "")

ggsave("Proteo_Genera_0.3_2024.04.16.png", plot = last_plot(),
       units = "cm", width = 40, height = 55, dpi = 300)

# well depth date tide order
ggplot(genera_table, aes(x = Percent, y = SampleID)) +
  geom_bar(stat = "identity", aes(fill = Genus))+
  theme_bw() +
  theme(strip.text = element_text(size = 11)) +
  #theme(legend.position = "none") +
  #scale_y_continuous(limits = c(0, 50)) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) + 
  scale_fill_manual(values = c(rev(colors_matt_43)), name = "Taxa") +
  labs(x = "Relative Abundance (%)\n", y = "")

ggsave("Proteo_Genera_WellDepth_0.3_2024.04.16.png", plot = last_plot(),
       units = "cm", width = 40, height = 55, dpi = 300)

# # flip order for legend
# phyla_table$Phylum = factor(phyla_table$Phylum, levels = c((phyla_list_0.1_other)))
# levels(phyla_table$Phylum)
# ggplot(phyla_table, aes(x = Percent, y = SampleID)) +
#   geom_bar(stat = "identity", aes(fill = Phylum)) +
#   theme_bw() +
#   theme(strip.text = element_text(size = 11)) +
#   #theme(legend.position = "none") +
#   #scale_y_continuous(limits = c(0, 50)) +
#   theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
#   #theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) + 
#   scale_fill_manual(values = rev(colors_match), name = "Phyla") +
#   labs(x = "Relative Abundance (%)\n", y = "")
# 
# ggsave("Phyla_Legend_0.15_2024.03.14.png", plot = last_plot(),
#        units = "cm", width = 40, height = 55, dpi = 300)
# 
# 
# including date tide
#check_data2_sort$depth_bin = as.character(check_data2_sort$depth_bin)
check_data3_sort = meta
check_data3_sort$SampleWDBDT = paste(check_data3_sort$W, " ", check_data3_sort$depth_bin, " ", check_data3_sort$Date_Tide)

#phyla_table_2$depth_bin = as.character(phyla_table_2$depth_bin)
phyla_table_3 = genera_table
phyla_table_3$SampleWDBDT = paste(phyla_table_3$W, " ", phyla_table_3$depth_bin, " ", phyla_table_3$Date_Tide)

phyla_table_3$SampleWDBDT = factor(phyla_table_3$SampleWDBDT, levels = c(rev(check_data3_sort$SampleWDBDT)))
levels(phyla_table_3$SampleWDBDT)


ggplot(phyla_table_3, aes(x = Percent, y = SampleWDBDT)) +
  geom_bar(stat = "identity", aes(fill = Genus)) +
  theme_bw() +
  theme(strip.text = element_text(size = 11)) +
  #theme(legend.position = "none") +
  #scale_y_continuous(limits = c(0, 50)) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) +
  scale_fill_manual(values = rev(colors_matt_43), name = "Taxa") +
  labs(x = "Relative Abundance (%)\n", y = "") +
  theme(panel.border = element_blank())

ggsave("Proteo_Genera_Stacked_DateTide_2024.04.16.png", plot = last_plot(),
       units = "cm", width = 40, height = 55, dpi = 300)

# legend
phyla_table_3$Genus = factor(phyla_table_3$Genus, levels = c((genera_list_0.3_other))) # reverse
levels(phyla_table_3$Genus)

ggplot(phyla_table_3, aes(x = Percent, y = SampleWDBDT)) +
  geom_bar(stat = "identity", aes(fill = Genus)) +
  theme_bw() +
  theme(strip.text = element_text(size = 11)) +
  #theme(legend.position = "none") +
  #scale_y_continuous(limits = c(0, 50)) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) +
  scale_fill_manual(values = (colors_matt_43), name = "Proteobacteria Taxa") +
  labs(x = "Relative Abundance (%)\n", y = "") +
  theme(panel.border = element_blank())

ggsave("Proteo_Genera_legend_2024.04.16.png", plot = last_plot(),
       units = "cm", width = 40, height = 55, dpi = 300)


# only samples already sent for sequencing
mgs = c("240222-H-W1D2", "240222-H-W1D3", "240222-H-W2D2", "240222-H-W2D3", "240222-H-W3D2", "240222-H-W3D3", "240222-H-W3D4", "240222-H-W4D1", "240222-H-W4D3", "250222-L-SW")

ggplot(subset(genera_table, SampleID %in% mgs), aes(x = Percent, y = SampleID)) +
  geom_bar(stat = "identity", aes(fill = Phylum_Genus)) +
  theme_bw() +
  theme(strip.text = element_text(size = 11)) +
  theme(legend.position = "none") +
  #scale_y_continuous(limits = c(0, 50)) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) + 
  scale_fill_manual(values = colors_match, name = "Taxa") +
  labs(x = "Relative Abundance (%)\n", y = "") +
  theme(panel.border = element_blank())

ggsave("MGs_Genera_2024.03.15.png", plot = last_plot(),
       units = "cm", width = 20, height = 5, dpi = 300)



```

```{r non-Proteobacteria}
# define Phylum_Genus
psclust_long_proteo = subset(psclust_long, Phylum != "Proteobacteria")
psclust_long_proteo$Phylum_Genus = paste(psclust_long_proteo$Phylum, psclust_long_proteo$Genus, sep = "; ")

# sum read counts by sample ID and phylum
psclust_sum_genera_proteo = psclust_long_proteo %>% group_by(SampleID, Phylum_Genus) %>% summarise(All = sum(Abundance)) # not proportional!

sample_all = psclust_long %>% group_by(SampleID) %>% summarise(All = sum(Abundance))
sample_proteo = psclust_long_proteo %>% group_by(SampleID) %>% summarise(All = sum(Abundance))
sample_proteo$Proportion = (sample_proteo$All)/(sample_all$All)*100

# total ASVs?
psclust_asvs_proteo = psclust_long_proteo %>% group_by(OTU) %>% summarise(All = sum(Abundance)) 
psclust_asvs_proteo = subset(psclust_asvs_proteo, All > 0)


# total reads
#sum(psclust_long$Abundance) # 7613584

# phyla comprising 0.1% of all reads (Ruiz Gonzales 2022)
all_genera_reads_proteo = psclust_long_proteo %>% group_by(Phylum_Genus) %>% summarise(Total = sum(Abundance)) 
all_genera_reads_proteo$Proportion = (all_genera_reads_proteo$Total)/(sum(psclust_long$Abundance))*100
#all_genera_reads_proteo$Proportion_Prot = (all_genera_reads_proteo$Total)/(sum(psclust_long_proteo$Abundance))*100

all_genera_reads_proteo = subset(all_genera_reads_proteo, Total > 0)
n_distinct(all_genera_reads_proteo$Phylum_Genus) #1140 (includes unclassifieds)

sub_genera_reads = subset(all_genera_reads_proteo, Proportion >= 0.24 ) 
sub_genera_reads = subset(sub_genera_reads, Phylum_Genus != "unclassified Bacteria; unclassified Bacteria")

sub_genera_reads = sub_genera_reads[order(-sub_genera_reads$Proportion),]
sub_genera_reads$Phylum_Genus # 41

# how much of the reads are they ?
sum(sub_genera_reads$Total)/sum(psclust_long$Abundance) # 0.507979 # 50.8% of total reads

# store the order for plotting
genera_list_0.3 = sub_genera_reads$Phylum_Genus

# total reads per sample
psclust_sum_sample = psclust_long %>% group_by(SampleID) %>% summarise(Reads = sum(Abundance)) # change to psclust_long_proteo for reads out of proteo
check_data2 = data.frame(sample_data(ps_pw_sw))
#check_data2 = merge(psclust_sum_sample, check_data, by = "SampleID")

# merge sample separated with total per sample
psclust_sum_genera2 = left_join(psclust_sum_genera_proteo, psclust_sum_sample, by = "SampleID")

# take proportion  per sample total # proteo!!
psclust_sum_genera2$Proportion = (psclust_sum_genera2$All)/(psclust_sum_genera2$Reads)
psclust_sum_genera2$Percent = (psclust_sum_genera2$All)/(psclust_sum_genera2$Reads)*100

# subset to genera > 0.3% of total reads
psclust_sum_genera2_sub = subset(psclust_sum_genera2, Phylum_Genus %in% c(genera_list_0.3))

unique(psclust_sum_genera2_sub$Phylum_Genus) # check

# create an "other"  group
sub_genera_total = psclust_sum_genera2_sub %>% group_by(SampleID) %>% summarise(Percent = sum(Percent)) # check manually for top
psclust_sum_genera2_other = psclust_sum_genera2_sub %>% group_by(SampleID) %>% summarise(Percent = 80 - sum(Percent)) # chagne to 100 for proteo
psclust_sum_genera2_other$Phylum_Genus = "Other"

# trim table 
psclust_sum_genera_0.3 = data.frame(psclust_sum_genera2_sub$SampleID, psclust_sum_genera2_sub$Percent, psclust_sum_genera2_sub$Phylum_Genus)
colnames(psclust_sum_genera_0.3) = c("SampleID", "Percent", "Phylum_Genus")



# bind dataframes
psclust_genera = rbind(psclust_sum_genera_0.3, psclust_sum_genera2_other)
#psclust_genera = psclust_sum_genera_0.3 # remove other

# merge with metadata
genera_table = merge(check_data2, psclust_genera, by = "SampleID")


# specify order of samples by metadata file # well, depth order
genera_table$SampleID = as.factor(genera_table$SampleID)
meta = read.csv("Stinson_16S_metadata_2024.03.14.csv") #204 with sand
meta = subset(meta, Type != "Sand")
genera_table$SampleID = factor(genera_table$SampleID, levels = c(rev(meta$SampleID)))
levels(genera_table$SampleID) # reverse order for plot

# specify order 
genera_list_0.3_other = c(genera_list_0.3,  "Other")
genera_table$Phylum_Genus = as.factor(genera_table$Phylum_Genus)
genera_table$Phylum_Genus = factor(genera_table$Phylum_Genus, levels = c(rev(genera_list_0.3_other))) # reverse
levels(genera_table$Phylum_Genus)

# define colors
colors_matt_30 =c( "white", "#a7b8c2", "#f0ece2", "#dfd3c3", "#c7b198" ,"#9c4f41", "#bf2651", "#4d6a94", "#b3785d", "#e6cc8a", "#997d76", "#ffd3a3" ,"#52216e", "#ffac7f" ,"#ff8766" ,"#e6a3a3", "#d6a57a", "#4b3d53" ,"#911d55", "#fafac3", "#99c2db", "#995c95", "#93c2a7", "#66333d", "#524a63", "#4c8f82" ,"#b2c5c7" ,"#357985" ,"#728794",  "#dfeded","#242b4a")

colors_matt_60 =c( "#a7b8c2", "#f0ece2", "#dfd3c3", "#c7b198" ,"#9c4f41", "#bf2651","#f6f694", "#4d6a94", "#b3785d", "#e6cc8a", "#997d76", "#ffd3a3" ,"#52216e", "#ffac7f" ,"#ff8766" ,"#e6a3a3", "#d6a57a", "#4b3d53" ,"#911d55", "#fafac3", "#99c2db", "#995c95", "#93c2a7", "#66333d", "#524a63", "#4c8f82" ,"#b2c5c7" ,"#357985" ,"#728794",  "#dfeded","#242b4a",
                  "#5161a6", "#744d8b",  "#9779a8", "#b9a6c5", "#aab7be", "#d4dbde", "#e5795b","#ff9375","#ff9f84","#ffab93","#ffb7a3","#ffc3b2","#ffcfc1","#ffdbd1","#ffe7e0",  "#6fa59b", "#93bbb4", "#666666", "#7f7f7f", "#999999", "#b2b2b2", "#d4dbde", "#66333d", "#7a9baf", "#89aec5", "#99c2db", "#43364a", "#5d5064", "#6e6375")

colors_matt_60 =c( "#a7b8c2", "#f0ece2", "#dfd3c3", "#c7b198" , "#bf2651","#f6f694", "#4d6a94",  "#e6cc8a", "#997d76", "#ffd3a3" ,"#52216e", "#ffac7f" ,"#ff8766" ,"#e6a3a3", "#d6a57a", "#4b3d53" ,"#911d55", "#fafac3", "#99c2db", "#995c95", "#93c2a7", "#66333d", "#524a63", "#4c8f82" ,"#b2c5c7" ,"#357985" ,"#728794",
                   
                   
                  "#5161a6", "#744d8b",  "#9779a8", "#b9a6c5", "#d4dbde", "#6fa59b", "#93bbb4", "#666666", "#7f7f7f","#66333d", "#ffcfc1","#ffdbd1","#ffe7e0", "#7a9baf")
                  
                  
                  "#e5795b","#ff9375","#ff9f84","#ffab93","#ffb7a3","#ffc3b2",  "#999999", "#b2b2b2", "#d4dbde", "#7a9baf", "#89aec5", "#6e6375")

colors_matt_50  = sample(colors_matt_60, 49)
colors_matt_50 = c(colors_matt_50, "white")


# 43 colors

 
#colors_matt_41  = sample(colors_matt_60, 41)
#colors_matt_43 = c("#242b4a", colors_matt_41, "white")

colors_matt_41_2  = sample(colors_matt_60, 41)
colors_matt_43_2 = c("#242b4a", colors_matt_41_2, "white")
colors_matt_43_2 = c(colors_matt_41_2,"#242b4a")


version1_colors = "#242b4a" "#ffcfc1" "#744d8b" "#ffac7f" "#fafac3" "#b2c5c7" "#ffd3a3" "#911d55" "#aab7be" "#357985" "#e6cc8a" "#66333d"
[13] "#ffe7e0" "#c7b198" "#d6a57a" "#4d6a94" "#a7b8c2" "#995c95" "#dfd3c3" "#5161a6" "#ff8766" "#6fa59b" "#66333d" "#bf2651"
[25] "#d4dbde" "#93c2a7" "#997d76" "#7f7f7f" "#e6a3a3" "#728794" "#ffdbd1" "#9779a8" "#52216e" "#f6f694" "#666666" "#99c2db"
[37] "#b9a6c5" "#93bbb4" "#f0ece2" "#4b3d53" "#4c8f82" "#524a63" "white"  

version2_colors = c("#242b4a", "#ffcfc1", "#744d8b", "#ffac7f", "#fafac3", "#6e6375", "#ffd3a3", "#911d55", "#7a9baf", "#357985", "#e6cc8a", "#66333d", "#ffe7e0", "#c7b198", "#d6a57a", "#4d6a94", "#a7b8c2", "#995c95", "#dfd3c3", "#5161a6", "#ff8766", "#6fa59b","#b3785d", "#bf2651","#d4dbde", "#93c2a7", "#997d76", "#7f7f7f", "#e6a3a3", "#728794", "#ffdbd1", "#9779a8", "#52216e", "#f6f694", "#666666", "#99c2db","#b9a6c5" ,"#93bbb4" ,"#f0ece2", "#4b3d53", "#4c8f82", "white"  )

version3_colors =c( "#a7b8c2", "#f0ece2", "#dfd3c3", "#c7b198" , "#bf2651","#f6f694", "#4d6a94",  "#e6cc8a", "#997d76", "#ffd3a3" ,"#52216e", "#ffac7f" ,"#ff8766" ,"#e6a3a3", "#d6a57a", "#4b3d53" ,"#911d55", "#fafac3", "#99c2db", "#995c95", "#93c2a7", "#66333d", "#524a63", "#4c8f82" ,"#b2c5c7" ,"#357985" ,"#728794",
                   
                   
                  "#5161a6", "#744d8b",  "#9779a8", "#b9a6c5", "#d4dbde", "#6fa59b", "#93bbb4", "#666666", "#7f7f7f","#66333d", "#ffcfc1","#ffdbd1","#ffe7e0", "#7a9baf")

colors_matt_shade_42 =c("white",  "#5161a6", "#744d8b",  "#b9a6c5", "#d4dbde", "#6fa59b", "#93bbb4", "#666666",  "#ffcfc1","#ffe7e0", "#7a9baf",
  "#242526", "#a7b8c2", "#f0ece2", "#dfd3c3", "#c7b198" ,"#9c4f41", "#bf2651", "#4d6a94", "#b3785d", "#e6cc8a", "#fafac3",  "#f6f694" ,"#52216e", "#ffac7f" ,"#ff8766" ,"#e6a3a3", "#d6a57a", "#4b3d53" ,"#911d55","#6e6375", "#99c2db", "#995c95", "#93c2a7", "#66333d", "#524a63", "#4c8f82" ,"#b2c5c7" ,"#357985" ,"#728794",  "#dfeded","#242b4a")


# manually specify order
nonprot_order = c("Nanoarchaeota; unclassified Woesearchaeales",    
"Nanoarchaeota; unclassified GW2011_GWC1_47_15",                          
"Nanoarchaeota; unclassified SCGC AAA011-D5",
"Verrucomicrobiota; Candidatus Omnitrophus"   ,                           
"Verrucomicrobiota; unclassified Omnitrophales" ,
 "Patescibacteria; unclassified Candidatus Peribacteria"    ,              
 "Patescibacteria; unclassified Candidatus Yanofskybacteria"   ,           
 "Patescibacteria; unclassified Gracilibacteria"     ,                     
"Patescibacteria; unclassified JGI 0000069-P22"     ,                     
 "Patescibacteria; unclassified Parcubacteria"  ,
"Bdellovibrionota; Bdellovibrio"     ,                                    
"Bdellovibrionota; Peredibacter"        ,                                 
"Bdellovibrionota; unclassified 0319-6G20"      ,                         
"Bdellovibrionota; unclassified Bacteriovoracaceae"    ,                  
"Bdellovibrionota; unclassified Oligoflexaceae"      ,
"Crenarchaeota; Candidatus Nitrosopelagicus"   ,                          
"Crenarchaeota; Candidatus Nitrosopumilus"     ,                          
 "Crenarchaeota; unclassified Nitrosopumilaceae"     ,
"Firmicutes; unclassified Bacillaceae"   ,                                
"Campylobacterota; Sulfurimonas"   ,                                      
"Campylobacterota; unclassified Campylobacterales"   ,
"Bacteroidota; Algoriphagus"     ,                                        
"Bacteroidota; Flavobacterium"     ,                                      
 "Bacteroidota; Fluviicola"           ,                                    
"Bacteroidota; unclassified Cryomorphaceae"      ,                        
"Bacteroidota; unclassified Flavobacteriaceae"      ,                    
 "Bacteroidota; unclassified Sphingobacteriales"   ,
"Myxococcota; unclassified bacteriap25"   ,   
 "Chloroflexi; unclassified Dehalococcoidia" ,                             
"Chloroflexi; unclassified SAR202 clade"     ,
"Planctomycetota; Candidatus Scalindua"    ,                              
 "Planctomycetota; unclassified Planctomycetota"     ,
"SAR324 clade(Marine group B); unclassified SAR324 clade(Marine group B)",
"Desulfobacterota; Desulfobulbus"  ,
 "Thermoplasmatota; unclassified Marine Group II"   ,
 "Acidobacteriota; unclassified Vicinamibacteraceae"     ,                 
"Acidobacteriota; unclassified Vicinamibacterales"    ,                   
"Aenigmarchaeota; unclassified Aenigmarchaeales"   ,                      
"Dependentiae; unclassified Babeliales"        ,                          
"Dependentiae; unclassified Vermiphilaceae"   ,                           
"Nitrospirota; Nitrospira"  ,                                             
 "Other" )
genera_table$Phylum_Genus = factor(genera_table$Phylum_Genus, levels = c(rev(nonprot_order))) # reverse

# cluster order
ggplot(genera_table, aes(x = Percent, y = reorder(SampleID, ClusterOrder, decreasing = T))) +
  geom_bar(stat = "identity", aes(fill = Phylum_Genus))+
  theme_bw() +
  theme(strip.text = element_text(size = 11)) +
  #theme(legend.position = "none") +
  #scale_y_continuous(limits = c(0, 50)) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) + 
  scale_fill_manual(values = c((colors_matt_shade_42)), name = "Taxa") +
  labs(x = "Relative Abundance (%)\n", y = "")

ggsave("non_Proteo_cluster_0.24_2024.04.16_v3.png", plot = last_plot(),
       units = "cm", width = 40, height = 55, dpi = 300)

# well depth date tide order
ggplot(genera_table, aes(x = Percent, y = SampleID)) +
  geom_bar(stat = "identity", aes(fill = Phylum_Genus))+
  theme_bw() +
  theme(strip.text = element_text(size = 11)) +
  #theme(legend.position = "none") +
  #scale_y_continuous(limits = c(0, 50)) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) + 
  scale_fill_manual(values = c((colors_matt_shade_42)), name = "Taxa") +
  labs(x = "Relative Abundance (%)\n", y = "")

ggsave("non_Proteo_Genera_WellDepth_0.25_2024.04.16.png", plot = last_plot(),
       units = "cm", width = 40, height = 55, dpi = 300)


# including date tide
#check_data2_sort$depth_bin = as.character(check_data2_sort$depth_bin)
check_data3_sort = meta
check_data3_sort$SampleWDBDT = paste(check_data3_sort$W, " ", check_data3_sort$depth_bin, " ", check_data3_sort$Date_Tide)

#phyla_table_2$depth_bin = as.character(phyla_table_2$depth_bin)
phyla_table_3 = genera_table
phyla_table_3$SampleWDBDT = paste(phyla_table_3$W, " ", phyla_table_3$depth_bin, " ", phyla_table_3$Date_Tide)

phyla_table_3$SampleWDBDT = factor(phyla_table_3$SampleWDBDT, levels = c(rev(check_data3_sort$SampleWDBDT)))
levels(phyla_table_3$SampleWDBDT)


ggplot(phyla_table_3, aes(x = Percent, y = SampleWDBDT)) +
  geom_bar(stat = "identity", aes(fill = Phylum_Genus)) +
  theme_bw() +
  theme(strip.text = element_text(size = 11)) +
  theme(legend.position = "none") +
  #scale_y_continuous(limits = c(0, 50)) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) +
  scale_fill_manual(values = (colors_matt_shade_42), name = "Taxa") +
  labs(x = "Relative Abundance (%)\n", y = "") +
  theme(panel.border = element_blank())

ggsave("non_Proteo_Genera_DateTide_2024.04.16.png", plot = last_plot(),
       units = "cm", width = 20, height = 55, dpi = 300)

# legend
phyla_table_3$Phylum_Genus = factor(phyla_table_3$Phylum_Genus, levels = c((nonprot_order))) # reverse
levels(phyla_table_3$Phylum_Genus)

ggplot(phyla_table_3, aes(x = Percent, y = SampleWDBDT)) +
  geom_bar(stat = "identity", aes(fill = Phylum_Genus)) +
  theme_bw() +
  theme(strip.text = element_text(size = 11)) +
  #theme(legend.position = "none") +
  #scale_y_continuous(limits = c(0, 50)) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) +
  scale_fill_manual(values = c(rev(colors_matt_shade_42)), name = "Non-Proteobacteria Taxa") +
  labs(x = "Relative Abundance (%)\n", y = "") +
  theme(panel.border = element_blank())

ggsave("non_Proteo_Genera_legend_2024.04.16.png", plot = last_plot(),
       units = "cm", width = 40, height = 55, dpi = 300)


# # only samples already sent for sequencing
# mgs = c("240222-H-W1D2", "240222-H-W1D3", "240222-H-W2D2", "240222-H-W2D3", "240222-H-W3D2", "240222-H-W3D3", "240222-H-W3D4", "240222-H-W4D1", "240222-H-W4D3", "250222-L-SW")
# 
# ggplot(subset(genera_table, SampleID %in% mgs), aes(x = Percent, y = SampleID)) +
#   geom_bar(stat = "identity", aes(fill = Phylum_Genus)) +
#   theme_bw() +
#   theme(strip.text = element_text(size = 11)) +
#   theme(legend.position = "none") +
#   #scale_y_continuous(limits = c(0, 50)) +
#   theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
#   #theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) + 
#   scale_fill_manual(values = colors_match, name = "Taxa") +
#   labs(x = "Relative Abundance (%)\n", y = "") +
#   theme(panel.border = element_blank())
# 
# ggsave("MGs_Genera_2024.03.15.png", plot = last_plot(),
#        units = "cm", width = 20, height = 5, dpi = 300)



```



```{r Nanoarchaeota}
# define Phylum_Genus
psclust_long_nano = subset(psclust_long, Phylum == "Nanoarchaeota")
#psclust_long$Phylum_Genus = paste(psclust_long$Phylum, psclust_long$Genus, sep = "; ")

# sum read counts by sample ID and phylum
psclust_sum_genera_nano = psclust_long_nano %>% group_by(SampleID, Genus) %>% summarise(All = sum(Abundance)) # not proportional!

# total reads
sum(psclust_long$Abundance) # 7613584

psclust_asvs_nano = psclust_long_nano %>% group_by(OTU) %>% summarise(All = sum(Abundance)) 
psclust_asvs_nano = subset(psclust_asvs_nano, All > 0)


# phyla comprising 0.1% of all reads (Ruiz Gonzales 2022)
all_genera_reads_nano = psclust_long_nano %>% group_by(Genus) %>% summarise(Total = sum(Abundance)) 
all_genera_reads_nano$Proportion_Total = (all_genera_reads_nano$Total)/(sum(psclust_long$Abundance))*100
all_genera_reads_nano$Proportion_Prot = (all_genera_reads_nano$Total)/(sum(psclust_long_nano$Abundance))*100

all_genera_reads_nano = subset(all_genera_reads_nano, Total > 0)
n_distinct(all_genera_reads_nano$Genus) #7 (includes unclassifieds)


# for each cluster
# Cluster 5 shallow
psclust_long_proteo_shallow = subset(psclust_long, Phylum == "Proteobacteria" & Cluster == "cluster5")

# sum read counts by sample ID and phylum
#psclust_sum_genera_proteo_shallow = psclust_long_proteo_shallow %>% group_by(SampleID, Genus) %>% summarise(All = sum(Abundance)) # not proportional!

all_genera_reads_proteo_shallow = psclust_long_proteo_shallow %>% group_by(Genus) %>% summarise(Total = sum(Abundance)) 
all_genera_reads_proteo_shallow$Proportion_Total = (all_genera_reads_proteo_shallow$Total)/(sum(psclust_long$Abundance))*100
all_genera_reads_proteo_shallow$Proportion_Prot = (all_genera_reads_proteo_shallow$Total)/(sum(psclust_long_proteo_shallow$Abundance))*100

all_genera_reads_proteo_shallow = subset(all_genera_reads_proteo_shallow, Total > 0)
n_distinct(all_genera_reads_proteo_shallow$Genus) #571 (includes unclassifieds)

# Cluster 4 intermediate
psclust_long_proteo_inter = subset(psclust_long, Phylum == "Proteobacteria" & Cluster == "cluster4")

# sum read counts by sample ID and phylum
#psclust_sum_genera_proteo_inter = psclust_long_proteo_inter %>% group_by(SampleID, Genus) %>% summarise(All = sum(Abundance)) # not proportional!

all_genera_reads_proteo_inter = psclust_long_proteo_inter %>% group_by(Genus) %>% summarise(Total = sum(Abundance)) 
all_genera_reads_proteo_inter$Proportion_Total = (all_genera_reads_proteo_inter$Total)/(sum(psclust_long$Abundance))*100
all_genera_reads_proteo_inter$Proportion_Prot = (all_genera_reads_proteo_inter$Total)/(sum(psclust_long_proteo_inter$Abundance))*100

all_genera_reads_proteo_inter = subset(all_genera_reads_proteo_inter, Total > 0)
n_distinct(all_genera_reads_proteo_inter$Genus) #503 (includes unclassifieds)

# Cluster 1 deep
psclust_long_proteo_deep = subset(psclust_long, Phylum == "Proteobacteria" & Cluster == "cluster1")

# sum read counts by sample ID and phylum
#psclust_sum_genera_proteo_inter = psclust_long_proteo_inter %>% group_by(SampleID, Genus) %>% summarise(All = sum(Abundance)) # not proportional!

all_genera_reads_proteo_deep = psclust_long_proteo_deep %>% group_by(Genus) %>% summarise(Total = sum(Abundance)) 
all_genera_reads_proteo_deep$Proportion_Total = (all_genera_reads_proteo_deep$Total)/(sum(psclust_long$Abundance))*100
all_genera_reads_proteo_deep$Proportion_Prot = (all_genera_reads_proteo_deep$Total)/(sum(psclust_long_proteo_deep$Abundance))*100

all_genera_reads_proteo_deep = subset(all_genera_reads_proteo_deep, Total > 0)
n_distinct(all_genera_reads_proteo_deep$Genus) #401 (includes unclassifieds)

# Cluster 3 W4 without overtopping
psclust_long_proteo_c3 = subset(psclust_long, Phylum == "Proteobacteria" & Cluster == "cluster3")

# sum read counts by sample ID and phylum
#psclust_sum_genera_proteo_inter = psclust_long_proteo_inter %>% group_by(SampleID, Genus) %>% summarise(All = sum(Abundance)) # not proportional!

all_genera_reads_proteo_c3 = psclust_long_proteo_c3 %>% group_by(Genus) %>% summarise(Total = sum(Abundance)) 
all_genera_reads_proteo_c3$Proportion_Total = (all_genera_reads_proteo_c3$Total)/(sum(psclust_long$Abundance))*100
all_genera_reads_proteo_c3$Proportion_Prot = (all_genera_reads_proteo_c3$Total)/(sum(psclust_long_proteo_c3$Abundance))*100

all_genera_reads_proteo_c3 = subset(all_genera_reads_proteo_c3, Total > 0)
n_distinct(all_genera_reads_proteo_c3$Genus) #480 (includes unclassifieds)

# Well 4 all
psclust_long_proteo_W4 = subset(psclust_long, Phylum == "Proteobacteria" & Well == "Well 4")

# sum read counts by sample ID and phylum
#psclust_sum_genera_proteo_inter = psclust_long_proteo_inter %>% group_by(SampleID, Genus) %>% summarise(All = sum(Abundance)) # not proportional!

all_genera_reads_proteo_W4 = psclust_long_proteo_W4 %>% group_by(Genus) %>% summarise(Total = sum(Abundance)) 
all_genera_reads_proteo_W4$Proportion_Total = (all_genera_reads_proteo_W4$Total)/(sum(psclust_long$Abundance))*100
all_genera_reads_proteo_W4$Proportion_Prot = (all_genera_reads_proteo_W4$Total)/(sum(psclust_long_proteo_W4$Abundance))*100

all_genera_reads_proteo_W4 = subset(all_genera_reads_proteo_W4, Total > 0)
n_distinct(all_genera_reads_proteo_W4$Genus) #492 (includes unclassifieds)

# Cluster2 SW and overtopping
psclust_long_proteo_c2 = subset(psclust_long, Phylum == "Proteobacteria" & Cluster == "cluster2")

# sum read counts by sample ID and phylum
#psclust_sum_genera_proteo_inter = psclust_long_proteo_inter %>% group_by(SampleID, Genus) %>% summarise(All = sum(Abundance)) # not proportional!

all_genera_reads_proteo_c2 = psclust_long_proteo_c2 %>% group_by(Genus) %>% summarise(Total = sum(Abundance)) 
all_genera_reads_proteo_c2$Proportion_Total = (all_genera_reads_proteo_c2$Total)/(sum(psclust_long$Abundance))*100
all_genera_reads_proteo_c2$Proportion_Prot = (all_genera_reads_proteo_c2$Total)/(sum(psclust_long_proteo_c2$Abundance))*100

all_genera_reads_proteo_c2 = subset(all_genera_reads_proteo_c2, Total > 0)
n_distinct(all_genera_reads_proteo_c2$Genus) #433 (includes unclassifieds)

# Seawater only
psclust_long_proteo_SW = subset(psclust_long, Phylum == "Proteobacteria" & Well == "Seawater")

# sum read counts by sample ID and phylum
#psclust_sum_genera_proteo_inter = psclust_long_proteo_inter %>% group_by(SampleID, Genus) %>% summarise(All = sum(Abundance)) # not proportional!

all_genera_reads_proteo_SW = psclust_long_proteo_SW %>% group_by(Genus) %>% summarise(Total = sum(Abundance)) 
all_genera_reads_proteo_SW$Proportion_Total = (all_genera_reads_proteo_SW$Total)/(sum(psclust_long$Abundance))*100
all_genera_reads_proteo_SW$Proportion_Prot = (all_genera_reads_proteo_SW$Total)/(sum(psclust_long_proteo_SW$Abundance))*100

all_genera_reads_proteo_SW = subset(all_genera_reads_proteo_SW, Total > 0)
n_distinct(all_genera_reads_proteo_SW$Genus) #295 (includes unclassifieds)



# ggplot(all_genera_reads, aes(x = reorder(Genus, Proportion), y = Proportion)) +
#   geom_bar(stat = "identity") +
#   coord_flip() +
#   theme_bw() +
#   labs(x = "", y = "Percent of Total Reads (%)")
# 
# ggsave("phyla_rank_percent_2024.03.14.png", plot = last_plot(), 
#        units = "cm", width = 28, height = 28, dpi = 300)
# 
# ggplot(all_phyla_reads, aes(x = reorder(Phylum, Total), y = Total)) +
#   geom_bar(stat = "identity") +
#   coord_flip() +
#   theme_bw() +
#   labs(x = "", y = "\nTotal Read Count")
# 
# ggsave("phyla_rank_reads_2024.03.14.png", plot = last_plot(), 
#        units = "cm", width = 28, height = 28, dpi = 300)


sub_genera_reads = subset(all_genera_reads, Proportion >= 0.3 )  #0.5%

sub_genera_reads = sub_genera_reads[order(-sub_genera_reads$Proportion),]
sub_genera_reads$Phylum_Genus # 49/1749

# how much of the reads are they ?
sum(sub_genera_reads$Total)/sum(psclust_long$Abundance) # 0.633369

# store the order for plotting
genera_list_0.3 = sub_genera_reads$Phylum_Genus

# total reads per sample
psclust_sum_sample = psclust_long %>% group_by(SampleID) %>% summarise(Reads = sum(Abundance)) 
check_data = data.frame(sample_data(ps_pw_sw))
check_data2 = merge(psclust_sum_sample, check_data, by = "SampleID")

# merge sample separated with total per sample
psclust_sum_genera2 = left_join(psclust_sum_genera, psclust_sum_sample, by = "SampleID")

# take proportion of sample-phylum per sample total
psclust_sum_genera2$Proportion = (psclust_sum_genera2$All)/(psclust_sum_genera2$Reads)
psclust_sum_genera2$Percent = (psclust_sum_genera2$All)/(psclust_sum_genera2$Reads)*100

# subset to genera > 0.3% of total reads
psclust_sum_genera2_sub = subset(psclust_sum_genera2, Phylum_Genus %in% c(genera_list_0.3))

unique(psclust_sum_genera2_sub$Phylum_Genus) # check

# create an "other"  group
sub_genera_total = psclust_sum_genera2_sub %>% group_by(SampleID) %>% summarise(Percent = sum(Percent)) # highest is 83
psclust_sum_genera2_other = psclust_sum_genera2_sub %>% group_by(SampleID) %>% summarise(Percent = 83 - sum(Percent))
psclust_sum_genera2_other$Phylum_Genus = "Other"

# trim table 
psclust_sum_genera_0.3 = data.frame(psclust_sum_genera2_sub$SampleID, psclust_sum_genera2_sub$Percent, psclust_sum_genera2_sub$Phylum_Genus)
colnames(psclust_sum_genera_0.3) = c("SampleID", "Percent", "Phylum_Genus")

# bind dataframes
psclust_genera = rbind(psclust_sum_genera_0.3, psclust_sum_genera2_other)

# merge with metadata
genera_table = merge(check_data2, psclust_genera, by = "SampleID")

# specify order of samples by metadata file # well, depth order
# genera_table$SampleID = as.factor(genera_table$SampleID)
# meta = read.csv("Stinson_16S_metadata_2024.03.14.csv") #204 with sand
# meta = subset(meta, Type != "Sand")
# genera_table$SampleID = factor(genera_table$SampleID, levels = c(rev(meta$SampleID)))
# levels(genera_table$SampleID) # reverse order for plot

# specify order 
genera_list_0.3_other = c(genera_list_0.3,  "Other")
genera_table$Phylum_Genus = as.factor(genera_table$Phylum_Genus)
genera_table$Phylum_Genus = factor(genera_table$Phylum_Genus, levels = c((genera_list_0.3_other)))
levels(genera_table$Phylum_Genus)

# define colors
colors_matt_30 =c( "white", "#a7b8c2", "#f0ece2", "#dfd3c3", "#c7b198" ,"#9c4f41", "#bf2651", "#4d6a94", "#b3785d", "#e6cc8a", "#997d76", "#ffd3a3" ,"#52216e", "#ffac7f" ,"#ff8766" ,"#e6a3a3", "#d6a57a", "#4b3d53" ,"#911d55", "#fafac3", "#99c2db", "#995c95", "#93c2a7", "#66333d", "#524a63", "#4c8f82" ,"#b2c5c7" ,"#357985" ,"#728794",  "#dfeded","#242b4a")

colors_matt_60 =c( "#a7b8c2", "#f0ece2", "#dfd3c3", "#c7b198" ,"#9c4f41", "#bf2651","#f6f694", "#4d6a94", "#b3785d", "#e6cc8a", "#997d76", "#ffd3a3" ,"#52216e", "#ffac7f" ,"#ff8766" ,"#e6a3a3", "#d6a57a", "#4b3d53" ,"#911d55", "#fafac3", "#99c2db", "#995c95", "#93c2a7", "#66333d", "#524a63", "#4c8f82" ,"#b2c5c7" ,"#357985" ,"#728794",  "#dfeded","#242b4a",
                  "#5161a6", "#744d8b",  "#9779a8", "#b9a6c5", "#aab7be", "#d4dbde", "#e5795b","#ff9375","#ff9f84","#ffab93","#ffb7a3","#ffc3b2","#ffcfc1","#ffdbd1","#ffe7e0",  "#6fa59b", "#93bbb4", "#666666", "#7f7f7f", "#999999", "#b2b2b2", "#d4dbde", "#66333d", "#7a9baf", "#89aec5", "#99c2db", "#43364a", "#5d5064", "#6e6375")

colors_matt_50  = sample(colors_matt_60, 49)
colors_matt_50 = c(colors_matt_50, "white")
 

# ordered by well depth
ggplot(genera_table, aes(x = Percent, y = reorder(SampleID, ClusterOrder, decreasing = T))) +
  geom_bar(stat = "identity", aes(fill = Phylum_Genus))+
  theme_bw() +
  theme(strip.text = element_text(size = 11)) +
  #theme(legend.position = "none") +
  #scale_y_continuous(limits = c(0, 50)) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) + 
  scale_fill_manual(values = colors_matt_50, name = "Taxa") +
  labs(x = "Relative Abundance (%)\n", y = "")

ggsave("Genera_Stacked_0.3_2024.04.02.png", plot = last_plot(),
       units = "cm", width = 40, height = 55, dpi = 300)

# # flip order for legend
# phyla_table$Phylum = factor(phyla_table$Phylum, levels = c((phyla_list_0.1_other)))
# levels(phyla_table$Phylum)
# ggplot(phyla_table, aes(x = Percent, y = SampleID)) +
#   geom_bar(stat = "identity", aes(fill = Phylum)) +
#   theme_bw() +
#   theme(strip.text = element_text(size = 11)) +
#   #theme(legend.position = "none") +
#   #scale_y_continuous(limits = c(0, 50)) +
#   theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
#   #theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) + 
#   scale_fill_manual(values = rev(colors_match), name = "Phyla") +
#   labs(x = "Relative Abundance (%)\n", y = "")
# 
# ggsave("Phyla_Legend_0.15_2024.03.14.png", plot = last_plot(),
#        units = "cm", width = 40, height = 55, dpi = 300)
# 
# 
# # including date tide
# #check_data2_sort$depth_bin = as.character(check_data2_sort$depth_bin)
# check_data3_sort = meta
# check_data3_sort$SampleWDBDT = paste(check_data3_sort$W, " ", check_data3_sort$depth_bin, " ", check_data3_sort$Date_Tide)
# 
# #phyla_table_2$depth_bin = as.character(phyla_table_2$depth_bin)
# phyla_table_3 = phyla_table
# phyla_table_3$SampleWDBDT = paste(phyla_table_3$W, " ", phyla_table_3$depth_bin, " ", phyla_table_3$Date_Tide)
# 
# phyla_table_3$SampleWDBDT = factor(phyla_table_3$SampleWDBDT, levels = c(rev(check_data3_sort$SampleWDBDT)))
# levels(phyla_table_3$SampleWDBDT)
# 
# ggplot(phyla_table_3, aes(x = Percent, y = SampleWDBDT)) +
#   geom_bar(stat = "identity", aes(fill = Phylum)) +
#   theme_bw() +
#   theme(strip.text = element_text(size = 11)) +
#   #theme(legend.position = "none") +
#   #scale_y_continuous(limits = c(0, 50)) +
#   theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
#   #theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) + 
#   scale_fill_manual(values = colors_match, name = "Phyla") +
#   labs(x = "Relative Abundance (%)\n", y = "") +
#   theme(panel.border = element_blank())
# 
# ggsave("Phyla_Stacked_DateTide_2024.03.14.png", plot = last_plot(),
#        units = "cm", width = 40, height = 55, dpi = 300)


# only samples already sent for sequencing
mgs = c("240222-H-W1D2", "240222-H-W1D3", "240222-H-W2D2", "240222-H-W2D3", "240222-H-W3D2", "240222-H-W3D3", "240222-H-W3D4", "240222-H-W4D1", "240222-H-W4D3", "250222-L-SW")

ggplot(subset(genera_table, SampleID %in% mgs), aes(x = Percent, y = SampleID)) +
  geom_bar(stat = "identity", aes(fill = Phylum_Genus)) +
  theme_bw() +
  theme(strip.text = element_text(size = 11)) +
  theme(legend.position = "none") +
  #scale_y_continuous(limits = c(0, 50)) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) + 
  scale_fill_manual(values = colors_match, name = "Taxa") +
  labs(x = "Relative Abundance (%)\n", y = "") +
  theme(panel.border = element_blank())

ggsave("MGs_Genera_2024.03.15.png", plot = last_plot(),
       units = "cm", width = 20, height = 5, dpi = 300)



```

```{r Verrucomicrobiota}
# define Phylum_Genus
psclust_long_verr = subset(psclust_long, Phylum == "Verrucomicrobiota")
#psclust_long$Phylum_Genus = paste(psclust_long$Phylum, psclust_long$Genus, sep = "; ")

# sum read counts by sample ID and phylum
psclust_sum_genera_verr = psclust_long_verr %>% group_by(SampleID, Genus) %>% summarise(All = sum(Abundance)) # not proportional!


psclust_asvs_verr = psclust_long_verr %>% group_by(OTU) %>% summarise(All = sum(Abundance)) 
psclust_asvs_verr = subset(psclust_asvs_verr, All > 0)


# phyla comprising 0.1% of all reads (Ruiz Gonzales 2022)
all_genera_reads_verr = psclust_long_verr %>% group_by(Genus) %>% summarise(Total = sum(Abundance)) 
all_genera_reads_verr$Proportion_Total = (all_genera_reads_verr$Total)/(sum(psclust_long$Abundance))*100
all_genera_reads_verr$Proportion_Prot = (all_genera_reads_verr$Total)/(sum(psclust_long_verr$Abundance))*100

all_genera_reads_verr = subset(all_genera_reads_verr, Total > 0)
n_distinct(all_genera_reads_verr$Genus) #91 (includes unclassifieds)

# Omnitrophia
psclust_long_omni = subset(psclust_long, Class == "Omnitrophia")
#psclust_long$Phylum_Genus = paste(psclust_long$Phylum, psclust_long$Genus, sep = "; ")

# sum read counts by sample ID and phylum
psclust_sum_genera_omni = psclust_long_verr %>% group_by(SampleID, Genus) %>% summarise(All = sum(Abundance)) # not proportional!


psclust_asvs_omni = psclust_long_omni %>% group_by(OTU) %>% summarise(All = sum(Abundance)) 
psclust_asvs_omni = subset(psclust_asvs_omni, All > 0)


# phyla comprising 0.1% of all reads (Ruiz Gonzales 2022)
all_genera_reads_omni = psclust_long_omni %>% group_by(Genus) %>% summarise(Total = sum(Abundance)) 
all_genera_reads_omni$Proportion_Total = (all_genera_reads_omni$Total)/(sum(psclust_long$Abundance))*100
all_genera_reads_omni$Proportion_Prot = (all_genera_reads_omni$Total)/(sum(psclust_long_omni$Abundance))*100

all_genera_reads_omni = subset(all_genera_reads_omni, Total > 0)
n_distinct(all_genera_reads_omni$Genus) #4 (includes unclassifieds)



# for each cluster
# Cluster 5 shallow
psclust_long_proteo_shallow = subset(psclust_long, Phylum == "Proteobacteria" & Cluster == "cluster5")

# sum read counts by sample ID and phylum
#psclust_sum_genera_proteo_shallow = psclust_long_proteo_shallow %>% group_by(SampleID, Genus) %>% summarise(All = sum(Abundance)) # not proportional!

all_genera_reads_proteo_shallow = psclust_long_proteo_shallow %>% group_by(Genus) %>% summarise(Total = sum(Abundance)) 
all_genera_reads_proteo_shallow$Proportion_Total = (all_genera_reads_proteo_shallow$Total)/(sum(psclust_long$Abundance))*100
all_genera_reads_proteo_shallow$Proportion_Prot = (all_genera_reads_proteo_shallow$Total)/(sum(psclust_long_proteo_shallow$Abundance))*100

all_genera_reads_proteo_shallow = subset(all_genera_reads_proteo_shallow, Total > 0)
n_distinct(all_genera_reads_proteo_shallow$Genus) #571 (includes unclassifieds)

# Cluster 4 intermediate
psclust_long_proteo_inter = subset(psclust_long, Phylum == "Proteobacteria" & Cluster == "cluster4")

# sum read counts by sample ID and phylum
#psclust_sum_genera_proteo_inter = psclust_long_proteo_inter %>% group_by(SampleID, Genus) %>% summarise(All = sum(Abundance)) # not proportional!

all_genera_reads_proteo_inter = psclust_long_proteo_inter %>% group_by(Genus) %>% summarise(Total = sum(Abundance)) 
all_genera_reads_proteo_inter$Proportion_Total = (all_genera_reads_proteo_inter$Total)/(sum(psclust_long$Abundance))*100
all_genera_reads_proteo_inter$Proportion_Prot = (all_genera_reads_proteo_inter$Total)/(sum(psclust_long_proteo_inter$Abundance))*100

all_genera_reads_proteo_inter = subset(all_genera_reads_proteo_inter, Total > 0)
n_distinct(all_genera_reads_proteo_inter$Genus) #503 (includes unclassifieds)

# Cluster 1 deep
psclust_long_proteo_deep = subset(psclust_long, Phylum == "Proteobacteria" & Cluster == "cluster1")

# sum read counts by sample ID and phylum
#psclust_sum_genera_proteo_inter = psclust_long_proteo_inter %>% group_by(SampleID, Genus) %>% summarise(All = sum(Abundance)) # not proportional!

all_genera_reads_proteo_deep = psclust_long_proteo_deep %>% group_by(Genus) %>% summarise(Total = sum(Abundance)) 
all_genera_reads_proteo_deep$Proportion_Total = (all_genera_reads_proteo_deep$Total)/(sum(psclust_long$Abundance))*100
all_genera_reads_proteo_deep$Proportion_Prot = (all_genera_reads_proteo_deep$Total)/(sum(psclust_long_proteo_deep$Abundance))*100

all_genera_reads_proteo_deep = subset(all_genera_reads_proteo_deep, Total > 0)
n_distinct(all_genera_reads_proteo_deep$Genus) #401 (includes unclassifieds)

# Cluster 3 W4 without overtopping
psclust_long_proteo_c3 = subset(psclust_long, Phylum == "Proteobacteria" & Cluster == "cluster3")

# sum read counts by sample ID and phylum
#psclust_sum_genera_proteo_inter = psclust_long_proteo_inter %>% group_by(SampleID, Genus) %>% summarise(All = sum(Abundance)) # not proportional!

all_genera_reads_proteo_c3 = psclust_long_proteo_c3 %>% group_by(Genus) %>% summarise(Total = sum(Abundance)) 
all_genera_reads_proteo_c3$Proportion_Total = (all_genera_reads_proteo_c3$Total)/(sum(psclust_long$Abundance))*100
all_genera_reads_proteo_c3$Proportion_Prot = (all_genera_reads_proteo_c3$Total)/(sum(psclust_long_proteo_c3$Abundance))*100

all_genera_reads_proteo_c3 = subset(all_genera_reads_proteo_c3, Total > 0)
n_distinct(all_genera_reads_proteo_c3$Genus) #480 (includes unclassifieds)

# Well 4 all
psclust_long_proteo_W4 = subset(psclust_long, Phylum == "Proteobacteria" & Well == "Well 4")

# sum read counts by sample ID and phylum
#psclust_sum_genera_proteo_inter = psclust_long_proteo_inter %>% group_by(SampleID, Genus) %>% summarise(All = sum(Abundance)) # not proportional!

all_genera_reads_proteo_W4 = psclust_long_proteo_W4 %>% group_by(Genus) %>% summarise(Total = sum(Abundance)) 
all_genera_reads_proteo_W4$Proportion_Total = (all_genera_reads_proteo_W4$Total)/(sum(psclust_long$Abundance))*100
all_genera_reads_proteo_W4$Proportion_Prot = (all_genera_reads_proteo_W4$Total)/(sum(psclust_long_proteo_W4$Abundance))*100

all_genera_reads_proteo_W4 = subset(all_genera_reads_proteo_W4, Total > 0)
n_distinct(all_genera_reads_proteo_W4$Genus) #492 (includes unclassifieds)

# Cluster2 SW and overtopping
psclust_long_proteo_c2 = subset(psclust_long, Phylum == "Proteobacteria" & Cluster == "cluster2")

# sum read counts by sample ID and phylum
#psclust_sum_genera_proteo_inter = psclust_long_proteo_inter %>% group_by(SampleID, Genus) %>% summarise(All = sum(Abundance)) # not proportional!

all_genera_reads_proteo_c2 = psclust_long_proteo_c2 %>% group_by(Genus) %>% summarise(Total = sum(Abundance)) 
all_genera_reads_proteo_c2$Proportion_Total = (all_genera_reads_proteo_c2$Total)/(sum(psclust_long$Abundance))*100
all_genera_reads_proteo_c2$Proportion_Prot = (all_genera_reads_proteo_c2$Total)/(sum(psclust_long_proteo_c2$Abundance))*100

all_genera_reads_proteo_c2 = subset(all_genera_reads_proteo_c2, Total > 0)
n_distinct(all_genera_reads_proteo_c2$Genus) #433 (includes unclassifieds)

# Seawater only
psclust_long_proteo_SW = subset(psclust_long, Phylum == "Proteobacteria" & Well == "Seawater")

# sum read counts by sample ID and phylum
#psclust_sum_genera_proteo_inter = psclust_long_proteo_inter %>% group_by(SampleID, Genus) %>% summarise(All = sum(Abundance)) # not proportional!

all_genera_reads_proteo_SW = psclust_long_proteo_SW %>% group_by(Genus) %>% summarise(Total = sum(Abundance)) 
all_genera_reads_proteo_SW$Proportion_Total = (all_genera_reads_proteo_SW$Total)/(sum(psclust_long$Abundance))*100
all_genera_reads_proteo_SW$Proportion_Prot = (all_genera_reads_proteo_SW$Total)/(sum(psclust_long_proteo_SW$Abundance))*100

all_genera_reads_proteo_SW = subset(all_genera_reads_proteo_SW, Total > 0)
n_distinct(all_genera_reads_proteo_SW$Genus) #295 (includes unclassifieds)



# ggplot(all_genera_reads, aes(x = reorder(Genus, Proportion), y = Proportion)) +
#   geom_bar(stat = "identity") +
#   coord_flip() +
#   theme_bw() +
#   labs(x = "", y = "Percent of Total Reads (%)")
# 
# ggsave("phyla_rank_percent_2024.03.14.png", plot = last_plot(), 
#        units = "cm", width = 28, height = 28, dpi = 300)
# 
# ggplot(all_phyla_reads, aes(x = reorder(Phylum, Total), y = Total)) +
#   geom_bar(stat = "identity") +
#   coord_flip() +
#   theme_bw() +
#   labs(x = "", y = "\nTotal Read Count")
# 
# ggsave("phyla_rank_reads_2024.03.14.png", plot = last_plot(), 
#        units = "cm", width = 28, height = 28, dpi = 300)


sub_genera_reads = subset(all_genera_reads, Proportion >= 0.3 )  #0.5%

sub_genera_reads = sub_genera_reads[order(-sub_genera_reads$Proportion),]
sub_genera_reads$Phylum_Genus # 49/1749

# how much of the reads are they ?
sum(sub_genera_reads$Total)/sum(psclust_long$Abundance) # 0.633369

# store the order for plotting
genera_list_0.3 = sub_genera_reads$Phylum_Genus

# total reads per sample
psclust_sum_sample = psclust_long %>% group_by(SampleID) %>% summarise(Reads = sum(Abundance)) 
check_data = data.frame(sample_data(ps_pw_sw))
check_data2 = merge(psclust_sum_sample, check_data, by = "SampleID")

# merge sample separated with total per sample
psclust_sum_genera2 = left_join(psclust_sum_genera, psclust_sum_sample, by = "SampleID")

# take proportion of sample-phylum per sample total
psclust_sum_genera2$Proportion = (psclust_sum_genera2$All)/(psclust_sum_genera2$Reads)
psclust_sum_genera2$Percent = (psclust_sum_genera2$All)/(psclust_sum_genera2$Reads)*100

# subset to genera > 0.3% of total reads
psclust_sum_genera2_sub = subset(psclust_sum_genera2, Phylum_Genus %in% c(genera_list_0.3))

unique(psclust_sum_genera2_sub$Phylum_Genus) # check

# create an "other"  group
sub_genera_total = psclust_sum_genera2_sub %>% group_by(SampleID) %>% summarise(Percent = sum(Percent)) # highest is 83
psclust_sum_genera2_other = psclust_sum_genera2_sub %>% group_by(SampleID) %>% summarise(Percent = 83 - sum(Percent))
psclust_sum_genera2_other$Phylum_Genus = "Other"

# trim table 
psclust_sum_genera_0.3 = data.frame(psclust_sum_genera2_sub$SampleID, psclust_sum_genera2_sub$Percent, psclust_sum_genera2_sub$Phylum_Genus)
colnames(psclust_sum_genera_0.3) = c("SampleID", "Percent", "Phylum_Genus")

# bind dataframes
psclust_genera = rbind(psclust_sum_genera_0.3, psclust_sum_genera2_other)

# merge with metadata
genera_table = merge(check_data2, psclust_genera, by = "SampleID")

# specify order of samples by metadata file # well, depth order
# genera_table$SampleID = as.factor(genera_table$SampleID)
# meta = read.csv("Stinson_16S_metadata_2024.03.14.csv") #204 with sand
# meta = subset(meta, Type != "Sand")
# genera_table$SampleID = factor(genera_table$SampleID, levels = c(rev(meta$SampleID)))
# levels(genera_table$SampleID) # reverse order for plot

# specify order 
genera_list_0.3_other = c(genera_list_0.3,  "Other")
genera_table$Phylum_Genus = as.factor(genera_table$Phylum_Genus)
genera_table$Phylum_Genus = factor(genera_table$Phylum_Genus, levels = c((genera_list_0.3_other)))
levels(genera_table$Phylum_Genus)

# define colors
colors_matt_30 =c( "white", "#a7b8c2", "#f0ece2", "#dfd3c3", "#c7b198" ,"#9c4f41", "#bf2651", "#4d6a94", "#b3785d", "#e6cc8a", "#997d76", "#ffd3a3" ,"#52216e", "#ffac7f" ,"#ff8766" ,"#e6a3a3", "#d6a57a", "#4b3d53" ,"#911d55", "#fafac3", "#99c2db", "#995c95", "#93c2a7", "#66333d", "#524a63", "#4c8f82" ,"#b2c5c7" ,"#357985" ,"#728794",  "#dfeded","#242b4a")

colors_matt_60 =c( "#a7b8c2", "#f0ece2", "#dfd3c3", "#c7b198" ,"#9c4f41", "#bf2651","#f6f694", "#4d6a94", "#b3785d", "#e6cc8a", "#997d76", "#ffd3a3" ,"#52216e", "#ffac7f" ,"#ff8766" ,"#e6a3a3", "#d6a57a", "#4b3d53" ,"#911d55", "#fafac3", "#99c2db", "#995c95", "#93c2a7", "#66333d", "#524a63", "#4c8f82" ,"#b2c5c7" ,"#357985" ,"#728794",  "#dfeded","#242b4a",
                  "#5161a6", "#744d8b",  "#9779a8", "#b9a6c5", "#aab7be", "#d4dbde", "#e5795b","#ff9375","#ff9f84","#ffab93","#ffb7a3","#ffc3b2","#ffcfc1","#ffdbd1","#ffe7e0",  "#6fa59b", "#93bbb4", "#666666", "#7f7f7f", "#999999", "#b2b2b2", "#d4dbde", "#66333d", "#7a9baf", "#89aec5", "#99c2db", "#43364a", "#5d5064", "#6e6375")

colors_matt_50  = sample(colors_matt_60, 49)
colors_matt_50 = c(colors_matt_50, "white")
 

# ordered by well depth
ggplot(genera_table, aes(x = Percent, y = reorder(SampleID, ClusterOrder, decreasing = T))) +
  geom_bar(stat = "identity", aes(fill = Phylum_Genus))+
  theme_bw() +
  theme(strip.text = element_text(size = 11)) +
  #theme(legend.position = "none") +
  #scale_y_continuous(limits = c(0, 50)) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) + 
  scale_fill_manual(values = colors_matt_50, name = "Taxa") +
  labs(x = "Relative Abundance (%)\n", y = "")

ggsave("Genera_Stacked_0.3_2024.04.02.png", plot = last_plot(),
       units = "cm", width = 40, height = 55, dpi = 300)

# # flip order for legend
# phyla_table$Phylum = factor(phyla_table$Phylum, levels = c((phyla_list_0.1_other)))
# levels(phyla_table$Phylum)
# ggplot(phyla_table, aes(x = Percent, y = SampleID)) +
#   geom_bar(stat = "identity", aes(fill = Phylum)) +
#   theme_bw() +
#   theme(strip.text = element_text(size = 11)) +
#   #theme(legend.position = "none") +
#   #scale_y_continuous(limits = c(0, 50)) +
#   theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
#   #theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) + 
#   scale_fill_manual(values = rev(colors_match), name = "Phyla") +
#   labs(x = "Relative Abundance (%)\n", y = "")
# 
# ggsave("Phyla_Legend_0.15_2024.03.14.png", plot = last_plot(),
#        units = "cm", width = 40, height = 55, dpi = 300)
# 
# 
# # including date tide
# #check_data2_sort$depth_bin = as.character(check_data2_sort$depth_bin)
# check_data3_sort = meta
# check_data3_sort$SampleWDBDT = paste(check_data3_sort$W, " ", check_data3_sort$depth_bin, " ", check_data3_sort$Date_Tide)
# 
# #phyla_table_2$depth_bin = as.character(phyla_table_2$depth_bin)
# phyla_table_3 = phyla_table
# phyla_table_3$SampleWDBDT = paste(phyla_table_3$W, " ", phyla_table_3$depth_bin, " ", phyla_table_3$Date_Tide)
# 
# phyla_table_3$SampleWDBDT = factor(phyla_table_3$SampleWDBDT, levels = c(rev(check_data3_sort$SampleWDBDT)))
# levels(phyla_table_3$SampleWDBDT)
# 
# ggplot(phyla_table_3, aes(x = Percent, y = SampleWDBDT)) +
#   geom_bar(stat = "identity", aes(fill = Phylum)) +
#   theme_bw() +
#   theme(strip.text = element_text(size = 11)) +
#   #theme(legend.position = "none") +
#   #scale_y_continuous(limits = c(0, 50)) +
#   theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
#   #theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) + 
#   scale_fill_manual(values = colors_match, name = "Phyla") +
#   labs(x = "Relative Abundance (%)\n", y = "") +
#   theme(panel.border = element_blank())
# 
# ggsave("Phyla_Stacked_DateTide_2024.03.14.png", plot = last_plot(),
#        units = "cm", width = 40, height = 55, dpi = 300)


# only samples already sent for sequencing
mgs = c("240222-H-W1D2", "240222-H-W1D3", "240222-H-W2D2", "240222-H-W2D3", "240222-H-W3D2", "240222-H-W3D3", "240222-H-W3D4", "240222-H-W4D1", "240222-H-W4D3", "250222-L-SW")

ggplot(subset(genera_table, SampleID %in% mgs), aes(x = Percent, y = SampleID)) +
  geom_bar(stat = "identity", aes(fill = Phylum_Genus)) +
  theme_bw() +
  theme(strip.text = element_text(size = 11)) +
  theme(legend.position = "none") +
  #scale_y_continuous(limits = c(0, 50)) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) + 
  scale_fill_manual(values = colors_match, name = "Taxa") +
  labs(x = "Relative Abundance (%)\n", y = "") +
  theme(panel.border = element_blank())

ggsave("MGs_Genera_2024.03.15.png", plot = last_plot(),
       units = "cm", width = 20, height = 5, dpi = 300)



```

```{r Patescibacteria}
# define Phylum_Genus
psclust_long_cpr = subset(psclust_long, Phylum == "Patescibacteria")
#psclust_long$Phylum_Genus = paste(psclust_long$Phylum, psclust_long$Genus, sep = "; ")

# sum read counts by sample ID and phylum
psclust_sum_genera_cpr = psclust_long_cpr %>% group_by(SampleID, Genus) %>% summarise(All = sum(Abundance)) # not proportional!


psclust_asvs_cpr = psclust_long_cpr %>% group_by(OTU) %>% summarise(All = sum(Abundance)) 
psclust_asvs_cpr = subset(psclust_asvs_cpr, All > 0)


# phyla comprising 0.1% of all reads (Ruiz Gonzales 2022)
all_genera_reads_cpr = psclust_long_cpr %>% group_by(Genus) %>% summarise(Total = sum(Abundance)) 
all_genera_reads_cpr$Proportion_Total = (all_genera_reads_cpr$Total)/(sum(psclust_long$Abundance))*100
all_genera_reads_cpr$Proportion_Prot = (all_genera_reads_cpr$Total)/(sum(psclust_long_cpr$Abundance))*100

all_genera_reads_cpr = subset(all_genera_reads_cpr, Total > 0)
n_distinct(all_genera_reads_cpr$Genus) #69 (includes unclassifieds)



# for each cluster
# Cluster 5 shallow
psclust_long_cpr_shallow = subset(psclust_long, Phylum == "Patescibacteria" & Cluster == "cluster5")

# sum read counts by sample ID and phylum
#psclust_sum_genera_proteo_shallow = psclust_long_proteo_shallow %>% group_by(SampleID, Genus) %>% summarise(All = sum(Abundance)) # not proportional!

all_genera_reads_cpr_shallow = psclust_long_cpr_shallow %>% group_by(Genus) %>% summarise(Total = sum(Abundance)) 
all_genera_reads_cpr_shallow$Proportion_Total = (all_genera_reads_cpr_shallow$Total)/(sum(psclust_long$Abundance))*100
all_genera_reads_cpr_shallow$Proportion_Prot = (all_genera_reads_cpr_shallow$Total)/(sum(psclust_long_cpr_shallow$Abundance))*100

all_genera_reads_cpr_shallow = subset(all_genera_reads_cpr_shallow, Total > 0)
n_distinct(all_genera_reads_cpr_shallow$Genus) #571 (includes unclassifieds)

# Cluster 4 intermediate
psclust_long_cpr_inter = subset(psclust_long, Phylum == "Patescibacteria" & Cluster == "cluster4")

# sum read counts by sample ID and phylum
#psclust_sum_genera_proteo_inter = psclust_long_proteo_inter %>% group_by(SampleID, Genus) %>% summarise(All = sum(Abundance)) # not proportional!

all_genera_reads_cpr_inter = psclust_long_cpr_inter %>% group_by(Genus) %>% summarise(Total = sum(Abundance)) 
all_genera_reads_cpr_inter$Proportion_Total = (all_genera_reads_cpr_inter$Total)/(sum(psclust_long$Abundance))*100
all_genera_reads_cpr_inter$Proportion_Prot = (all_genera_reads_cpr_inter$Total)/(sum(psclust_long_cpr_inter$Abundance))*100

all_genera_reads_cpr_inter = subset(all_genera_reads_cpr_inter, Total > 0)
n_distinct(all_genera_reads_cpr_inter$Genus) #503 (includes unclassifieds)

# Cluster 1 deep
psclust_long_cpr_deep = subset(psclust_long, Phylum == "Patescibacteria" & Cluster == "cluster1")

# sum read counts by sample ID and phylum
#psclust_sum_genera_proteo_inter = psclust_long_proteo_inter %>% group_by(SampleID, Genus) %>% summarise(All = sum(Abundance)) # not proportional!

all_genera_reads_cpr_deep = psclust_long_cpr_deep %>% group_by(Genus) %>% summarise(Total = sum(Abundance)) 
all_genera_reads_cpr_deep$Proportion_Total = (all_genera_reads_cpr_deep$Total)/(sum(psclust_long$Abundance))*100
all_genera_reads_cpr_deep$Proportion_Prot = (all_genera_reads_cpr_deep$Total)/(sum(psclust_long_cpr_deep$Abundance))*100

all_genera_reads_cpr_deep = subset(all_genera_reads_cpr_deep, Total > 0)
n_distinct(all_genera_reads_cpr_deep$Genus) #66 (includes unclassifieds)

# Cluster 3 W4 without overtopping
psclust_long_cpr_c3 = subset(psclust_long, Phylum == "Proteobacteria" & Cluster == "cluster3")

# sum read counts by sample ID and phylum
#psclust_sum_genera_proteo_inter = psclust_long_proteo_inter %>% group_by(SampleID, Genus) %>% summarise(All = sum(Abundance)) # not proportional!

all_genera_reads_cpr_c3 = psclust_long_proteo_c3 %>% group_by(Genus) %>% summarise(Total = sum(Abundance)) 
all_genera_reads_cpr_c3$Proportion_Total = (all_genera_reads_cpr_c3$Total)/(sum(psclust_long$Abundance))*100
all_genera_reads_cpr_c3$Proportion_Prot = (all_genera_reads_cpr_c3$Total)/(sum(psclust_long_cpr_c3$Abundance))*100

all_genera_reads_proteo_c3 = subset(all_genera_reads_proteo_c3, Total > 0)
n_distinct(all_genera_reads_proteo_c3$Genus) #480 (includes unclassifieds)

# Well 4 all
psclust_long_cpr_W4 = subset(psclust_long, Phylum == "Patescibacteria" & Well == "Well 4")

# sum read counts by sample ID and phylum
#psclust_sum_genera_proteo_inter = psclust_long_proteo_inter %>% group_by(SampleID, Genus) %>% summarise(All = sum(Abundance)) # not proportional!

all_genera_reads_cpr_W4 = psclust_long_cpr_W4 %>% group_by(Genus) %>% summarise(Total = sum(Abundance)) 
all_genera_reads_cpr_W4$Proportion_Total = (all_genera_reads_cpr_W4$Total)/(sum(psclust_long$Abundance))*100
all_genera_reads_cpr_W4$Proportion_Prot = (all_genera_reads_cpr_W4$Total)/(sum(psclust_long_cpr_W4$Abundance))*100

all_genera_reads_cpr_W4 = subset(all_genera_reads_cpr_W4, Total > 0)
n_distinct(all_genera_reads_cpr_W4$Genus) #492 (includes unclassifieds)

# Cluster2 SW and overtopping
psclust_long_cpr_c2 = subset(psclust_long, Phylum == "Patescibacteria" & Cluster == "cluster2")

# sum read counts by sample ID and phylum
#psclust_sum_genera_proteo_inter = psclust_long_proteo_inter %>% group_by(SampleID, Genus) %>% summarise(All = sum(Abundance)) # not proportional!

all_genera_reads_cpr_c2 = psclust_long_cpr_c2 %>% group_by(Genus) %>% summarise(Total = sum(Abundance)) 
all_genera_reads_cpr_c2$Proportion_Total = (all_genera_reads_cpr_c2$Total)/(sum(psclust_long$Abundance))*100
all_genera_reads_cpr_c2$Proportion_Prot = (all_genera_reads_cpr_c2$Total)/(sum(psclust_long_cpr_c2$Abundance))*100

all_genera_reads_cpr_c2 = subset(all_genera_reads_cpr_c2, Total > 0)
n_distinct(all_genera_reads_cpr_c2$Genus) #433 (includes unclassifieds)

# Seawater only
psclust_long_cpr_SW = subset(psclust_long, Phylum == "Patescibacteria" & Well == "Seawater")

# sum read counts by sample ID and phylum
psclust_sum_genera_cpr_SW = psclust_long_cpr_SW %>% group_by(SampleID, Genus) %>% summarise(All = sum(Abundance)) # not proportional!

all_genera_reads_cpr_SW = psclust_long_cpr_SW %>% group_by(Genus) %>% summarise(Total = sum(Abundance)) 
all_genera_reads_cpr_SW$Proportion_Total = (all_genera_reads_cpr_SW$Total)/(sum(psclust_long$Abundance))*100
all_genera_reads_cpr_SW$Proportion_Prot = (all_genera_reads_cpr_SW$Total)/(sum(psclust_long_cpr_SW$Abundance))*100

all_genera_reads_cpr_SW = subset(all_genera_reads_cpr_SW, Total > 0)
n_distinct(all_genera_reads_cpr_SW$Genus) #295 (includes unclassifieds)



# ggplot(all_genera_reads, aes(x = reorder(Genus, Proportion), y = Proportion)) +
#   geom_bar(stat = "identity") +
#   coord_flip() +
#   theme_bw() +
#   labs(x = "", y = "Percent of Total Reads (%)")
# 
# ggsave("phyla_rank_percent_2024.03.14.png", plot = last_plot(), 
#        units = "cm", width = 28, height = 28, dpi = 300)
# 
# ggplot(all_phyla_reads, aes(x = reorder(Phylum, Total), y = Total)) +
#   geom_bar(stat = "identity") +
#   coord_flip() +
#   theme_bw() +
#   labs(x = "", y = "\nTotal Read Count")
# 
# ggsave("phyla_rank_reads_2024.03.14.png", plot = last_plot(), 
#        units = "cm", width = 28, height = 28, dpi = 300)


sub_genera_reads = subset(all_genera_reads, Proportion >= 0.3 )  #0.5%

sub_genera_reads = sub_genera_reads[order(-sub_genera_reads$Proportion),]
sub_genera_reads$Phylum_Genus # 49/1749

# how much of the reads are they ?
sum(sub_genera_reads$Total)/sum(psclust_long$Abundance) # 0.633369

# store the order for plotting
genera_list_0.3 = sub_genera_reads$Phylum_Genus

# total reads per sample
psclust_sum_sample = psclust_long %>% group_by(SampleID) %>% summarise(Reads = sum(Abundance)) 
check_data = data.frame(sample_data(ps_pw_sw))
check_data2 = merge(psclust_sum_sample, check_data, by = "SampleID")

# merge sample separated with total per sample
psclust_sum_genera2 = left_join(psclust_sum_genera, psclust_sum_sample, by = "SampleID")

# take proportion of sample-phylum per sample total
psclust_sum_genera2$Proportion = (psclust_sum_genera2$All)/(psclust_sum_genera2$Reads)
psclust_sum_genera2$Percent = (psclust_sum_genera2$All)/(psclust_sum_genera2$Reads)*100

# subset to genera > 0.3% of total reads
psclust_sum_genera2_sub = subset(psclust_sum_genera2, Phylum_Genus %in% c(genera_list_0.3))

unique(psclust_sum_genera2_sub$Phylum_Genus) # check

# create an "other"  group
sub_genera_total = psclust_sum_genera2_sub %>% group_by(SampleID) %>% summarise(Percent = sum(Percent)) # highest is 83
psclust_sum_genera2_other = psclust_sum_genera2_sub %>% group_by(SampleID) %>% summarise(Percent = 83 - sum(Percent))
psclust_sum_genera2_other$Phylum_Genus = "Other"

# trim table 
psclust_sum_genera_0.3 = data.frame(psclust_sum_genera2_sub$SampleID, psclust_sum_genera2_sub$Percent, psclust_sum_genera2_sub$Phylum_Genus)
colnames(psclust_sum_genera_0.3) = c("SampleID", "Percent", "Phylum_Genus")

# bind dataframes
psclust_genera = rbind(psclust_sum_genera_0.3, psclust_sum_genera2_other)

# merge with metadata
genera_table = merge(check_data2, psclust_genera, by = "SampleID")

# specify order of samples by metadata file # well, depth order
# genera_table$SampleID = as.factor(genera_table$SampleID)
# meta = read.csv("Stinson_16S_metadata_2024.03.14.csv") #204 with sand
# meta = subset(meta, Type != "Sand")
# genera_table$SampleID = factor(genera_table$SampleID, levels = c(rev(meta$SampleID)))
# levels(genera_table$SampleID) # reverse order for plot

# specify order 
genera_list_0.3_other = c(genera_list_0.3,  "Other")
genera_table$Phylum_Genus = as.factor(genera_table$Phylum_Genus)
genera_table$Phylum_Genus = factor(genera_table$Phylum_Genus, levels = c((genera_list_0.3_other)))
levels(genera_table$Phylum_Genus)

# define colors
colors_matt_30 =c( "white", "#a7b8c2", "#f0ece2", "#dfd3c3", "#c7b198" ,"#9c4f41", "#bf2651", "#4d6a94", "#b3785d", "#e6cc8a", "#997d76", "#ffd3a3" ,"#52216e", "#ffac7f" ,"#ff8766" ,"#e6a3a3", "#d6a57a", "#4b3d53" ,"#911d55", "#fafac3", "#99c2db", "#995c95", "#93c2a7", "#66333d", "#524a63", "#4c8f82" ,"#b2c5c7" ,"#357985" ,"#728794",  "#dfeded","#242b4a")

colors_matt_60 =c( "#a7b8c2", "#f0ece2", "#dfd3c3", "#c7b198" ,"#9c4f41", "#bf2651","#f6f694", "#4d6a94", "#b3785d", "#e6cc8a", "#997d76", "#ffd3a3" ,"#52216e", "#ffac7f" ,"#ff8766" ,"#e6a3a3", "#d6a57a", "#4b3d53" ,"#911d55", "#fafac3", "#99c2db", "#995c95", "#93c2a7", "#66333d", "#524a63", "#4c8f82" ,"#b2c5c7" ,"#357985" ,"#728794",  "#dfeded","#242b4a",
                  "#5161a6", "#744d8b",  "#9779a8", "#b9a6c5", "#aab7be", "#d4dbde", "#e5795b","#ff9375","#ff9f84","#ffab93","#ffb7a3","#ffc3b2","#ffcfc1","#ffdbd1","#ffe7e0",  "#6fa59b", "#93bbb4", "#666666", "#7f7f7f", "#999999", "#b2b2b2", "#d4dbde", "#66333d", "#7a9baf", "#89aec5", "#99c2db", "#43364a", "#5d5064", "#6e6375")

colors_matt_50  = sample(colors_matt_60, 49)
colors_matt_50 = c(colors_matt_50, "white")
 

# ordered by well depth
ggplot(genera_table, aes(x = Percent, y = reorder(SampleID, ClusterOrder, decreasing = T))) +
  geom_bar(stat = "identity", aes(fill = Phylum_Genus))+
  theme_bw() +
  theme(strip.text = element_text(size = 11)) +
  #theme(legend.position = "none") +
  #scale_y_continuous(limits = c(0, 50)) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) + 
  scale_fill_manual(values = colors_matt_50, name = "Taxa") +
  labs(x = "Relative Abundance (%)\n", y = "")

ggsave("Genera_Stacked_0.3_2024.04.02.png", plot = last_plot(),
       units = "cm", width = 40, height = 55, dpi = 300)

# # flip order for legend
# phyla_table$Phylum = factor(phyla_table$Phylum, levels = c((phyla_list_0.1_other)))
# levels(phyla_table$Phylum)
# ggplot(phyla_table, aes(x = Percent, y = SampleID)) +
#   geom_bar(stat = "identity", aes(fill = Phylum)) +
#   theme_bw() +
#   theme(strip.text = element_text(size = 11)) +
#   #theme(legend.position = "none") +
#   #scale_y_continuous(limits = c(0, 50)) +
#   theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
#   #theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) + 
#   scale_fill_manual(values = rev(colors_match), name = "Phyla") +
#   labs(x = "Relative Abundance (%)\n", y = "")
# 
# ggsave("Phyla_Legend_0.15_2024.03.14.png", plot = last_plot(),
#        units = "cm", width = 40, height = 55, dpi = 300)
# 
# 
# # including date tide
# #check_data2_sort$depth_bin = as.character(check_data2_sort$depth_bin)
# check_data3_sort = meta
# check_data3_sort$SampleWDBDT = paste(check_data3_sort$W, " ", check_data3_sort$depth_bin, " ", check_data3_sort$Date_Tide)
# 
# #phyla_table_2$depth_bin = as.character(phyla_table_2$depth_bin)
# phyla_table_3 = phyla_table
# phyla_table_3$SampleWDBDT = paste(phyla_table_3$W, " ", phyla_table_3$depth_bin, " ", phyla_table_3$Date_Tide)
# 
# phyla_table_3$SampleWDBDT = factor(phyla_table_3$SampleWDBDT, levels = c(rev(check_data3_sort$SampleWDBDT)))
# levels(phyla_table_3$SampleWDBDT)
# 
# ggplot(phyla_table_3, aes(x = Percent, y = SampleWDBDT)) +
#   geom_bar(stat = "identity", aes(fill = Phylum)) +
#   theme_bw() +
#   theme(strip.text = element_text(size = 11)) +
#   #theme(legend.position = "none") +
#   #scale_y_continuous(limits = c(0, 50)) +
#   theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
#   #theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) + 
#   scale_fill_manual(values = colors_match, name = "Phyla") +
#   labs(x = "Relative Abundance (%)\n", y = "") +
#   theme(panel.border = element_blank())
# 
# ggsave("Phyla_Stacked_DateTide_2024.03.14.png", plot = last_plot(),
#        units = "cm", width = 40, height = 55, dpi = 300)


# only samples already sent for sequencing
mgs = c("240222-H-W1D2", "240222-H-W1D3", "240222-H-W2D2", "240222-H-W2D3", "240222-H-W3D2", "240222-H-W3D3", "240222-H-W3D4", "240222-H-W4D1", "240222-H-W4D3", "250222-L-SW")

ggplot(subset(genera_table, SampleID %in% mgs), aes(x = Percent, y = SampleID)) +
  geom_bar(stat = "identity", aes(fill = Phylum_Genus)) +
  theme_bw() +
  theme(strip.text = element_text(size = 11)) +
  theme(legend.position = "none") +
  #scale_y_continuous(limits = c(0, 50)) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) + 
  scale_fill_manual(values = colors_match, name = "Taxa") +
  labs(x = "Relative Abundance (%)\n", y = "") +
  theme(panel.border = element_blank())

ggsave("MGs_Genera_2024.03.15.png", plot = last_plot(),
       units = "cm", width = 20, height = 5, dpi = 300)



```

```{r Bacteroidota}
# define Phylum_Genus
psclust_long_bact = subset(psclust_long, Phylum == "Bacteroidota")
#psclust_long$Phylum_Genus = paste(psclust_long$Phylum, psclust_long$Genus, sep = "; ")

# sum read counts by sample ID and phylum
psclust_sum_genera_bact = psclust_long_bact %>% group_by(SampleID, Genus) %>% summarise(All = sum(Abundance)) # not proportional!


psclust_asvs_bact = psclust_long_bact %>% group_by(OTU) %>% summarise(All = sum(Abundance)) 
psclust_asvs_bact = subset(psclust_asvs_bact, All > 0)


# phyla comprising 0.1% of all reads (Ruiz Gonzales 2022)
all_genera_reads_bact = psclust_long_bact %>% group_by(Genus) %>% summarise(Total = sum(Abundance)) 
all_genera_reads_bact$Proportion_Total = (all_genera_reads_bact$Total)/(sum(psclust_long$Abundance))*100
all_genera_reads_bact$Proportion_Prot = (all_genera_reads_bact$Total)/(sum(psclust_long_cpr$Abundance))*100

all_genera_reads_bact = subset(all_genera_reads_bact, Total > 0)
n_distinct(all_genera_reads_bact$Genus) #69 (includes unclassifieds)



# for each cluster
# Cluster 5 shallow
psclust_long_cpr_shallow = subset(psclust_long, Phylum == "Bacteroidota" & Cluster == "cluster5")

# sum read counts by sample ID and phylum
#psclust_sum_genera_proteo_shallow = psclust_long_proteo_shallow %>% group_by(SampleID, Genus) %>% summarise(All = sum(Abundance)) # not proportional!

all_genera_reads_cpr_shallow = psclust_long_cpr_shallow %>% group_by(Genus) %>% summarise(Total = sum(Abundance)) 
all_genera_reads_cpr_shallow$Proportion_Total = (all_genera_reads_cpr_shallow$Total)/(sum(psclust_long$Abundance))*100
all_genera_reads_cpr_shallow$Proportion_Prot = (all_genera_reads_cpr_shallow$Total)/(sum(psclust_long_cpr_shallow$Abundance))*100

all_genera_reads_cpr_shallow = subset(all_genera_reads_cpr_shallow, Total > 0)
n_distinct(all_genera_reads_cpr_shallow$Genus) #571 (includes unclassifieds)

# Cluster 4 intermediate
psclust_long_cpr_inter = subset(psclust_long, Phylum == "Bacteroidota" & Cluster == "cluster4")

# sum read counts by sample ID and phylum
#psclust_sum_genera_proteo_inter = psclust_long_proteo_inter %>% group_by(SampleID, Genus) %>% summarise(All = sum(Abundance)) # not proportional!

all_genera_reads_cpr_inter = psclust_long_cpr_inter %>% group_by(Genus) %>% summarise(Total = sum(Abundance)) 
all_genera_reads_cpr_inter$Proportion_Total = (all_genera_reads_cpr_inter$Total)/(sum(psclust_long$Abundance))*100
all_genera_reads_cpr_inter$Proportion_Prot = (all_genera_reads_cpr_inter$Total)/(sum(psclust_long_cpr_inter$Abundance))*100

all_genera_reads_cpr_inter = subset(all_genera_reads_cpr_inter, Total > 0)
n_distinct(all_genera_reads_cpr_inter$Genus) #503 (includes unclassifieds)

# Cluster 1 deep
psclust_long_cpr_deep = subset(psclust_long, Phylum == "Bacteroidota" & Cluster == "cluster1")

# sum read counts by sample ID and phylum
#psclust_sum_genera_proteo_inter = psclust_long_proteo_inter %>% group_by(SampleID, Genus) %>% summarise(All = sum(Abundance)) # not proportional!

all_genera_reads_cpr_deep = psclust_long_cpr_deep %>% group_by(Genus) %>% summarise(Total = sum(Abundance)) 
all_genera_reads_cpr_deep$Proportion_Total = (all_genera_reads_cpr_deep$Total)/(sum(psclust_long$Abundance))*100
all_genera_reads_cpr_deep$Proportion_Prot = (all_genera_reads_cpr_deep$Total)/(sum(psclust_long_cpr_deep$Abundance))*100

all_genera_reads_cpr_deep = subset(all_genera_reads_cpr_deep, Total > 0)
n_distinct(all_genera_reads_cpr_deep$Genus) #66 (includes unclassifieds)

# Cluster 3 W4 without overtopping
psclust_long_cpr_c3 = subset(psclust_long, Phylum == "Bacteroidota" & Cluster == "cluster3")

# sum read counts by sample ID and phylum
#psclust_sum_genera_proteo_inter = psclust_long_proteo_inter %>% group_by(SampleID, Genus) %>% summarise(All = sum(Abundance)) # not proportional!

all_genera_reads_cpr_c3 = psclust_long_cpr_c3 %>% group_by(Genus) %>% summarise(Total = sum(Abundance)) 
all_genera_reads_cpr_c3$Proportion_Total = (all_genera_reads_cpr_c3$Total)/(sum(psclust_long$Abundance))*100
all_genera_reads_cpr_c3$Proportion_Prot = (all_genera_reads_cpr_c3$Total)/(sum(psclust_long_cpr_c3$Abundance))*100

all_genera_reads_proteo_c3 = subset(all_genera_reads_proteo_c3, Total > 0)
n_distinct(all_genera_reads_proteo_c3$Genus) #480 (includes unclassifieds)

# Well 4 all
psclust_long_cpr_W4 = subset(psclust_long, Phylum == "Bacteroidota" & Well == "Well 4")

# sum read counts by sample ID and phylum
#psclust_sum_genera_proteo_inter = psclust_long_proteo_inter %>% group_by(SampleID, Genus) %>% summarise(All = sum(Abundance)) # not proportional!

all_genera_reads_cpr_W4 = psclust_long_cpr_W4 %>% group_by(Genus) %>% summarise(Total = sum(Abundance)) 
all_genera_reads_cpr_W4$Proportion_Total = (all_genera_reads_cpr_W4$Total)/(sum(psclust_long$Abundance))*100
all_genera_reads_cpr_W4$Proportion_Prot = (all_genera_reads_cpr_W4$Total)/(sum(psclust_long_cpr_W4$Abundance))*100

all_genera_reads_cpr_W4 = subset(all_genera_reads_cpr_W4, Total > 0)
n_distinct(all_genera_reads_cpr_W4$Genus) #492 (includes unclassifieds)

# Cluster2 SW and overtopping
psclust_long_cpr_c2 = subset(psclust_long, Phylum == "Bacteroidota" & Cluster == "cluster2")

# sum read counts by sample ID and phylum
#psclust_sum_genera_proteo_inter = psclust_long_proteo_inter %>% group_by(SampleID, Genus) %>% summarise(All = sum(Abundance)) # not proportional!

all_genera_reads_cpr_c2 = psclust_long_cpr_c2 %>% group_by(Genus) %>% summarise(Total = sum(Abundance)) 
all_genera_reads_cpr_c2$Proportion_Total = (all_genera_reads_cpr_c2$Total)/(sum(psclust_long$Abundance))*100
all_genera_reads_cpr_c2$Proportion_Prot = (all_genera_reads_cpr_c2$Total)/(sum(psclust_long_cpr_c2$Abundance))*100

all_genera_reads_cpr_c2 = subset(all_genera_reads_cpr_c2, Total > 0)
n_distinct(all_genera_reads_cpr_c2$Genus) #433 (includes unclassifieds)

# Seawater only
psclust_long_cpr_SW = subset(psclust_long, Phylum == "Bacteroidota" & Well == "Seawater")

# sum read counts by sample ID and phylum
psclust_sum_genera_cpr_SW = psclust_long_cpr_SW %>% group_by(SampleID, Genus) %>% summarise(All = sum(Abundance)) # not proportional!

all_genera_reads_cpr_SW = psclust_long_cpr_SW %>% group_by(Genus) %>% summarise(Total = sum(Abundance)) 
all_genera_reads_cpr_SW$Proportion_Total = (all_genera_reads_cpr_SW$Total)/(sum(psclust_long$Abundance))*100
all_genera_reads_cpr_SW$Proportion_Prot = (all_genera_reads_cpr_SW$Total)/(sum(psclust_long_cpr_SW$Abundance))*100

all_genera_reads_cpr_SW = subset(all_genera_reads_cpr_SW, Total > 0)
n_distinct(all_genera_reads_cpr_SW$Genus) #295 (includes unclassifieds)



# ggplot(all_genera_reads, aes(x = reorder(Genus, Proportion), y = Proportion)) +
#   geom_bar(stat = "identity") +
#   coord_flip() +
#   theme_bw() +
#   labs(x = "", y = "Percent of Total Reads (%)")
# 
# ggsave("phyla_rank_percent_2024.03.14.png", plot = last_plot(), 
#        units = "cm", width = 28, height = 28, dpi = 300)
# 
# ggplot(all_phyla_reads, aes(x = reorder(Phylum, Total), y = Total)) +
#   geom_bar(stat = "identity") +
#   coord_flip() +
#   theme_bw() +
#   labs(x = "", y = "\nTotal Read Count")
# 
# ggsave("phyla_rank_reads_2024.03.14.png", plot = last_plot(), 
#        units = "cm", width = 28, height = 28, dpi = 300)


sub_genera_reads = subset(all_genera_reads, Proportion >= 0.3 )  #0.5%

sub_genera_reads = sub_genera_reads[order(-sub_genera_reads$Proportion),]
sub_genera_reads$Phylum_Genus # 49/1749

# how much of the reads are they ?
sum(sub_genera_reads$Total)/sum(psclust_long$Abundance) # 0.633369

# store the order for plotting
genera_list_0.3 = sub_genera_reads$Phylum_Genus

# total reads per sample
psclust_sum_sample = psclust_long %>% group_by(SampleID) %>% summarise(Reads = sum(Abundance)) 
check_data = data.frame(sample_data(ps_pw_sw))
check_data2 = merge(psclust_sum_sample, check_data, by = "SampleID")

# merge sample separated with total per sample
psclust_sum_genera2 = left_join(psclust_sum_genera, psclust_sum_sample, by = "SampleID")

# take proportion of sample-phylum per sample total
psclust_sum_genera2$Proportion = (psclust_sum_genera2$All)/(psclust_sum_genera2$Reads)
psclust_sum_genera2$Percent = (psclust_sum_genera2$All)/(psclust_sum_genera2$Reads)*100

# subset to genera > 0.3% of total reads
psclust_sum_genera2_sub = subset(psclust_sum_genera2, Phylum_Genus %in% c(genera_list_0.3))

unique(psclust_sum_genera2_sub$Phylum_Genus) # check

# create an "other"  group
sub_genera_total = psclust_sum_genera2_sub %>% group_by(SampleID) %>% summarise(Percent = sum(Percent)) # highest is 83
psclust_sum_genera2_other = psclust_sum_genera2_sub %>% group_by(SampleID) %>% summarise(Percent = 83 - sum(Percent))
psclust_sum_genera2_other$Phylum_Genus = "Other"

# trim table 
psclust_sum_genera_0.3 = data.frame(psclust_sum_genera2_sub$SampleID, psclust_sum_genera2_sub$Percent, psclust_sum_genera2_sub$Phylum_Genus)
colnames(psclust_sum_genera_0.3) = c("SampleID", "Percent", "Phylum_Genus")

# bind dataframes
psclust_genera = rbind(psclust_sum_genera_0.3, psclust_sum_genera2_other)

# merge with metadata
genera_table = merge(check_data2, psclust_genera, by = "SampleID")

# specify order of samples by metadata file # well, depth order
# genera_table$SampleID = as.factor(genera_table$SampleID)
# meta = read.csv("Stinson_16S_metadata_2024.03.14.csv") #204 with sand
# meta = subset(meta, Type != "Sand")
# genera_table$SampleID = factor(genera_table$SampleID, levels = c(rev(meta$SampleID)))
# levels(genera_table$SampleID) # reverse order for plot

# specify order 
genera_list_0.3_other = c(genera_list_0.3,  "Other")
genera_table$Phylum_Genus = as.factor(genera_table$Phylum_Genus)
genera_table$Phylum_Genus = factor(genera_table$Phylum_Genus, levels = c((genera_list_0.3_other)))
levels(genera_table$Phylum_Genus)

# define colors
colors_matt_30 =c( "white", "#a7b8c2", "#f0ece2", "#dfd3c3", "#c7b198" ,"#9c4f41", "#bf2651", "#4d6a94", "#b3785d", "#e6cc8a", "#997d76", "#ffd3a3" ,"#52216e", "#ffac7f" ,"#ff8766" ,"#e6a3a3", "#d6a57a", "#4b3d53" ,"#911d55", "#fafac3", "#99c2db", "#995c95", "#93c2a7", "#66333d", "#524a63", "#4c8f82" ,"#b2c5c7" ,"#357985" ,"#728794",  "#dfeded","#242b4a")

colors_matt_60 =c( "#a7b8c2", "#f0ece2", "#dfd3c3", "#c7b198" ,"#9c4f41", "#bf2651","#f6f694", "#4d6a94", "#b3785d", "#e6cc8a", "#997d76", "#ffd3a3" ,"#52216e", "#ffac7f" ,"#ff8766" ,"#e6a3a3", "#d6a57a", "#4b3d53" ,"#911d55", "#fafac3", "#99c2db", "#995c95", "#93c2a7", "#66333d", "#524a63", "#4c8f82" ,"#b2c5c7" ,"#357985" ,"#728794",  "#dfeded","#242b4a",
                  "#5161a6", "#744d8b",  "#9779a8", "#b9a6c5", "#aab7be", "#d4dbde", "#e5795b","#ff9375","#ff9f84","#ffab93","#ffb7a3","#ffc3b2","#ffcfc1","#ffdbd1","#ffe7e0",  "#6fa59b", "#93bbb4", "#666666", "#7f7f7f", "#999999", "#b2b2b2", "#d4dbde", "#66333d", "#7a9baf", "#89aec5", "#99c2db", "#43364a", "#5d5064", "#6e6375")

colors_matt_50  = sample(colors_matt_60, 49)
colors_matt_50 = c(colors_matt_50, "white")
 

# ordered by well depth
ggplot(genera_table, aes(x = Percent, y = reorder(SampleID, ClusterOrder, decreasing = T))) +
  geom_bar(stat = "identity", aes(fill = Phylum_Genus))+
  theme_bw() +
  theme(strip.text = element_text(size = 11)) +
  #theme(legend.position = "none") +
  #scale_y_continuous(limits = c(0, 50)) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) + 
  scale_fill_manual(values = colors_matt_50, name = "Taxa") +
  labs(x = "Relative Abundance (%)\n", y = "")

ggsave("Genera_Stacked_0.3_2024.04.02.png", plot = last_plot(),
       units = "cm", width = 40, height = 55, dpi = 300)

# # flip order for legend
# phyla_table$Phylum = factor(phyla_table$Phylum, levels = c((phyla_list_0.1_other)))
# levels(phyla_table$Phylum)
# ggplot(phyla_table, aes(x = Percent, y = SampleID)) +
#   geom_bar(stat = "identity", aes(fill = Phylum)) +
#   theme_bw() +
#   theme(strip.text = element_text(size = 11)) +
#   #theme(legend.position = "none") +
#   #scale_y_continuous(limits = c(0, 50)) +
#   theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
#   #theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) + 
#   scale_fill_manual(values = rev(colors_match), name = "Phyla") +
#   labs(x = "Relative Abundance (%)\n", y = "")
# 
# ggsave("Phyla_Legend_0.15_2024.03.14.png", plot = last_plot(),
#        units = "cm", width = 40, height = 55, dpi = 300)
# 
# 
# # including date tide
# #check_data2_sort$depth_bin = as.character(check_data2_sort$depth_bin)
# check_data3_sort = meta
# check_data3_sort$SampleWDBDT = paste(check_data3_sort$W, " ", check_data3_sort$depth_bin, " ", check_data3_sort$Date_Tide)
# 
# #phyla_table_2$depth_bin = as.character(phyla_table_2$depth_bin)
# phyla_table_3 = phyla_table
# phyla_table_3$SampleWDBDT = paste(phyla_table_3$W, " ", phyla_table_3$depth_bin, " ", phyla_table_3$Date_Tide)
# 
# phyla_table_3$SampleWDBDT = factor(phyla_table_3$SampleWDBDT, levels = c(rev(check_data3_sort$SampleWDBDT)))
# levels(phyla_table_3$SampleWDBDT)
# 
# ggplot(phyla_table_3, aes(x = Percent, y = SampleWDBDT)) +
#   geom_bar(stat = "identity", aes(fill = Phylum)) +
#   theme_bw() +
#   theme(strip.text = element_text(size = 11)) +
#   #theme(legend.position = "none") +
#   #scale_y_continuous(limits = c(0, 50)) +
#   theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
#   #theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) + 
#   scale_fill_manual(values = colors_match, name = "Phyla") +
#   labs(x = "Relative Abundance (%)\n", y = "") +
#   theme(panel.border = element_blank())
# 
# ggsave("Phyla_Stacked_DateTide_2024.03.14.png", plot = last_plot(),
#        units = "cm", width = 40, height = 55, dpi = 300)


# only samples already sent for sequencing
mgs = c("240222-H-W1D2", "240222-H-W1D3", "240222-H-W2D2", "240222-H-W2D3", "240222-H-W3D2", "240222-H-W3D3", "240222-H-W3D4", "240222-H-W4D1", "240222-H-W4D3", "250222-L-SW")

ggplot(subset(genera_table, SampleID %in% mgs), aes(x = Percent, y = SampleID)) +
  geom_bar(stat = "identity", aes(fill = Phylum_Genus)) +
  theme_bw() +
  theme(strip.text = element_text(size = 11)) +
  theme(legend.position = "none") +
  #scale_y_continuous(limits = c(0, 50)) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) + 
  scale_fill_manual(values = colors_match, name = "Taxa") +
  labs(x = "Relative Abundance (%)\n", y = "") +
  theme(panel.border = element_blank())

ggsave("MGs_Genera_2024.03.15.png", plot = last_plot(),
       units = "cm", width = 20, height = 5, dpi = 300)



```


```{r genera over 2% of any sample}

#Genera over 2% of any sample
sub_genera_reads_2 = subset(psclust_sum_genera2, Percent > 2 )  
genera_list_check = unique(psclust_sum_genera2$Phylum_Genus) #1778
genera_list_2 = unique(sub_genera_reads_2$Phylum_Genus) #152 with 75 unclassified

sub_genera_reads_2 = subset(psclust_sum_genera2, Percent > 3 )  
#genera_list_check = unique(psclust_sum_genera2$Phylum_Genus) #1778
genera_list_2 = unique(sub_genera_reads_2$Phylum_Genus) #105 

sub_genera_reads_2 = subset(psclust_sum_genera2, Percent > 4 )  
#genera_list_check = unique(psclust_sum_genera2$Phylum_Genus) #1778
genera_list_2 = sort(unique(sub_genera_reads_2$Phylum_Genus)) #77 (17 phyla) 
genera_list_2 = genera_list_2[genera_list_2 != "unclassified Bacteria; unclassified Bacteria"]
genera_list_2 = genera_list_2[genera_list_2 != "unclassified Archaea; unclassified Archaea"]
genera_list_2 # 75/1776

sub_genera = subset(psclust_sum_genera2, Phylum_Genus %in% genera_list_2)
#sub_genera = sub_genera_reads_2[order(-sub_genera_reads_2$Proportion),]

# how much of the reads are they ?
sum(sub_genera$All)/sum(psclust_long$Abundance) # 0.7398896 for 2% ## 0.5984525 for 4%

# store the order for plotting
#genera_list_0.5 = sub_genera_reads$Phylum_Genus

# create an "other"  group
sub_genera_total = sub_genera %>% group_by(SampleID) %>% summarise(Percent = sum(Percent)) # highest is 81.5

sub_genera_other = sub_genera %>% group_by(SampleID) %>% summarise(Percent = 82 - sum(Percent))
sub_genera_other$Phylum_Genus = "Other"

# trim table of phyla over 0.1% #####
sub_genera_0.5 = data.frame(sub_genera$SampleID, sub_genera$Percent, sub_genera$Phylum_Genus)
colnames(sub_genera_0.5) = c("SampleID", "Percent", "Phylum_Genus")

# bind dataframes
psclust_genera2 = rbind(sub_genera_0.5, sub_genera_other)

# merge with metadata
genera_table2 = merge(check_data2, psclust_genera2, by = "SampleID")

# specify order of samples by metadata file
genera_table2$SampleID = as.factor(genera_table2$SampleID)
genera_table2$SampleID = factor(genera_table2$SampleID, levels = c(rev(meta$SampleID)))
levels(genera_table2$SampleID) # reverse order for plot

genera_list_order = c("Proteobacteria; Aliikangiella",        # separate alpha, beta, gamma proteo!    30 total           
 "Proteobacteria; Amylibacter",                          "Proteobacteria; Candidatus Riegeria",                 
 "Proteobacteria; Cellvibrio"  ,                         "Proteobacteria; Clade Ia"   ,                         
 "Proteobacteria; Colwellia"    ,                        "Proteobacteria; Gallionella" ,                        
 "Proteobacteria; Marinobacterium",                      "Proteobacteria; Motiliproteus",                       
 "Proteobacteria; Oceaniserpentilla",                    "Proteobacteria; Pseudoalteromonas",                   
 "Proteobacteria; Pseudomonas"       ,                   "Proteobacteria; Psychromonas"      ,                  
 "Proteobacteria; Rheinheimera"       ,                  "Proteobacteria; Shewanella"         ,                 
 "Proteobacteria; Simiduia"            ,                 "Proteobacteria; Sulfitobacter"       ,                
 "Proteobacteria; Sulfurifustis"        ,                "Proteobacteria; SUP05 cluster"        ,               
 "Proteobacteria; Thalassotalea"         ,               "Proteobacteria; Thioalkalispira-Sulfurivermis",       
 "Proteobacteria; Thiobacillus"           ,              "Proteobacteria; unclassified Aeromonadaceae"   ,      
 "Proteobacteria; unclassified Gammaproteobacteria",     "Proteobacteria; unclassified Magnetospiraceae"  ,     
 "Proteobacteria; unclassified MBMPE27"             ,    "Proteobacteria; unclassified SAR86 clade"        ,    
 "Proteobacteria; unclassified Thiotrichaceae"       ,   "Proteobacteria; Vibrio"                           ,   
 "Proteobacteria; Woeseia",
 "Nanoarchaeota; unclassified GW2011_GWC1_47_15",       
 "Nanoarchaeota; unclassified SCGC AAA011-D5"  ,         "Nanoarchaeota; unclassified Woesearchaeales",
 "Verrucomicrobiota; Candidatus Omnitrophus",            "Verrucomicrobiota; unclassified Omnitrophales" ,      
 "Verrucomicrobiota; unclassified S-BQ2-57 soil group",
 "Patescibacteria; unclassified JGI 0000069-P22",        "Patescibacteria; unclassified Parcubacteria",
 "Bacteroidota; Algoriphagus"  ,                         "Bacteroidota; Flavobacterium"   ,                     
 "Bacteroidota; Fluviicola"      ,                       "Bacteroidota; Gramella"    ,                          
"Bacteroidota; Maritimimonas"      ,                    "Bacteroidota; NS5 marine group"    ,                  
 "Bacteroidota; unclassified Bacteroidetes VC2.1 Bac22", "Bacteroidota; unclassified Cryomorphaceae"   ,        
"Bacteroidota; unclassified Flavobacteriaceae" ,        "Bacteroidota; unclassified NS9 marine group"  ,       
"Bdellovibrionota; Peredibacter"    ,                   "Bdellovibrionota; unclassified 0319-6G20"    ,        
 "Bdellovibrionota; unclassified Oligoflexaceae",
"Chloroflexi; SCGC-AB-539-J10"    ,                    
 "Chloroflexi; unclassified Anaerolineaceae"  ,          "Chloroflexi; unclassified Dehalococcoidia"   ,        
 "Chloroflexi; unclassified SAR202 clade" ,
"Planctomycetota; Rhodopirellula"  , 
"Crenarchaeota; Candidatus Nitrosopelagicus"   ,       
"Crenarchaeota; Candidatus Nitrosopumilus"  ,           "Crenarchaeota; unclassified Bathyarchaeia"   ,        
"Crenarchaeota; unclassified Nitrosopumilaceae",
"Acidobacteriota; Subgroup 10"  ,                       "Acidobacteriota; unclassified Aminicenantales" ,      
"Acidobacteriota; unclassified Vicinamibacterales",
"Firmicutes; unclassified Bacillaceae"   , "Myxococcota; unclassified bacteriap25"  ,
 "Campylobacterota; Sulfurimonas"  ,                    "Campylobacterota; unclassified Campylobacterales"  ,
"Desulfobacterota; Desulfatiglans" ,                   
 "Desulfobacterota; Desulfobulbus"   ,                   "Desulfobacterota; Geopsychrobacter"   ,               
"Desulfobacterota; unclassified Desulfuromonadaceae" ,
"Actinobacteriota; Candidatus Actinomarina" ,           "Actinobacteriota; unclassified Actinomarinales" ,
"Thermoplasmatota; unclassified Marine Group II"    ,  "Asgardarchaeota; unclassified Lokiarchaeia"   , 
"Other"
)

genera_table2$Phylum_Genus = as.factor(genera_table2$Phylum_Genus)
genera_table2$Phylum_Genus = factor(genera_table2$Phylum_Genus, levels = c(rev(genera_list_order)))
levels(genera_table2$Phylum_Genus) # reverse order for plot

a <- "#040508"
b <- "#5161a6"
c <- "#5b6baf"
d <- "#fafafc"
prot_col1 = colorRampPalette(colors=c(a,b))(15)
prot_col2 = colorRampPalette(colors=c(c,d))(15)

nano_col =c("#744d8b",  "#9779a8", "#b9a6c5")
verruco_col =c("#728794","#aab7be", "#d4dbde")
patesci_col =c( "#357985" ,"#5d939d")
bacteroid_col =c("#e5795b","#ff8766","#ff9375","#ff9f84","#ffab93","#ffb7a3","#ffc3b2","#ffcfc1","#ffdbd1","#ffe7e0")
bdello_col = c("#4c8f82", "#6fa59b", "#93bbb4")
chloro_col = c("#666666", "#7f7f7f", "#999999", "#b2b2b2")
plancto_col = "#66333d"
crena_col =c("#759b85", "#84ae96", "#93c2a7", "#9dc8af") 
acid_col = c("#7a9baf", "#89aec5", "#99c2db") #3
firm_col = "#995c95" #1
myxo_col = "#911d55" #1
campy_col = c("#f6f694", "#fafac3")#2
desulf_col = c("#43364a", "#4b3d53" ,"#5d5064", "#6e6375") #4
actino_col =c ("#c0946d", "#d6a57a")
thermo_col = "#4d6a94" #1
asgard_col = "#9c4f41" #1


colors_genera = c(prot_col1, prot_col2, nano_col, verruco_col, patesci_col, bacteroid_col, bdello_col, chloro_col, plancto_col, crena_col, acid_col, firm_col, myxo_col, campy_col, desulf_col, actino_col, thermo_col, asgard_col, "#191919")

colors_matt_shade =c( "#242526", "#a7b8c2", "#f0ece2", "#dfd3c3", "#c7b198" ,"#9c4f41", "#bf2651", "#4d6a94", "#b3785d", "#e6cc8a", "#997d76", "#ffd3a3" ,"#52216e", "#ffac7f" ,"#ff8766" ,"#e6a3a3", "#d6a57a", "#4b3d53" ,"#911d55", "#fafac3", "#99c2db", "#995c95", "#93c2a7", "#66333d", "#524a63", "#4c8f82" ,"#b2c5c7" ,"#357985" ,"#728794",  "#dfeded","#242b4a")

colors_match =c( "#242526","#7a7d81", "#414244", 
                 "#f0ece2", "#cbd2b8", "#bf2651",
                 "#c7b198" ,"#9c4f41",  "#4d6a94", 
                 "#b3785d", "#e6cc8a", "#997d76", 
                 "#ffac7f", "#ffd3a3" ,"#52216e" ,
                 "#ff8766" ,"#e6a3a3", "#d6a57a", 
                 "#4b2e56" , "#fafac3", "#911d55", 
                 "#995c95","#99c2db", "#78b392", 
                 "#66333d", "#635c72", "#4c8f82" ,
                 "#b2c5c7" ,"#357985" ,"#728794",  
                 "#dfeded","#242b4a")

n_distinct(genera_table2$Phylum_Genus) # 76

ggplot(genera_table2, aes(x = Percent, y = SampleID)) +
  geom_bar(stat = "identity", aes(fill = Phylum_Genus), col = "#191919") +
  theme_bw() +
  theme(strip.text = element_text(size = 11)) +
  theme(legend.position = "none") +
  #scale_y_continuous(limits = c(0, 50)) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) + 
  scale_fill_manual(values = rev(colors_genera), name = "Taxa") +
  labs(x = "Relative Abundance (%)\n", y = "")

ggsave("Genera_4_2024.03.15.png", plot = last_plot(),
       units = "cm", width = 40, height = 55, dpi = 300)

# order by dendrogram
genera_table2_noSand = subset(genera_table2, Type != "Sand")

ggplot(genera_table2_noSand, aes(x = Percent, y = reorder(SampleID, ClusterOrder))) +
  geom_bar(stat = "identity", aes(fill = Phylum_Genus), col = "#191919") +
  theme_bw() +
  theme(strip.text = element_text(size = 11)) +
  theme(legend.position = "none") +
  scale_y_discrete(limits =rev) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) + 
  scale_fill_manual(values = rev(colors_genera), name = "Taxa") +
  labs(x = "Relative Abundance (%)\n", y = "")

ggsave("Genera_4_Dendro_2024.03.15.png", plot = last_plot(),
       units = "cm", width = 40, height = 55, dpi = 300)


# legend
genera_table2$Phylum_Genus = factor(genera_table2$Phylum_Genus, levels = c((genera_list_order)))
ggplot(genera_table2, aes(x = Percent, y = SampleID)) +
  geom_bar(stat = "identity", aes(fill = Phylum_Genus), col = "#191919") +
  theme_bw() +
  theme(strip.text = element_text(size = 11)) +
  #theme(legend.position = "none") +
  #scale_y_continuous(limits = c(0, 50)) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) + 
  scale_fill_manual(values = (colors_genera), name = "Taxa") +
  labs(x = "Relative Abundance (%)\n", y = "")

ggsave("Genera_legend_2024.03.15.png", plot = last_plot(),
       units = "cm", width = 40, height = 55, dpi = 300)

# only samples already sent for sequencing
mgs = c("240222-H-W1D2", "240222-H-W1D3", "240222-H-W2D2", "240222-H-W2D3", "240222-H-W3D2", "240222-H-W3D3", "240222-H-W3D4", "240222-H-W4D1", "240222-H-W4D3", "250222-L-SW")
genera_table2$Phylum_Genus = factor(genera_table2$Phylum_Genus, levels = c(rev(genera_list_order)))

ggplot(subset(genera_table2, SampleID %in% mgs), aes(x = Percent, y = SampleID)) +
  geom_bar(stat = "identity", aes(fill = Phylum_Genus)) +
  theme_bw() +
  theme(strip.text = element_text(size = 11)) +
  theme(legend.position = "none") +
  #scale_y_continuous(limits = c(0, 50)) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) + 
  scale_fill_manual(values = rev(colors_genera), name = "Phyla") +
  labs(x = "Relative Abundance (%)\n", y = "") +
  theme(panel.border = element_blank())

ggsave("MGs_Genera_2024.03.15.png", plot = last_plot(),
       units = "cm", width = 20, height = 5, dpi = 300)

```


# Part 6. ASV-level analysis

## 1. Stacked Bar ASV
```{r}
# define Phylum_Genus_OTU
psclust_long$Phylum_Genus_ASV = paste(psclust_long$Phylum, psclust_long$Genus, psclust_long$OTU, sep = "; ")

# sum read counts by sample ID and phylum
psclust_sum_ASV = psclust_long %>% group_by(SampleID, Phylum_Genus_ASV) %>% summarise(All = sum(Abundance)) # not proportional!

# total reads
sum(psclust_long$Abundance) # 8215467

# ASVs comprising 0.1% of all reads (Ruiz Gonzales 2022)
all_asv_reads = psclust_long %>% group_by(Phylum_Genus_ASV) %>% summarise(Total = sum(Abundance)) 
all_asv_reads$Proportion = (all_asv_reads$Total)/(sum(psclust_long$Abundance))*100
all_asv_reads = subset(all_asv_reads, Total > 0)
n_distinct(all_asv_reads$Phylum_Genus_ASV) #63737 (includes unclassifieds)

# total reads per sample
#psclust_sum_sample = psclust_long %>% group_by(SampleID) %>% summarise(Reads = sum(Abundance)) 
#check_data = data.frame(sample_data(ps))
#check_data2 = merge(psclust_sum_sample, check_data, by = "SampleID")

# merge sample separated with total per sample
psclust_sum_genera2 = left_join(psclust_sum_genera, psclust_sum_sample, by = "SampleID")

# take proportion of sample-phylum per sample total
psclust_sum_genera2$Proportion = (psclust_sum_genera2$All)/(psclust_sum_genera2$Reads)
psclust_sum_genera2$Percent = (psclust_sum_genera2$All)/(psclust_sum_genera2$Reads)*100


####
#Genera over 2% of any sample
sub_genera_reads_2 = subset(psclust_sum_genera2, Percent > 2 )  
genera_list_check = unique(psclust_sum_genera2$Phylum_Genus) #1778
genera_list_2 = unique(sub_genera_reads_2$Phylum_Genus) #152 with 75 unclassified

sub_genera_reads_2 = subset(psclust_sum_genera2, Percent > 3 )  
#genera_list_check = unique(psclust_sum_genera2$Phylum_Genus) #1778
genera_list_2 = unique(sub_genera_reads_2$Phylum_Genus) #105 

sub_genera_reads_2 = subset(psclust_sum_genera2, Percent > 4 )  
#genera_list_check = unique(psclust_sum_genera2$Phylum_Genus) #1778
genera_list_2 = sort(unique(sub_genera_reads_2$Phylum_Genus)) #77 (17 phyla) 
genera_list_2 = genera_list_2[genera_list_2 != "unclassified Bacteria; unclassified Bacteria"]
genera_list_2 = genera_list_2[genera_list_2 != "unclassified Archaea; unclassified Archaea"]
sort(genera_list_2) # 75/1776

sub_genera = subset(psclust_sum_genera2, Phylum_Genus %in% genera_list_2)
#sub_genera = sub_genera_reads_2[order(-sub_genera_reads_2$Proportion),]

# how much of the reads are they ?
sum(sub_genera$All)/sum(psclust_long$Abundance) # 0.1464273

# store the order for plotting
#genera_list_0.5 = sub_genera_reads$Phylum_Genus

# create an "other"  group
sub_genera_total = sub_genera %>% group_by(SampleID) %>% summarise(Percent = sum(Percent)) # highest is 57.9

sub_genera_other = sub_genera %>% group_by(SampleID) %>% summarise(Percent = 58 - sum(Percent))
sub_genera_other$Phylum_Genus = "Other"

# trim table of phyla over 0.1% #####
sub_genera_0.5 = data.frame(sub_genera$SampleID, sub_genera$Percent, sub_genera$Phylum_Genus)
colnames(sub_genera_0.5) = c("SampleID", "Percent", "Phylum_Genus")

# bind dataframes
psclust_genera2 = rbind(sub_genera_0.5, sub_genera_other)

# merge with metadata
genera_table2 = merge(check_data2, psclust_genera2, by = "SampleID")

# specify order of samples by metadata file
genera_table2$SampleID = as.factor(genera_table2$SampleID)
genera_table2$SampleID = factor(genera_table2$SampleID, levels = c(rev(meta$SampleID)))
levels(genera_table2$SampleID) # reverse order for plot

genera_list_order = c( "Proteobacteria; Amylibacter; ASV.12" ,                       
                       "Proteobacteria; Candidatus Riegeria; ASV.687"   ,              
 "Proteobacteria; Cellvibrio; ASV.5"     ,                       
"Proteobacteria; Clade Ia; ASV.15"      ,                       
 "Proteobacteria; Clade Ia; ASV.8"       ,                       
 "Proteobacteria; Colwellia; ASV.119"     ,                      
"Proteobacteria; Colwellia; ASV.322"       ,                    
"Proteobacteria; Gallionella; ASV.31"       ,                   
"Proteobacteria; Gallionella; ASV.87"        ,                  
 "Proteobacteria; Motiliproteus; ASV.230"     ,                  
 "Proteobacteria; Oceaniserpentilla; ASV.138"  ,                 
 "Proteobacteria; Pseudoalteromonas; ASV.123"   ,                
 "Proteobacteria; Pseudoalteromonas; ASV.33"     ,               
 "Proteobacteria; Pseudoalteromonas; ASV.43"      ,              
 "Proteobacteria; Pseudomonas; ASV.16"             ,             
 "Proteobacteria; Pseudomonas; ASV.2"               ,            
"Proteobacteria; Pseudomonas; ASV.254"               ,          
 "Proteobacteria; Psychromonas; ASV.429"              ,          
 "Proteobacteria; Rheinheimera; ASV.40"                ,         
"Proteobacteria; Rheinheimera; ASV.41"                  ,       
"Proteobacteria; Shewanella; ASV.1"                      ,      
 "Proteobacteria; Shewanella; ASV.34"                     ,      
"Proteobacteria; Shewanella; ASV.47"                       ,    
 "Proteobacteria; Shewanella; ASV.7"                        ,    
 "Proteobacteria; Simiduia; ASV.161"                         ,   
 "Proteobacteria; SUP05 cluster; ASV.49"                      ,  
"Proteobacteria; Thalassotalea; ASV.72"                        ,
 "Proteobacteria; Thioalkalispira-Sulfurivermis; ASV.79"        ,
 "Proteobacteria; Thiobacillus; ASV.46"                         ,
 "Proteobacteria; unclassified Aeromonadaceae; ASV.22"          ,
"Proteobacteria; unclassified Gammaproteobacteria; ASV.44"     ,
 "Proteobacteria; unclassified Magnetospiraceae; ASV.168"       ,
 "Proteobacteria; unclassified Thiotrichaceae; ASV.52"          ,
"Proteobacteria; Vibrio; ASV.70"           ,
"Nanoarchaeota; unclassified Woesearchaeales; ASV.13"          ,
"Nanoarchaeota; unclassified Woesearchaeales; ASV.219"         ,
 "Nanoarchaeota; unclassified Woesearchaeales; ASV.234"         ,
"Nanoarchaeota; unclassified Woesearchaeales; ASV.25"          ,
 "Nanoarchaeota; unclassified Woesearchaeales; ASV.28"          ,
 "Verrucomicrobiota; Candidatus Omnitrophus; ASV.67"  ,
"Patescibacteria; unclassified JGI 0000069-P22; ASV.21"        ,
 "Bacteroidota; Flavobacterium; ASV.333"               ,         
"Bacteroidota; Fluviicola; ASV.35"                     ,        
  "Bacteroidota; Gramella; ASV.50"                        ,       
 "Bacteroidota; Maritimimonas; ASV.100"                   ,      
 "Bacteroidota; unclassified Bacteroidetes VC2.1 Bac22; ASV.307",
 "Bacteroidota; unclassified Flavobacteriaceae; ASV.17"         ,
 "Bdellovibrionota; Peredibacter; ASV.121"                      ,
 "Bdellovibrionota; unclassified 0319-6G20; ASV.82"             ,
"Firmicutes; unclassified Bacillaceae; ASV.6"                  ,
"Campylobacterota; Sulfurimonas; ASV.11"                       ,
 "Campylobacterota; Sulfurimonas; ASV.81"                       ,
 "Campylobacterota; Sulfurimonas; ASV.91"                       ,
 "Campylobacterota; unclassified Campylobacterales; ASV.10"     ,
 "Desulfobacterota; Desulfobulbus; ASV.24"                      ,
 "Desulfobacterota; Geopsychrobacter; ASV.226"                  ,
 "Desulfobacterota; unclassified Desulfuromonadaceae; ASV.203"  ,
  "Actinobacteriota; Candidatus Actinomarina; ASV.129"    ,       
 "unclassified Archaea; unclassified Archaea; ASV.153"          ,
"unclassified Bacteria; unclassified Bacteria; ASV.621"        ,
"Other"
)

genera_table2$Phylum_Genus = as.factor(genera_table2$Phylum_Genus)
genera_table2$Phylum_Genus = factor(genera_table2$Phylum_Genus, levels = c(rev(genera_list_order)))
levels(genera_table2$Phylum_Genus) # reverse order for plot

a <- "#040508"
b <- "#5161a6"
c <- "#5b6baf"
d <- "#fafafc"
prot_col1 = colorRampPalette(colors=c(a,b))(17)
prot_col2 = colorRampPalette(colors=c(c,d))(17)

nano_col = colorRampPalette(colors=c("#744d8b","#fafafc"))(5)
verruco_col =c("#728794")
patesci_col =c( "#357985" )
bacteroid_col =c("#e5795b","#ff8766","#ff9375","#ff9f84","#ffab93","#ffb7a3")
bdello_col = c("#4c8f82", "#6fa59b")
firm_col = "#995c95" #1
campy_col =  colorRampPalette(colors=c("#f6f694", "#fafac3"))(4)
desulf_col = c( "#4b3d53" ,"#5d5064", "#6e6375") #4
actino_col =c ("#c0946d")
thermo_col = "#4d6a94" #1
asgard_col = "#9c4f41" #1


colors_genera = c(prot_col1, prot_col2, nano_col, verruco_col, patesci_col, bacteroid_col, bdello_col,  firm_col,  campy_col, desulf_col, actino_col, thermo_col, asgard_col, "#191919")

n_distinct(genera_table2$Phylum_Genus) # 61

ggplot(genera_table2, aes(x = Percent, y = SampleID)) +
  geom_bar(stat = "identity", aes(fill = Phylum_Genus), col = "#191919") +
  theme_bw() +
  theme(strip.text = element_text(size = 11)) +
  theme(legend.position = "none") +
  #scale_y_continuous(limits = c(0, 50)) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) + 
  scale_fill_manual(values = rev(colors_genera), name = "Taxa") +
  labs(x = "Relative Abundance (%)\n", y = "")

ggsave("ASVs_4_2024.03.15.png", plot = last_plot(),
       units = "cm", width = 40, height = 55, dpi = 300)

# order by dendrogram
genera_table2_noSand = subset(genera_table2, Type != "Sand")

ggplot(genera_table2_noSand, aes(x = Percent, y = reorder(SampleID, ClusterOrder))) +
  geom_bar(stat = "identity", aes(fill = Phylum_Genus)) +
  theme_bw() +
  theme(strip.text = element_text(size = 11)) +
  theme(legend.position = "none") +
  scale_y_discrete(limits =rev) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) + 
  scale_fill_manual(values = rev(colors_genera), name = "Taxa") +
  labs(x = "Relative Abundance (%)\n", y = "")

ggsave("ASVs_4_Dendro_2024.03.15.png", plot = last_plot(),
       units = "cm", width = 40, height = 55, dpi = 300)


# legend
genera_table2$Phylum_Genus = factor(genera_table2$Phylum_Genus, levels = c((genera_list_order)))
ggplot(genera_table2, aes(x = Percent, y = SampleID)) +
  geom_bar(stat = "identity", aes(fill = Phylum_Genus), col = "#191919") +
  theme_bw() +
  theme(strip.text = element_text(size = 11)) +
  #theme(legend.position = "none") +
  #scale_y_continuous(limits = c(0, 50)) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) + 
  scale_fill_manual(values = (colors_genera), name = "Taxa") +
  labs(x = "Relative Abundance (%)\n", y = "")

ggsave("ASVs_legend_2024.03.15.png", plot = last_plot(),
       units = "cm", width = 40, height = 55, dpi = 300)

# only samples already sent for sequencing
mgs = c("240222-H-W1D2", "240222-H-W1D3", "240222-H-W2D2", "240222-H-W2D3", "240222-H-W3D2", "240222-H-W3D3", "240222-H-W3D4", "240222-H-W4D1", "240222-H-W4D3", "250222-L-SW")
genera_table2$Phylum_Genus = factor(genera_table2$Phylum_Genus, levels = c(rev(genera_list_order)))

ggplot(subset(genera_table2, SampleID %in% mgs), aes(x = Percent, y = SampleID)) +
  geom_bar(stat = "identity", aes(fill = Phylum_Genus)) +
  theme_bw() +
  theme(strip.text = element_text(size = 11)) +
  theme(legend.position = "none") +
  #scale_y_continuous(limits = c(0, 50)) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) + 
  scale_fill_manual(values = rev(colors_genera), name = "Phyla") +
  labs(x = "Relative Abundance (%)\n", y = "") +
  theme(panel.border = element_blank())

ggsave("MGs_ASVs_2024.03.15.png", plot = last_plot(),
       units = "cm", width = 20, height = 5, dpi = 300)


```



## 2. DESeq2 to ID Indicator ASVs

```{r table of ASV read counts}
all_asv_reads = data.frame(psclust_long %>% group_by(OTU) %>% summarise(Total = sum(Abundance))) 
all_asv_reads$Percent = (all_asv_reads$Total)/(sum(psclust_long$Abundance))*100
all_asv_reads = subset(all_asv_reads, Total > 0)
n_distinct(all_asv_reads$OTU) # 63344
rownames(all_asv_reads) = all_asv_reads$OTU
```


```{r DESeq ASVs ~ season/campaign}
ps_pw_sw = subset_samples(ps, Type != "Sand")

# convert to deseq
library(DESeq2)
diagdds <- phyloseq_to_deseq2(ps_pw_sw, ~ Campaign)
diagdds <- DESeq(diagdds, test = "Wald", fitType = "parametric", sfType = "poscounts")

# February 2022

incubation = "Campaign"
treat = "February 2022"
control = "October 2022"

# Inspect results with significance threshold
res <- results(diagdds, cooksCutoff = FALSE, contrast = c(incubation, treat, control))
alpha <- 0.05
sigtab <- res[which(res$padj < alpha), ]
#sigtab <- sigtab[which(res$log2FoldChange > 0), ] # only want enriched
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(ps)[rownames(sigtab), ], "matrix"))
write.csv(sigtab, "DESeq_CampaignFeb_ASVs_2024.03.28.csv", row.names = F)

# October 2022

incubation = "Campaign"
treat = "October 2022"
control = "February 2022"

# Inspect results with significance threshold
res <- results(diagdds, cooksCutoff = FALSE, contrast = c(incubation, treat, control))
alpha <- 0.05
sigtab <- res[which(res$padj < alpha), ]
#sigtab <- sigtab[which(res$log2FoldChange > 0), ] # only want enriched
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(ps)[rownames(sigtab), ], "matrix"))
write.csv(sigtab, "DESeq_Campaign_Oct_ASVs_2024.03.28.csv", row.names = F)
```


```{r DESeq ASVs ~ cluster locations}
# Cluster 1
diagdds <- phyloseq_to_deseq2(ps_pw_sw, ~ Cluster1)
diagdds <- DESeq(diagdds, test = "Wald", fitType = "parametric", sfType = "poscounts")

incubation = "Cluster1" # deep
treat = "yes"
control = "no"

res <- results(diagdds, cooksCutoff = FALSE, contrast = c(incubation, treat, control))
alpha <- 0.05
sigtab <- res[which(res$padj < alpha), ]
#sigtab <- sigtab[which(res$log2FoldChange > 0), ] # only want enriched
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(ps_pw_sw)[rownames(sigtab), ], "matrix"), all_asv_reads[rownames(sigtab),])
write.csv(sigtab, "DESeq_Cluster1_deep_2024.04.11.csv", row.names = F)

# Cluster 2
diagdds <- phyloseq_to_deseq2(ps_pw_sw, ~ Cluster2)
diagdds <- DESeq(diagdds, test = "Wald", fitType = "parametric", sfType = "poscounts")

incubation = "Cluster2" # seawater
treat = "yes"
control = "no"

res <- results(diagdds, cooksCutoff = FALSE, contrast = c(incubation, treat, control))
alpha <- 0.05
sigtab <- res[which(res$padj < alpha), ]
#sigtab <- sigtab[which(res$log2FoldChange > 0), ] # only want enriched
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(ps_pw_sw)[rownames(sigtab), ], "matrix"), all_asv_reads[rownames(sigtab),])
write.csv(sigtab, "DESeq_Cluster2_sw_2024.04.11.csv", row.names = F)

# Seawater only
diagdds <- phyloseq_to_deseq2(ps_pw_sw, ~ Type)
diagdds <- DESeq(diagdds, test = "Wald", fitType = "parametric", sfType = "poscounts")

incubation = "Type" # seawater
treat = "Seawater"
control = "Porewater"

res <- results(diagdds, cooksCutoff = FALSE, contrast = c(incubation, treat, control))
alpha <- 0.05
sigtab <- res[which(res$padj < alpha), ]
#sigtab <- sigtab[which(res$log2FoldChange > 0), ] # only want enriched
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(ps_pw_sw)[rownames(sigtab), ], "matrix"), all_asv_reads[rownames(sigtab),])
write.csv(sigtab, "DESeq_Seawater_2024.04.11.csv", row.names = F)

# February Seawater only (vs porewater pre-overtopping)
ps_pre_overtop = subset_samples(ps_pw_sw, SampleID %in% c("230222-L-W1D2",
 "230222-L-W1D3", "230222-L-W2D2", "230222-L-W2D3", "230222-L-W3D2", "230222-L-W3D3", "230222-L-W3D4", "230222-L-W4D2",
 "230222-L-W4D3", "230222-L-W4D4" ,"240222-H-W1D2", "240222-H-W1D3", "240222-H-W1D4" ,"240222-H-W2D2", "240222-H-W2D3",
 "240222-H-W3D2", "240222-H-W3D3", "240222-H-W3D4" ,"240222-H-W4D1", "240222-H-W4D2", "240222-H-W4D3" ,"240222-H-W4D4",
"250222-H-W1D2", "250222-H-W1D3", "250222-H-W1D4" ,"250222-H-W2D2" ,"250222-H-W2D3", "250222-H-W3D2" ,"250222-H-W3D3",
 "250222-H-W3D4" ,"250222-H-W4D1", "250222-H-W4D2", "250222-H-W4D3" ,"250222-H-W4D4" ,"250222-L-SW" ,  "250222-L-W1D2",
"250222-L-W1D3" ,"250222-L-W1D4" ,"250222-L-W2D2" ,"250222-L-W2D3", "250222-L-W3D2", "250222-L-W3D3", "250222-L-W3D4",
"250222-L-W4D2", "250222-L-W4D3", "250222-L-W4D4","010322-H-SW"  ))

diagdds <- phyloseq_to_deseq2(ps_pre_overtop, ~ Type)
diagdds <- DESeq(diagdds, test = "Wald", fitType = "parametric", sfType = "poscounts")

incubation = "Type" # seawater
treat = "Seawater"
control = "Porewater"

res <- results(diagdds, cooksCutoff = FALSE, contrast = c(incubation, treat, control))
alpha <- 0.05
sigtab <- res[which(res$padj < alpha), ]
#sigtab <- sigtab[which(res$log2FoldChange > 0), ] # only want enriched
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(ps_pre_overtop)[rownames(sigtab), ], "matrix"), all_asv_reads[rownames(sigtab),])
write.csv(sigtab, "DESeq_Seawater_PreOvertop_2024.04.11.csv", row.names = F)

# Cluster 3
diagdds <- phyloseq_to_deseq2(ps_pw_sw, ~ Cluster3)
diagdds <- DESeq(diagdds, test = "Wald", fitType = "parametric", sfType = "poscounts")

incubation = "Cluster3" # well 4
treat = "yes"
control = "no"

res <- results(diagdds, cooksCutoff = FALSE, contrast = c(incubation, treat, control))
alpha <- 0.05
sigtab <- res[which(res$padj < alpha), ]
#sigtab <- sigtab[which(res$log2FoldChange > 0), ] # only want enriched
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(ps_pw_sw)[rownames(sigtab), ], "matrix"), all_asv_reads[rownames(sigtab),])
write.csv(sigtab, "DESeq_Cluster3_W4_2024.04.11.csv", row.names = F)

# Cluster 4
diagdds <- phyloseq_to_deseq2(ps_pw_sw, ~ Cluster4)
diagdds <- DESeq(diagdds, test = "Wald", fitType = "parametric", sfType = "poscounts")

incubation = "Cluster4" # intermediate
treat = "yes"
control = "no"

res <- results(diagdds, cooksCutoff = FALSE, contrast = c(incubation, treat, control))
alpha <- 0.05
sigtab <- res[which(res$padj < alpha), ]
#sigtab <- sigtab[which(res$log2FoldChange > 0), ] # only want enriched
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(ps_pw_sw)[rownames(sigtab), ], "matrix"), all_asv_reads[rownames(sigtab),])
write.csv(sigtab, "DESeq_Cluster4_intermediate_2024.04.11.csv", row.names = F)

# Cluster 5
diagdds <- phyloseq_to_deseq2(ps_pw_sw, ~ Cluster5)
diagdds <- DESeq(diagdds, test = "Wald", fitType = "parametric", sfType = "poscounts")

incubation = "Cluster5" # shallow
treat = "yes"
control = "no"

res <- results(diagdds, cooksCutoff = FALSE, contrast = c(incubation, treat, control))
alpha <- 0.05
sigtab <- res[which(res$padj < alpha), ]
#sigtab <- sigtab[which(res$log2FoldChange > 0), ] # only want enriched
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(ps_pw_sw)[rownames(sigtab), ], "matrix"), all_asv_reads[rownames(sigtab),])
write.csv(sigtab, "DESeq_Cluster5_shallow_2024.04.11.csv", row.names = F)
```

## 3. Region Indicator ASVs

```{r indicator ASV heatmap}
# top 15 enriched from each cluster

indicators = c(
  #shallow
  "ASV.64",
  "ASV.209",
  "ASV.104",
  "ASV.65",
  "ASV.114",
  "ASV.121",
"ASV.1434",
"ASV.82",
  "ASV.510",
"ASV.1618",

  #intermediate

"ASV.687",
"ASV.678",
"ASV.643",
"ASV.722",
"ASV.1357",
"ASV.414",
"ASV.760",
"ASV.407",
"ASV.1319",
"ASV.1211",

  #deep,

  "ASV.611",
   "ASV.193",
  "ASV.1218",
  "ASV.538",
  "ASV.203",
"ASV.226",
  "ASV.430",
"ASV.334",
"ASV.349",
"ASV.271",

# well 4 (c2),

"ASV.285",
"ASV.806",
"ASV.974",
"ASV.1029",
"ASV.288",
"ASV.841",
"ASV.146",
"ASV.558",
"ASV.516",
"ASV.964",

# sw,

"ASV.932",
"ASV.263",
"ASV.1116",
"ASV.129",
"ASV.775")


indicator = subset(psclust_long, OTU %in% indicators)

indicator$Taxonomy = paste(indicator$Genus, indicator$OTU, sep = "; ")

#indicator$Well = as.factor(indicator$Well)
#indicator$Well = factor(indicator$Well, levels = c("Well 1", "Well 2", "Well 3", "Well 4", "Seawater"))

#indicator$Depth = as.factor(indicator$Depth)
#indicator$Date_Tide = as.factor(indicator$Date_Tide)


sample_all = psclust_long %>% group_by(SampleID) %>% summarise(All = sum(Abundance))
indicator = left_join(indicator, sample_all, by = "SampleID")
indicator$Proportion = (indicator$Abundance)/(indicator$All)*100


# order OTU
indicator$OTU = factor(indicator$OTU, levels = rev(indicators))

# order SampleID
indicator$SampleID = as.factor(indicator$SampleID)

# geom_tile with sample ID on x axis, ASV on y axis, fill = Proportion
ggplot(indicator, aes(x = reorder(SampleID, ClusterOrder, decreasing = T), y = OTU)) + # change OTU to taxonomy
  geom_tile(aes(fill = Proportion)) + #transform!
  scale_fill_gradient(low = "white", high = "navy") +
  scale_y_discrete() +
  #scale_fill_viridis(option = "magma", direction = -1) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) +
  labs(x = "", y = "") 

ggsave("Indicator_blue_ASVs_2024.06.20.png", plot = last_plot(),
       units = "cm", width = 30, height = 20, dpi = 300)

ggplot(indicator, aes(x = reorder(SampleID, ClusterOrder, decreasing = T), y = OTU)) + # change OTU to taxonomy
  geom_tile(aes(fill = Proportion)) +
  scale_y_discrete() +
  scale_fill_viridis(option = "magma", direction = -1) +
  #scale_fill_gradientn(colours = rev(c("#000004FF", "#000004FF", "#000004FF", "#000004FF", "#000004FF", "#07071DFF", "#160F3BFF", "#29115AFF", "#400F73FF", "#56147DFF", "#6B1D81FF", "#802582FF", "#952C80FF", "#AB337CFF", "#C03A76FF", "#D6456CFF", "#E85362FF", "#F4685CFF", "#FA815FFF", "#FD9A6AFF", "#FEB37BFF", "#FECC8FFF", "#FDE4A6FF", "#FCFDBFFF")), values = c(0.0000000, 0.2826087, 0.5652174, 0.8478261, 1.1304348, 1.4130435, 1.6956522, 1.9782609, 2.2608696, 2.5434783, 2.8260870, 3.1086957,3.3913043, 3.6739130, 3.9565217, 4.2391304, 4.5217391, 4.8043478, 5.0869565, 5.3695652, 5.6521739, 5.9347826, 6.2173913, 6.5000000)/6.5, limits = c(0, 6.5), name = "value") +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) +
  labs(x = "", y = "") 

ggplot(indicator, aes(x = reorder(SampleID, ClusterOrder, decreasing = T), y = OTU)) + # change OTU to taxonomy
  geom_tile(aes(fill = Proportion)) +
  scale_y_discrete() +
  #scale_fill_viridis(option = "magma", direction = -1) +
  scale_fill_gradientn(colours = rev(c("#000004FF", "#000004FF", "#000004FF", "#000004FF", "#000004FF", "#07071DFF", "#160F3BFF", "#29115AFF", "#400F73FF", "#400F73FF", "#56147DFF", "#6B1D81FF", "#802582FF", "#952C80FF", "#AB337CFF", "#C03A76FF", "#C03A76FF", "#D6456CFF",   "#E85362FF","#F4685CFF", "#FA815FFF", "#FD9A6AFF", "#FEB37BFF", "#FCFDBFFF")),
                        values = c(seq(from = 0, to = 6.5, length.out = 24))/6.5, limits = c(0, 6.5),
                        name = "Relative \nAbundance (%)") + # needs to scale to 0-1
  theme_bw() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5)) +
  labs(x = "", y = "") 


ggsave("Indicator_red_ASVs_2024.06.21.png", plot = last_plot(),
       units = "cm", width = 30, height = 20, dpi = 300)


```

## 4. Overtopping SW Tracer ASVs
```{r overtopping}
tracers = c("ASV.12",
"ASV.90",
"ASV.49",
"ASV.101",
"ASV.109",
"ASV.379",
"ASV.155",
"ASV.76",
"ASV.188",
"ASV.42",
"ASV.992",
"ASV.321",
"ASV.15",
"ASV.462",
"ASV.8",
"ASV.926")



overtop = subset(psclust_long, OTU %in% tracers)
overtop = subset(overtop, Campaign == "Winter 2022")
overtop$Taxonomy = paste(overtop$Phylum, overtop$Genus, overtop$OTU, sep = "; ")


overtop$Well = as.factor(overtop$Well)
overtop$Well = factor(overtop$Well, levels = c("Well 1", "Well 2", "Well 3", "Well 4", "Seawater"))

overtop$Depth = as.factor(overtop$Depth)
overtop$Date_Tide = as.factor(overtop$Date_Tide)


sample_all = psclust_long %>% group_by(SampleID) %>% summarise(All = sum(Abundance))
overtop = left_join(overtop, sample_all, by = "SampleID")
overtop$Proportion = (overtop$Abundance)/(overtop$All)*100


ggplot(overtop, aes(x = Date_Tide, y= Proportion)) +
  geom_bar(stat = "identity", aes(fill = Taxonomy)) +
  facet_grid(cols = vars(Well), rows = vars(Depth)) +
  scale_fill_manual(values = viridis(16)) +
  #scale_fill_brewer(palette = "Dark2") +
  theme_bw() +
  labs(x="", y="Relative Abundance (%)\n") +
  theme(legend.title=element_blank()) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5))


ggsave("Overtop_ASVs_2024.04.18.png", plot = last_plot(),
       units = "cm", width = 25, height = 15, dpi = 300)

# save totals
overtop_totals = overtop %>% group_by(SampleID) %>% summarise(Proportion_SW = sum(Proportion), Reads_SW = sum(Abundance))

# merge with sample data
overtop_totals = left_join(overtop_totals, sample_data(ps), by = "SampleID")

write.csv(overtop_totals, "Overtop_ASVs_totals_2024.04.19.csv", row.names = F)



```


